{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "Here we walk through version 1.6 of the DADA2 pipeline on a small multi-sample dataset. Our starting point is a set of Illumina-sequenced paired-end fastq files that have been split (or \"demultiplexed\") by sample and from which the barcodes/adapters have already been removed. The end product is an **amplicon sequence variant (ASV) table**, a higher-resolution analogue of the traditional \"OTU table\", which records the number of times each amplicon sequence variant was observed in each sample. We also assign taxonomy to the output sequences, and demonstrate how the data can be imported into the popular [phyloseq](https://joey711.github.io/phyloseq/) R package for the analysis of microbiome data.\n",
    "\n",
    "-----------------------\n",
    "\n",
    "# Starting point\n",
    "\n",
    "This workflow assumes that your sequencing data meets certain criteria:\n",
    "\n",
    "* Samples have been demultiplexed, i.e. split into individual per-sample fastq files.\n",
    "* Non-biological nucleotides have been removed, e.g. primers, adapters, linkers, etc.\n",
    "* If paired-end sequencing data, the forward and reverse fastq files contain reads in matched order.\n",
    "\n",
    "If these criteria are not true for your data (**are you sure there aren't any primers hanging around?**) you need to remedy those issues before beginning this workflow. See [the FAQ](faq.html) for some recommendations for common issues.\n",
    "\n",
    "# Getting ready\n",
    "\n",
    "First we load the `dada2` package If you don't already it, see the [dada2 installation instructions](dada-installation.html):\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(dada2); packageVersion(\"dada2\")\n",
    "library(readr)\n",
    "\n",
    "library(phyloseq); packageVersion(\"phyloseq\")\n",
    "library(ggplot2); packageVersion(\"ggplot2\")\n",
    "theme_set(theme_bw())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Older versions of this workflow associated with previous release versions or the dada2 R package are also available: [version 1.2](tutorial_1_2.html), [version 1.4](tutorial_1_4.html).*\n",
    "\n",
    "The data we will work with are the same as those used in the [Mothur Miseq SOP](http://www.mothur.org/wiki/MiSeq_SOP). Download the [example data](http://www.mothur.org/w/images/d/d6/MiSeqSOPData.zip) and unzip. These fastq files were generated by amplicon sequencing (Illumina MiSeq, 2x250, V4 region of the 16S rRNA gene) of gut samples collected longitudinally from a mouse post-weaning, and one mock community control. For now just consider them paired-end fastq files to be processed. Define the following path variable so that it points to the extracted directory on **your** machine:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set Paths\n",
    "We need to set up a \"scratch\" directory for saving files that we generate while running dada2, but don't need to save long term."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "scratch.dir = path.expand(\"~/work/scratch/miseq_sop\")\n",
    "dir.create(scratch.dir, recursive = TRUE)\n",
    "\n",
    "miseqsop.dir = file.path(scratch.dir,\"MiSeq_SOP\")\n",
    "\n",
    "silva.url = \"https://zenodo.org/record/1172783/files/silva_nr_v132_train_set.fa.gz\"\n",
    "silva.ref = file.path(scratch.dir, basename(silva.url))\n",
    "# silva.species.ref = \"/data/references/dada/silva_species_assignment_v128.fa.gz\"\n",
    "\n",
    "ps.rds = file.path(scratch.dir, \"miseq_sop.rds\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "miseq_sop_url = \"http://www.mothur.org/w/images/d/d6/MiSeqSOPData.zip\"\n",
    "miseq_sop_data_zip = file.path(scratch.dir,basename(miseq_sop_url))\n",
    "download.file(miseq_sop_url, destfile = miseq_sop_data_zip)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "download.file(miseq_sop_url, destfile = miseq_sop_data_zip)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "download.file(silva.url, destfile = silva.ref)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "list.files(scratch.dir)\n",
    "unzip(miseq_sop_data_zip, exdir=scratch.dir)\n",
    "list.files(miseqsop.dir)\n",
    "unlink(miseq_sop_data_zip)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "list.files(scratch.dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filter and Trim\n",
    "\n",
    "First we read in the names of the fastq files, and perform some string manipulation to get lists of the forward and reverse fastq files in matched order:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Forward and reverse fastq filenames have format: SAMPLENAME_R1_001.fastq and SAMPLENAME_R2_001.fastq\n",
    "fnFs <- sort(list.files(miseqsop.dir, pattern=\"_R1_001.fastq\", full.names = TRUE))\n",
    "fnRs <- sort(list.files(miseqsop.dir, pattern=\"_R2_001.fastq\", full.names = TRUE))\n",
    "# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq\n",
    "sample.names <- sapply(strsplit(basename(fnFs), \"_\"), `[`, 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Examine quality profiles of forward and reverse reads\n",
    "\n",
    "We start by visualizing the quality profiles of the forward reads:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plotQualityProfile(fnFs[1:2])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The forward reads are good quality. We generally advise trimming the last few nucleotides to avoid less well-controlled errors that can arise there. These quality profiles do not suggest that any additional trimming is needed, so we will truncate the forward reads at position 240 (trimming the last 10 nucleotides).\n",
    "\n",
    "Now we visualize the quality profile of the reverse reads:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plotQualityProfile(fnRs[1:2])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The reverse reads are of significantly worse quality, especially at the end, which is common in Illumina sequencing. This isn't too worrisome, as DADA2 incorporates quality information into its error model which makes the algorithm [robust to lower quality sequence](https://twitter.com/bejcal/status/771010634074820608), but trimming as the average qualities crash will improve the algorithm's sensitivity to rare sequence variants. Based on these profiles, we will truncate the reverse reads at position 160 where the quality distribution crashes.\n",
    "\n",
    "\n",
    "## Perform filtering and trimming\n",
    "\n",
    "Assign the filenames for the filtered fastq.gz files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "filt_path <- file.path(scratch.dir, \"filtered\") # Place filtered files in filtered/ subdirectory\n",
    "filtFs <- file.path(filt_path, paste0(sample.names, \"_F_filt.fastq.gz\"))\n",
    "filtRs <- file.path(filt_path, paste0(sample.names, \"_R_filt.fastq.gz\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We'll use standard filtering parameters: `maxN=0` (DADA2 requires no Ns), `truncQ=2`, `rm.phix=TRUE` and `maxEE=2`. The `maxEE` parameter sets the maximum number of \"expected errors\" allowed in a read, which is [a better filter than simply averaging quality scores](http://www.drive5.com/usearch/manual/expected_errors.html).\n",
    "\n",
    "**Filter the forward and reverse reads**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, trimLeft=10, truncLen=c(240,160),\n",
    "              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,\n",
    "              compress=TRUE, multithread=FALSE) # On Windows set multithread=FALSE\n",
    "head(out)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Learn the Error Rates\n",
    "\n",
    "The DADA2 algorithm depends on a parametric error model (`err`) and every amplicon dataset has a different set of error rates. The `learnErrors` method learns the error model from the data, by alternating estimation of the error rates and inference of sample composition until they converge on a jointly consistent solution. As in many optimization problems, the algorithm must begin with an initial guess, for which the maximum possible error rates in this data are used (the error rates if only the most abundant sequence is correct and all the rest are errors)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "errF <- learnErrors(filtFs, multithread=TRUE)\n",
    "errR <- learnErrors(filtRs, multithread=TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It is always worthwhile, as a sanity check if nothing else, to visualize the estimated error rates:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plotErrors(errF, nominalQ=TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The error rates for each possible transition (eg. A->C, A->G, ...) are shown. Points are the observed error rates for each consensus quality score. The black line shows the estimated error rates after convergence. The red line shows the error rates expected under the nominal definition of the Q-value. Here the black line (the estimated rates) fits the observed rates well, and the error rates drop with increased quality as expected. Everything looks reasonable and we proceed with confidence.\n",
    "\n",
    "\n",
    "# Dereplication\n",
    "\n",
    "Dereplication combines all identical sequencing reads into into \"unique sequences\" with a corresponding \"abundance\": the number of reads with that unique sequence. Dereplication substantially reduces computation time by eliminating redundant comparisons.\n",
    "\n",
    "Dereplication in the DADA2 pipeline has one crucial addition from other pipelines: **DADA2 retains a summary of the quality information associated with each unique sequence**. The consensus quality profile of a unique sequence is the average of the positional qualities from the dereplicated reads. These quality profiles inform the error model of the subsequent denoising step, significantly increasing DADA2's accuracy.\n",
    "\n",
    "**Dereplicate the filtered fastq files**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "derepFs <- derepFastq(filtFs, verbose=TRUE)\n",
    "derepRs <- derepFastq(filtRs, verbose=TRUE)\n",
    "# Name the derep-class objects by the sample names\n",
    "names(derepFs) <- sample.names\n",
    "names(derepRs) <- sample.names"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sample Inference\n",
    "\n",
    "We are now ready to apply the core sequence-variant inference algorithm to the dereplicated data. \n",
    "\n",
    "**Infer the sequence variants in each sample**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dadaFs <- dada(derepFs, err=errF, multithread=TRUE)\n",
    "dadaRs <- dada(derepRs, err=errR, multithread=TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Inspecting the dada-class object returned by dada:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dadaFs[[1]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cat(\"blah\", length(dadaFs[[1]]$sequence))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cat(\"The DADA2 algorithm inferred\", \n",
    "    length(dadaFs[[1]]$sequence), \n",
    "    \"real sequence variants from the\", \n",
    "    length(dadaFs[[1]]$map),\n",
    "    \"unique sequences in the first sample\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There is much more to the `dada-class` return object than this (see `help(\"dada-class\")` for some info), including multiple diagnostics about the quality of each inferred sequence variant, but that is beyond the scope of an introductory tutorial.\n",
    "\n",
    "# Merge paired reads\n",
    "\n",
    "Spurious sequence variants are further reduced by merging overlapping reads. The core function here is `mergePairs`, which depends on the forward and reverse reads being in matching order at the time they were dereplicated.\n",
    "\n",
    "**Merge the denoised forward and reverse reads**:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=TRUE)\n",
    "# Inspect the merger data.frame from the first sample\n",
    "head(mergers[[1]])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We now have a `data.frame` for each sample with the merged `$sequence`, its `$abundance`, and the indices of the merged `$forward` and `$reverse` denoised sequences. Paired reads that did not exactly overlap were removed by `mergePairs`.\n",
    "\n",
    "# Construct sequence table\n",
    "\n",
    "We can now construct a sequence table of our mouse samples, a higher-resolution version of the OTU table produced by traditional methods."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "seqtab <- makeSequenceTable(mergers)\n",
    "dim(seqtab)\n",
    "# Inspect distribution of sequence lengths\n",
    "table(nchar(getSequences(seqtab)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The sequence table is a `matrix` with rows corresponding to (and named by) the samples, and columns corresponding to (and named by) the sequence variants. The lengths of our merged sequences all fall within the expected range for this V4 amplicon.\n",
    "\n",
    "\n",
    "# Remove chimeras\n",
    "\n",
    "The core `dada` method removes substitution and indel errors, but chimeras remain. Fortunately, the accuracy of the sequences after denoising makes identifying chimeras simpler than it is when dealing with fuzzy OTUs: all sequences which can be exactly reconstructed as a bimera (two-parent chimera) from more abundant sequences.\n",
    "\n",
    "**Remove chimeric sequences**:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "seqtab.nochim <- removeBimeraDenovo(seqtab, method=\"consensus\", multithread=TRUE, verbose=TRUE)\n",
    "dim(seqtab.nochim)\n",
    "sum(seqtab.nochim)/sum(seqtab)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cat(\"The fraction of chimeras varies based on factors including experimental\",\n",
    "    \"procedures and sample complexity, but can be substantial.\", \n",
    "    \"Here chimeras make up about\", \n",
    "    round(100*(ncol(seqtab)-ncol(seqtab.nochim))/ncol(seqtab)), \n",
    "    \"% of the inferred sequence variants, but those variants account for only about\",\n",
    "    round(100*(sum(seqtab)-sum(seqtab.nochim))/sum(seqtab)),\n",
    "    \"% of the total sequence reads.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Track reads through the pipeline\n",
    "\n",
    "As a final check of our progress, we'll look at the number of reads that made it through each step in the pipeline:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "getN <- function(x) sum(getUniques(x))\n",
    "track <- cbind(out, sapply(dadaFs, getN), sapply(mergers, getN), rowSums(seqtab), rowSums(seqtab.nochim))\n",
    "# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)\n",
    "colnames(track) <- c(\"input\", \"filtered\", \"denoised\", \"merged\", \"tabled\", \"nonchim\")\n",
    "rownames(track) <- sample.names\n",
    "head(track)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Looks good, we kept the majority of our raw reads, and there is no over-large drop associated with any single step.\n",
    "\n",
    "\n",
    "# Assign taxonomy\n",
    "\n",
    "It is common at this point, especially in 16S/18S/ITS amplicon sequencing, to classify sequence variants taxonomically. The DADA2 package provides a native implementation of [the RDP's naive Bayesian classifier](http://www.ncbi.nlm.nih.gov/pubmed/17586664) for this purpose. The `assignTaxonomy` function takes a set of sequences and a training set of taxonomically classified sequences, and outputs the taxonomic assignments with at least `minBoot` bootstrap confidence. \n",
    "\n",
    "We maintain [formatted training fastas for the RDP training set, GreenGenes clustered at 97\\% identity, and the Silva reference database](training.html). For fungal taxonomy, the General Fasta release files from the [UNITE ITS database](https://unite.ut.ee/repository.php) can be used as is. To follow along, download the `silva_nr_v128_train_set.fa.gz` file, and place it in the directory with the fastq files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "taxa <- assignTaxonomy(seqtab.nochim, silva.ref, multithread=TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "\n",
    "**Optional:** The dada2 package also implements a method to make [species level assignments based on **exact matching**](assign.html#species-assignment) between ASVs and sequenced reference strains. Currently species-assignment training fastas are available for the Silva and RDP 16S databases. To follow the optional species addition step, download the `silva_species_assignment_v128.fa.gz` file, and place it in the directory with the fastq files.\n",
    "\n",
    "```{r species}\n",
    "taxa <- addSpecies(taxa, silva.species.ref)\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's inspect the taxonomic assignments:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "taxa.print <- taxa # Removing sequence rownames for display only\n",
    "rownames(taxa.print) <- NULL\n",
    "head(taxa.print)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Unsurprisingly, the Bacteroidetes are well represented among the most abundant taxa in these fecal samples. Few species assignments were made, both because it is often not possible to make unambiguous species assignments from segments of the 16S gene, and because there is surprisingly little coverage of the indigenous mouse gut microbiota in reference databases."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Bonus: Handoff to phyloseq\n",
    "\n",
    "The [phyloseq R package is a powerful framework for further analysis of microbiome data](https://joey711.github.io/phyloseq/). We now demosntrate how to straightforwardly import the tables produced by the DADA2 pipeline into phyloseq. We'll also add the small amount of metadata we have -- the samples are named by the gender (G), mouse subject number (X) and the day post-weaning (Y) it was sampled (eg. GXDY).\n",
    "\n",
    "## Import into phyloseq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can construct a simple sample data.frame based on the filenames. Usually this step would instead involve reading the sample data in from a file.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "samples.out <- rownames(seqtab.nochim)\n",
    "subject <- sapply(strsplit(samples.out, \"D\"), `[`, 1)\n",
    "gender <- substr(subject,1,1)\n",
    "subject <- substr(subject,2,999)\n",
    "day <- as.integer(sapply(strsplit(samples.out, \"D\"), `[`, 2))\n",
    "samdf <- data.frame(Subject=subject, Gender=gender, Day=day)\n",
    "samdf$When <- \"Early\"\n",
    "samdf$When[samdf$Day>100] <- \"Late\"\n",
    "rownames(samdf) <- samples.out"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can now construct a phyloseq object directly from the dada2 outputs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ps <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE), \n",
    "               sample_data(samdf), \n",
    "               tax_table(taxa))\n",
    "ps <- prune_samples(sample_names(ps) != \"Mock\", ps) # Remove mock sample\n",
    "ps"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Any R object can be saved to an RDS file.  It is a good idea to do this for any object that is time consuming to generate and is reasonably small in size.  Even when the object was generated reproducibly, it can be frustrating to wait minutes or hours to regenerate when you are ready to perform downstream analyses.\n",
    "\n",
    "We will do this for out phyloseq object to a file since it is quite small (especially compared to the size of the input FASTQ files), and there were several time consuming computational steps required to generate it.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "write_rds(ps, ps.rds)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can now confirm that it worked!\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "loaded.ps = read_rds(ps.rds)\n",
    "print(loaded.ps)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We are now ready to use phyloseq!\n",
    "\n",
    "## Visualize alpha-diversity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plot_richness(ps, x=\"Day\", measures=c(\"Shannon\", \"Simpson\"), color=\"When\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "No obvious systematic difference in alpha-diversity between early and late samples.\n",
    "\n",
    "## Ordinate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Transform data to proportions as appropriate for Bray-Curtis distances\n",
    "ps.prop <- transform_sample_counts(ps, function(otu) otu/sum(otu))\n",
    "ord.nmds.bray <- ordinate(ps.prop, method=\"NMDS\", distance=\"bray\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plot_ordination(ps.prop, ord.nmds.bray, color=\"When\", title=\"Bray NMDS\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ordination picks out a clear separation between the early and late samples.\n",
    "\n",
    "## Bar plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "top20 <- names(sort(taxa_sums(ps), decreasing=TRUE))[1:20]\n",
    "ps.top20 <- transform_sample_counts(ps, function(OTU) OTU/sum(OTU))\n",
    "ps.top20 <- prune_taxa(top20, ps.top20)\n",
    "plot_bar(ps.top20, x=\"Day\", fill=\"Family\") + facet_wrap(~When, scales=\"free_x\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Nothing glaringly obvious jumps out from the taxonomic distribution of the top 20 sequences to explain the early-late differentiation.\n",
    "\n",
    "This was just a bare bones demonstration of how the data from DADA2 can be easily imported into phyloseq and interrogated. For further examples on the many analyses possible with phyloseq, see [the phyloseq web site](https://joey711.github.io/phyloseq/)!\n",
    "\n",
    "# Session Info\n",
    "Always print `sessionInfo` for reproducibility!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sessionInfo()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "-------------------\n",
    "\n",
    "This tutorial is based on the [Official DADA2 v1.6 Tutorial](https://raw.githubusercontent.com/benjjneb/dada2/gh-pages/tutorial_1_6.Rmd)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
