

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Quality Control &mdash; HTS2019 1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../_static/jquery.js"></script>
        <script type="text/javascript" src="../_static/underscore.js"></script>
        <script type="text/javascript" src="../_static/doctools.js"></script>
        <script type="text/javascript" src="../_static/language_data.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "document", "processClass": "math|output_area"}})</script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> HTS2019
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Slides</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../slides.html">Slide sets</a></li>
</ul>
<p class="caption"><span class="caption-text">Quick Intro of Analysis Tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../janice/1_IntroToCompBoot.html">Duke HTS Summer Course 2018</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/2_Intro_to_Unix_and_bash.html">Introduction to Unix and bash</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/3_IntroductionToR.html">Introduction to R</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/4_Rlibraries_and_bioconductor.html">R libraries and Bioconductor</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/5_Tidyverse_HandsOn-Filled.html">Hands-On Tidyverse - with count data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/6_Statistical_Inference.html">Simulations and Statistical Inference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/7_Statistical_Inference_Solutions.html">Estimation and Hypothesis Testing</a></li>
</ul>
<p class="caption"><span class="caption-text">Unix Shell</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix01_File_And_Directory_Solutions.html">Working with Unix</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix02_FileIO_Solutions.html">File I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix03_File_Storage_Solutions.html">File archival and recovery</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix04_Text_Manipulation_Solutinos.html">Working with text</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix05_Variables_Solutions.html">Unix Variables</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix06_Bash_Bioinformatics.html">Using <code class="docutils literal notranslate"><span class="pre">bash</span></code> in bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix07_Capstone_Exercise.html">Unix capstone exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix07_Capstone_Exercise_Solutions.html">Unix capstone exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix_Appendix01_Regular_Expressions.html">The Unix Shell: Regular Expressions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix_Appendix02_Review.html">Unix Review of Basics</a></li>
</ul>
<p class="caption"><span class="caption-text">R Graphics and Data Manipulation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R00_Review_Basics.html">Basics of R (Review)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R00_Review_Basics.html">Basics of R (Review)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R01_Data_Manipulation_Solutions.html">Data Manipulation in the <code class="docutils literal notranslate"><span class="pre">tidyverse</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R02_Tidying_Data_Solutions.html">Tidy Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R03_FileIO.html">File I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R04_Unsupervised_Learning.html">Unsupervised Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R05_Unsupervised_Learning_More_Examples.html">Unsupervised Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R06_Graphics_Overview.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R07_Graphics_Base.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R08_Graphics_ggplot2.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R09_Graphics_Exercise.html">R Graphics Exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R09_Graphics_Exercise_Solutions.html">R Graphics Exercise (Solutions)</a></li>
</ul>
<p class="caption"><span class="caption-text">Analysis of Pilot Data (Gene Count, DESeq2 &amp; DE Analysis)</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/8_Pilot2019_getstarcounts-KO.html">Get Star Count of the Pilot 2019 Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/00_deseq_prepare_data.html">DESeq2: Create count matrix and metadata data frame</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/01_deseq_dataobject.html">DESeq2: Create Data Objects</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/02_deseq_DE.html">DESeq2: Basic Differential Expression (DE) analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/03_deseq_interaction.html">DESeq2: Basic Differential Expression (DE) analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/04-deseq2-plots.html">Generate figures from DESEq2 analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">Reproducible Analysis (Git)</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../computing/data_handling_demo.html">Data handling Demo</a></li>
<li class="toctree-l1"><a class="reference internal" href="../computing/git_lesson.html">Git Lesson</a></li>
<li class="toctree-l1"><a class="reference internal" href="../computing/git_in_class.html">Git in class</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">HTS2019</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>Quality Control</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/bioinformatics/fastqc.ipynb.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 5ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast,
.nboutput.nblast {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast + .nbinput {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="Quality-Control">
<h1>Quality Control<a class="headerlink" href="#Quality-Control" title="Permalink to this headline">¶</a></h1>
<!--
** Add Intro Text***
--><div class="section" id="Shell-Variables">
<h2>Shell Variables<a class="headerlink" href="#Shell-Variables" title="Permalink to this headline">¶</a></h2>
<p>Assign the variables in this notebook.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">set</span> -u

<span class="c1"># We used these in the last notebook</span>
<span class="nv">DATA_BASE</span><span class="o">=</span><span class="s2">&quot;/data/hts_2019_data&quot;</span>
<span class="nv">RAW_FASTQS</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$DATA_BASE</span><span class="s2">/hts2019_pilot_rawdata&quot;</span>

<span class="c1"># The following are new for this notebook</span>
<span class="nv">CUROUT</span><span class="o">=</span><span class="nv">$HOME</span>/work/scratch/bioinf_intro
<span class="nv">QC</span><span class="o">=</span><span class="nv">$CUROUT</span>/qc_output
</pre></div>
</div>
</div>
</div>
<div class="section" id="Making-New-Directories">
<h2>Making New Directories<a class="headerlink" href="#Making-New-Directories" title="Permalink to this headline">¶</a></h2>
<p>The directories <em>do</em> carry over between notebooks, they are a function of the server, so we only need to make the directories that are new in this notebook</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>mkdir -p <span class="nv">$QC</span>
</pre></div>
</div>
</div>
<p>Now let’s check to be sure that worked. We will run <code class="docutils literal notranslate"><span class="pre">ls</span></code> and check that these directories now exist in the <code class="docutils literal notranslate"><span class="pre">$OUTPUT</span></code> directory.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls <span class="nv">$CUROUT</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-blue-intense-fg ansi-bold">count_out</span>  <span class="ansi-blue-intense-fg ansi-bold">igv</span>     <span class="ansi-blue-intense-fg ansi-bold">qc_output</span>  <span class="ansi-red-intense-fg ansi-bold">stuff_for_igv_shorter_intron.tgz</span>  <span class="ansi-blue-intense-fg ansi-bold">trimmed_fastqs</span>
<span class="ansi-blue-intense-fg ansi-bold">genome</span>     <span class="ansi-blue-intense-fg ansi-bold">myinfo</span>  <span class="ansi-blue-intense-fg ansi-bold">star_out</span>   <span class="ansi-red-intense-fg ansi-bold">stuff_for_igv.tgz</span>
</pre></div></div>
</div>
</div>
<div class="section" id="Fastqc">
<h2>Fastqc<a class="headerlink" href="#Fastqc" title="Permalink to this headline">¶</a></h2>
<p>Now that we have had a quick look at our fastq file, let’s run some basic analysis using a program called fastqc. This will take less than a minute to run. Remember that while “long” jobs are running it will say “In [*]:” in the left margin, once it is done, a number will replace the asterisk.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>fastqc -h
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>

            FastQC - A high throughput sequence QC analysis tool

SYNOPSIS

        fastqc seqfile1 seqfile2 .. seqfileN

    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam]
           [-c contaminant file] seqfile1 .. seqfileN

DESCRIPTION

    FastQC reads a set of sequence files and produces from each one a quality
    control report consisting of a number of different modules, each one of
    which will help to identify a different potential type of problem in your
    data.

    If no files to process are specified on the command line then the program
    will start as an interactive graphical application.  If files are provided
    on the command line then the program will run with no user interaction
    required.  In this mode it is suitable for inclusion into a standardised
    analysis pipeline.

    The options for the program as as follows:

    -h --help       Print this help file and exit

    -v --version    Print the version of the program and exit

    -o --outdir     Create all output files in the specified output directory.
                    Please note that this directory must exist as the program
                    will not create it.  If this option is not set then the
                    output file for each sequence file is created in the same
                    directory as the sequence file which was processed.

    --casava        Files come from raw casava output. Files in the same sample
                    group (differing only by the group number) will be analysed
                    as a set rather than individually. Sequences with the filter
                    flag set in the header will be excluded from the analysis.
                    Files must have the same names given to them by casava
                    (including being gzipped and ending with .gz) otherwise they
                    won&#39;t be grouped together correctly.

    --nofilter      If running with --casava then don&#39;t remove read flagged by
                    casava as poor quality when performing the QC analysis.

    --extract       If set then the zipped output file will be uncompressed in
                    the same directory after it has been created.  By default
                    this option will be set if fastqc is run in non-interactive
                    mode.

    -j --java       Provides the full path to the java binary you want to use to
                    launch fastqc. If not supplied then java is assumed to be in
                    your path.

    --noextract     Do not uncompress the output file after creating it.  You
                    should set this option if you do not wish to uncompress
                    the output when running in non-interactive mode.

    --nogroup       Disable grouping of bases for reads &gt;50bp. All reports will
                    show data for every base in the read.  WARNING: Using this
                    option will cause fastqc to crash and burn if you use it on
                    really long reads, and your plots may end up a ridiculous size.
                    You have been warned!

    -f --format     Bypasses the normal sequence file format detection and
                    forces the program to use the specified format.  Valid
                    formats are bam,sam,bam_mapped,sam_mapped and fastq

    -t --threads    Specifies the number of files which can be processed
                    simultaneously.  Each thread will be allocated 250MB of
                    memory so you shouldn&#39;t run more threads than your
                    available memory will cope with, and not more than
                    6 threads on a 32 bit machine

    -c              Specifies a non-default file which contains the list of
    --contaminants  contaminants to screen overrepresented sequences against.
                    The file must contain sets of named contaminants in the
                    form name[tab]sequence.  Lines prefixed with a hash will
                    be ignored.

    -a              Specifies a non-default file which contains the list of
    --adapters      adapter sequences which will be explicity searched against
                    the library. The file must contain sets of named adapters
                    in the form name[tab]sequence.  Lines prefixed with a hash
                    will be ignored.

    -l              Specifies a non-default file which contains a set of criteria
    --limits        which will be used to determine the warn/error limits for the
                    various modules.  This file can also be used to selectively
                    remove some modules from the output all together.  The format
                    needs to mirror the default limits.txt file found in the
                    Configuration folder.

   -k --kmers       Specifies the length of Kmer to look for in the Kmer content
                    module. Specified Kmer length must be between 2 and 10. Default
                    length is 7 if not specified.

   -q --quiet       Supress all progress messages on stdout and only report errors.

   -d --dir         Selects a directory to be used for temporary files written when
                    generating report images. Defaults to system temp directory if
                    not specified.

BUGS

    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk
    or in www.bioinformatics.babraham.ac.uk/bugzilla/

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="c1"># fastqc --threads 2 --quiet $DEMUX/r1.8A_pilot.fq.gz -o $QC</span>
fastqc --extract <span class="nv">$RAW_FASTQS</span>/21_2019_P_M1_S21_L002_R1_001.fastq.gz -o <span class="nv">$QC</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Started analysis of 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 5% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 10% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 15% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 20% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 25% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 30% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 35% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 40% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 45% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 50% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 55% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 60% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 65% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 70% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 75% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 80% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 85% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 90% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Approx 95% complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
Analysis complete for 21_2019_P_M1_S21_L002_R1_001.fastq.gz
</pre></div></div>
</div>
<p>Once fastqc is done running we can view the results by finding the output in the Jupyter browser, it should be in:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">echo</span> <span class="nv">$QC</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
/home/jovyan/work/scratch/bioinf_intro/qc_output
</pre></div></div>
</div>
</div>
<div class="section" id="MultiQC">
<h2>MultiQC<a class="headerlink" href="#MultiQC" title="Permalink to this headline">¶</a></h2>
<p>FastQC is a useful tool, but it has one problem: it generates one report for each FASTQ file. When you have more than a handful of FASTQs (as most projects will), it is tedious to look at each one, and there is no simple way to compare them.</p>
<p>MultiQC is a solution to this problem. It mines the results from FastQC (and other HTS analysis tools) and generates reports that combine and summarize results for all the FASTQs analyzed.</p>
<div class="section" id="Run-FastQC-on-Multiple-FASTQs">
<h3>Run FastQC on Multiple FASTQs<a class="headerlink" href="#Run-FastQC-on-Multiple-FASTQs" title="Permalink to this headline">¶</a></h3>
<p>Let’s pick a selection of FASTQs to look at - we will run FastQC on the Lane 1 results for samples 10 through 19 (because it is easy to specify these files with a simple command).</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls <span class="nv">$RAW_FASTQS</span>/1?_2019_*_L001_R1_001.fastq.gz
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-intense-fg ansi-bold">/data/hts_2019_data/hts2019_pilot_rawdata/10_2019_P_M1_S10_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">/data/hts_2019_data/hts2019_pilot_rawdata/11_2019_P_M1_S11_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">/data/hts_2019_data/hts2019_pilot_rawdata/12_2019_P_M1_S12_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">/data/hts_2019_data/hts2019_pilot_rawdata/13_2019_P_M1_S13_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">/data/hts_2019_data/hts2019_pilot_rawdata/14_2019_P_M1_S14_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">/data/hts_2019_data/hts2019_pilot_rawdata/15_2019_P_M1_S15_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">/data/hts_2019_data/hts2019_pilot_rawdata/16_2019_P_M1_S16_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">/data/hts_2019_data/hts2019_pilot_rawdata/17_2019_P_M1_S17_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">/data/hts_2019_data/hts2019_pilot_rawdata/18_2019_P_M1_S18_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">/data/hts_2019_data/hts2019_pilot_rawdata/19_2019_P_M1_S19_L001_R1_001.fastq.gz</span>
</pre></div></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">--threads</span> <span class="pre">2</span></code> tells fastqc to run using 2 cores (of the 16 that we all share). In general you should not do this because it is potentially inconsiderate of other users. I am using it so we don’t have to wait too long for this to run during class.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>fastqc --quiet --threads <span class="m">2</span> --extract <span class="nv">$RAW_FASTQS</span>/1?_2019_*_L001_R1_001.fastq.gz -o <span class="nv">$QC</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="Run-MultiQC">
<h3>Run MultiQC<a class="headerlink" href="#Run-MultiQC" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>multiqc -h
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
/home/jovyan/.local/lib/python3.5/site-packages/multiqc/utils/config.py:45: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  configs = yaml.load(f)
/home/jovyan/.local/lib/python3.5/site-packages/multiqc/utils/config.py:51: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  sp = yaml.load(f)
Usage: multiqc [OPTIONS] &lt;analysis directory&gt;

  MultiQC aggregates results from bioinformatics analyses across many
  samples into a single report.

  It searches a given directory for analysis logs and compiles a HTML
  report. It&#39;s a general use tool, perfect for summarising the output from
  numerous bioinformatics tools.

  To run, supply with one or more directory to scan for analysis results. To
  run here, use &#39;multiqc .&#39;

  See http://multiqc.info for more details.

  Author: Phil Ewels (http://phil.ewels.co.uk)

Options:
  -f, --force                     Overwrite any existing reports
  -d, --dirs                      Prepend directory to sample names
  -dd, --dirs-depth INTEGER       Prepend [INT] directories to sample names.
                                  Negative number to take from start of path.
  -s, --fullnames                 Do not clean the sample names (leave as full
                                  file name)
  -i, --title TEXT                Report title. Printed as page header, used
                                  for filename if not otherwise specified.
  -b, --comment TEXT              Custom comment, will be printed at the top
                                  of the report.
  -n, --filename TEXT             Report filename. Use &#39;stdout&#39; to print to
                                  standard out.
  -o, --outdir TEXT               Create report in the specified output
                                  directory.
  -t, --template [default_dev|geo|simple|sections|default]
                                  Report template to use.
  --tag TEXT                      Use only modules which tagged with this
                                  keyword, eg. RNA
  --view-tags, --view_tags        View the available tags and which modules
                                  they load
  -x, --ignore TEXT               Ignore analysis files (glob expression)
  --ignore-samples TEXT           Ignore sample names (glob expression)
  --ignore-symlinks               Ignore symlinked directories and files
  --sample-names PATH             File containing alternative sample names
  -l, --file-list                 Supply a file containing a list of file
                                  paths to be searched, one per row
  -e, --exclude [module name]     Do not use this module. Can specify multiple
                                  times.
  -m, --module [module name]      Use only this module. Can specify multiple
                                  times.
  --data-dir                      Force the parsed data directory to be
                                  created.
  --no-data-dir                   Prevent the parsed data directory from being
                                  created.
  -k, --data-format [json|tsv|yaml]
                                  Output parsed data in a different format.
                                  Default: tsv
  -z, --zip-data-dir              Compress the data directory.
  -p, --export                    Export plots as static images in addition to
                                  the report
  -fp, --flat                     Use only flat plots (static images)
  -ip, --interactive              Use only interactive plots (HighCharts
                                  Javascript)
  --lint                          Use strict linting (validation) to help code
                                  development
  --pdf                           Creates PDF report with &#39;simple&#39; template.
                                  Requires Pandoc to be installed.
  --no-megaqc-upload              Don&#39;t upload generated report to MegaQC,
                                  even if MegaQC options are found
  -c, --config PATH               Specific config file to load, after those in
                                  MultiQC dir / home dir / working dir.
  --cl-config, --cl_config TEXT   Specify MultiQC config YAML on the command
                                  line
  -v, --verbose                   Increase output verbosity.
  -q, --quiet                     Only show log warnings
  --version                       Show the version and exit.
  -h, --help                      Show this message and exit.
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>multiqc <span class="nv">$QC</span> --outdir <span class="nv">$QC</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
/home/jovyan/.local/lib/python3.5/site-packages/multiqc/utils/config.py:45: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  configs = yaml.load(f)
/home/jovyan/.local/lib/python3.5/site-packages/multiqc/utils/config.py:51: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  sp = yaml.load(f)
[INFO   ]         multiqc : This is MultiQC v1.7
[INFO   ]         multiqc : Template    : default
[INFO   ]         multiqc : Searching &#39;/home/jovyan/work/scratch/bioinf_intro/qc_output&#39;
Searching 221 files..  [####################################]  100%
[INFO   ]          fastqc : Found 11 reports
[INFO   ]         multiqc : Compressing plot data
[WARNING]         multiqc : Previous MultiQC output found! Adjusting filenames..
[WARNING]         multiqc : Use -f or --force to overwrite existing reports instead
[INFO   ]         multiqc : Report      : ../../scratch/bioinf_intro/qc_output/multiqc_report_1.html
[INFO   ]         multiqc : Data        : ../../scratch/bioinf_intro/qc_output/multiqc_data_1
[INFO   ]         multiqc : MultiQC complete
</pre></div></div>
</div>
<p>Once multiqc is done running we can view the results by finding the output in the Jupyter browser, it should be in a file named <code class="docutils literal notranslate"><span class="pre">multiqc_report.html</span></code> in :</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">echo</span> <span class="nv">$QC</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
/home/jovyan/work/scratch/bioinf_intro/qc_output
</pre></div></div>
</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>