

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Download and Index Genome &mdash; HTS2019 1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../_static/jquery.js"></script>
        <script type="text/javascript" src="../_static/underscore.js"></script>
        <script type="text/javascript" src="../_static/doctools.js"></script>
        <script type="text/javascript" src="../_static/language_data.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "document", "processClass": "math|output_area"}})</script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> HTS2019
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Slides</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../slides.html">Slide sets</a></li>
</ul>
<p class="caption"><span class="caption-text">Quick Intro of Analysis Tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../janice/1_IntroToCompBoot.html">Duke HTS Summer Course 2018</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/2_Intro_to_Unix_and_bash.html">Introduction to Unix and bash</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/3_IntroductionToR.html">Introduction to R</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/4_Rlibraries_and_bioconductor.html">R libraries and Bioconductor</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/5_Tidyverse_HandsOn-Filled.html">Hands-On Tidyverse - with count data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/6_Statistical_Inference.html">Simulations and Statistical Inference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/7_Statistical_Inference_Solutions.html">Estimation and Hypothesis Testing</a></li>
</ul>
<p class="caption"><span class="caption-text">Unix Shell</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix01_File_And_Directory_Solutions.html">Working with Unix</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix02_FileIO_Solutions.html">File I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix03_File_Storage_Solutions.html">File archival and recovery</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix04_Text_Manipulation_Solutinos.html">Working with text</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix05_Variables_Solutions.html">Unix Variables</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix06_Bash_Bioinformatics.html">Using <code class="docutils literal notranslate"><span class="pre">bash</span></code> in bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix07_Capstone_Exercise.html">Unix capstone exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix07_Capstone_Exercise_Solutions.html">Unix capstone exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix_Appendix01_Regular_Expressions.html">The Unix Shell: Regular Expressions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix_Appendix02_Review.html">Unix Review of Basics</a></li>
</ul>
<p class="caption"><span class="caption-text">R Graphics and Data Manipulation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R00_Review_Basics.html">Basics of R (Review)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R00_Review_Basics.html">Basics of R (Review)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R01_Data_Manipulation_Solutions.html">Data Manipulation in the <code class="docutils literal notranslate"><span class="pre">tidyverse</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R02_Tidying_Data_Solutions.html">Tidy Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R03_FileIO.html">File I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R04_Unsupervised_Learning.html">Unsupervised Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R05_Unsupervised_Learning_More_Examples.html">Unsupervised Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R06_Graphics_Overview.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R07_Graphics_Base.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R08_Graphics_ggplot2.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R09_Graphics_Exercise.html">R Graphics Exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R09_Graphics_Exercise_Solutions.html">R Graphics Exercise (Solutions)</a></li>
</ul>
<p class="caption"><span class="caption-text">Analysis of Pilot Data (Gene Count, DESeq2 &amp; DE Analysis)</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/8_Pilot2019_getstarcounts-KO.html">Get Star Count of the Pilot 2019 Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/00_deseq_prepare_data.html">DESeq2: Create count matrix and metadata data frame</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/01_deseq_dataobject.html">DESeq2: Create Data Objects</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/02_deseq_DE.html">DESeq2: Basic Differential Expression (DE) analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/03_deseq_interaction.html">DESeq2: Basic Differential Expression (DE) analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/04-deseq2-plots.html">Generate figures from DESEq2 analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">Reproducible Analysis (Git)</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../computing/data_handling_demo.html">Data handling Demo</a></li>
<li class="toctree-l1"><a class="reference internal" href="../computing/git_lesson.html">Git Lesson</a></li>
<li class="toctree-l1"><a class="reference internal" href="../computing/git_in_class.html">Git in class</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">HTS2019</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>Download and Index Genome</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/bioinformatics/genome_prep.ipynb.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 5ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast,
.nboutput.nblast {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast + .nbinput {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="Download-and-Index-Genome">
<h1>Download and Index Genome<a class="headerlink" href="#Download-and-Index-Genome" title="Permalink to this headline">¶</a></h1>
<!---
** Add Intro Text***
--><div class="section" id="Shell-Variables">
<h2>Shell Variables<a class="headerlink" href="#Shell-Variables" title="Permalink to this headline">¶</a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="c1"># Source the config script</span>
<span class="nb">source</span> bioinf_intro_config.sh

<span class="c1"># Make the new directories</span>
mkdir -p <span class="nv">$GENOME_DIR</span> <span class="nv">$STAR_OUT</span>

ls <span class="nv">$CUROUT</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-blue-intense-fg ansi-bold">count_out</span>  <span class="ansi-blue-intense-fg ansi-bold">igv</span>     <span class="ansi-blue-intense-fg ansi-bold">qc_output</span>  <span class="ansi-red-intense-fg ansi-bold">stuff_for_igv_shorter_intron.tgz</span>  <span class="ansi-blue-intense-fg ansi-bold">trimmed_fastqs</span>
<span class="ansi-blue-intense-fg ansi-bold">genome</span>     <span class="ansi-blue-intense-fg ansi-bold">myinfo</span>  <span class="ansi-blue-intense-fg ansi-bold">star_out</span>   <span class="ansi-red-intense-fg ansi-bold">stuff_for_igv.tgz</span>
</pre></div></div>
</div>
</div>
<div class="section" id="STAR">
<h2>STAR<a class="headerlink" href="#STAR" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>Publication: <a class="reference external" href="https://doi.org/10.1093/bioinformatics/bts635">https://doi.org/10.1093/bioinformatics/bts635</a></p></li>
<li><p>Github repo: <a class="reference external" href="https://github.com/alexdobin/STAR">https://github.com/alexdobin/STAR</a></p></li>
<li><p>Manual: <a class="reference external" href="https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf">https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf</a></p></li>
</ul>
</div>
<div class="section" id="Reference-Genome-and-Annotation">
<h2>Reference Genome and Annotation<a class="headerlink" href="#Reference-Genome-and-Annotation" title="Permalink to this headline">¶</a></h2>
<p>We have some preparation to do before we can map our data. First we need to download a reference genome and its annotation file. It is very important that <strong>the genome sequence and annotation are the same version</strong>, if they are not, things could go horribly wrong. The best way to ensure that your sequence and annotation are compatible is to download both from the same place, at the same time, and double check that they have the same version number. There are several good places to get genomes
data:</p>
<ul class="simple">
<li><p>Illumina’s website <a class="reference external" href="http://support.illumina.com/sequencing/sequencing_software/igenome.html">http://support.illumina.com/sequencing/sequencing_software/igenome.html</a>.</p></li>
<li><p>NCBI <a class="reference external" href="http://www.ncbi.nlm.nih.gov">http://www.ncbi.nlm.nih.gov</a></p></li>
<li><p>Ensembl <a class="reference external" href="http://www.ensembl.org/info/about/species.html">http://www.ensembl.org/info/about/species.html</a></p></li>
</ul>
<p>Illumina does not have any <em>Cryptococcus neoformans</em> genomes, so we need to look elsewhere.</p>
<div class="section" id="NCBI">
<h3>NCBI<a class="headerlink" href="#NCBI" title="Permalink to this headline">¶</a></h3>
<p>NCBI has most published genomes, but it is a bit tricky to find exactly what we are looking for. A good place to start is the <a class="reference external" href="http://www.ncbi.nlm.nih.gov/assembly/">NCBI Genome Assembly page</a> where we can search for “Cryptococcus neoformans H99”.</p>
<p>But the mapping software that we will be using, STAR, does not like the GFF format that NCBI uses for annotation. We could get the GFF from NCBI and convert it to a format that STAR likes, but it is easier to look elsewhere to see if we can find a GTF formatted file that STAR likes.</p>
</div>
<div class="section" id="Ensembl">
<h3>Ensembl<a class="headerlink" href="#Ensembl" title="Permalink to this headline">¶</a></h3>
<div class="line-block">
<div class="line">Our next stop is Ensembl, which is also difficult to navigate.</div>
<div class="line">1. Start at the <a class="reference external" href="http://fungi.ensembl.org/index.html">Ensembl Fungi</a> page 2. Click on <a class="reference external" href="http://fungi.ensembl.org/species.html">View full list of all Ensembl Fungi species</a> 3. Find “Cryptococcus neoformans var. grubii H99” (it is helpful to type “H99” in the filter box) 4. There is a link to the <a class="reference external" href="ftp://ftp.ensemblgenomes.org/pub/fungi/release-39/fasta/fungi_basidiomycota1_collection/cryptococcus_neoformans_var_grubii_h99">different FASTA files</a>, we want :
<a class="reference external" href="ftp://ftp.ensemblgenomes.org/pub/release-39/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_neoformans_var_grubii_h99/dna/Cryptococcus_neoformans_var_grubii_h99.CNA3.dna.toplevel.fa.gz">ftp://ftp.ensemblgenomes.org/pub/release-39/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_neoformans_var_grubii_h99/dna/Cryptococcus_neoformans_var_grubii_h99.CNA3.dna.toplevel.fa.gz</a> 5. There is also a link to GFF annotation, but we want GTF and the trail has gone cold. h ere is no obvious sign that there is a GTF file, you just have to know where to find it:
<a class="reference external" href="ftp://ftp.ensemblgenomes.org/pub/release-39/fungi/gtf/fungi_basidiomycota1_collection/cryptococcus_neoformans_var_grubii_h99/Cryptococcus_neoformans_var_grubii_h99.CNA3.39.gtf.gz">ftp://ftp.ensemblgenomes.org/pub/release-39/fungi/gtf/fungi_basidiomycota1_collection/cryptococcus_neoformans_var_grubii_h99/Cryptococcus_neoformans_var_grubii_h99.CNA3.39.gtf.gz</a></div>
</div>
</div>
<div class="section" id="Downloading-with-wget">
<h3>Downloading with wget<a class="headerlink" href="#Downloading-with-wget" title="Permalink to this headline">¶</a></h3>
<p>Now we can use the <code class="docutils literal notranslate"><span class="pre">wget</span></code> command to actually download these files. We will get the files from NCBI. Here is what we will want to tell wget:</p>
<ul class="simple">
<li><p>–directory-prefix : the directory to save the files in</p></li>
<li><p>–no-verbose : don’t output a lot of information while downloading</p></li>
<li><p>URL : what to download</p></li>
</ul>
<p>We are going to make a “genome” directory for these files so that things don’t get too messy. Soon we will generate several files based on these that tophat needs. I generally like to keep the original file names, but we are changing the names to make typing easier later.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">echo</span> <span class="nv">$FA_URL</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
ftp://ftp.ensemblgenomes.org/pub/release-39/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_neoformans_var_grubii_h99/dna/Cryptococcus_neoformans_var_grubii_h99.CNA3.dna.toplevel.fa.gz
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">echo</span> <span class="nv">$GTF_URL</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
ftp://ftp.ensemblgenomes.org/pub/release-39/fungi/gtf/fungi_basidiomycota1_collection/cryptococcus_neoformans_var_grubii_h99/Cryptococcus_neoformans_var_grubii_h99.CNA3.39.gtf.gz
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>wget --no-verbose --directory-prefix <span class="si">${</span><span class="nv">GENOME_DIR</span><span class="si">}</span> <span class="nv">$GTF_URL</span>

wget --no-verbose --directory-prefix <span class="si">${</span><span class="nv">GENOME_DIR</span><span class="si">}</span> <span class="nv">$FA_URL</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
2019-06-26 16:05:17 URL: ftp://ftp.ensemblgenomes.org/pub/release-39/fungi/gtf/fungi_basidiomycota1_collection/cryptococcus_neoformans_var_grubii_h99/Cryptococcus_neoformans_var_grubii_h99.CNA3.39.gtf.gz [1796344] -&gt; &#34;/home/jovyan/work/scratch/bioinf_intro/genome/Cryptococcus_neoformans_var_grubii_h99.CNA3.39.gtf.gz&#34; [1]
2019-06-26 16:05:19 URL: ftp://ftp.ensemblgenomes.org/pub/release-39/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_neoformans_var_grubii_h99/dna/Cryptococcus_neoformans_var_grubii_h99.CNA3.dna.toplevel.fa.gz [5922212] -&gt; &#34;/home/jovyan/work/scratch/bioinf_intro/genome/Cryptococcus_neoformans_var_grubii_h99.CNA3.dna.toplevel.fa.gz&#34; [1]
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls <span class="si">${</span><span class="nv">GENOME_DIR</span><span class="si">}</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-intense-fg ansi-bold">Cryptococcus_neoformans_var_grubii_h99.CNA3.39.gtf.gz</span>
<span class="ansi-red-intense-fg ansi-bold">Cryptococcus_neoformans_var_grubii_h99.CNA3.dna.toplevel.fa.gz</span>
</pre></div></div>
</div>
</div>
<div class="section" id="Decompressing-the-reference-files">
<h3>Decompressing the reference files<a class="headerlink" href="#Decompressing-the-reference-files" title="Permalink to this headline">¶</a></h3>
<p>Next we need to uncompress the files using <code class="docutils literal notranslate"><span class="pre">gunzip</span></code></p>
<p>We use <code class="docutils literal notranslate"><span class="pre">gunzip</span> <span class="pre">--force</span></code> here because if the uncompressed file already exists (i.e. if we run this notebook twice without deleting the output), gunzip will hang trying to ask if it should overwrite the existing uncompressed verison of the file.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>gunzip --force <span class="si">${</span><span class="nv">GENOME_DIR</span><span class="si">}</span>/Cryptococcus_neoformans_var_grubii_h99.CNA3.39.gtf.gz
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>gunzip --force <span class="si">${</span><span class="nv">GENOME_DIR</span><span class="si">}</span>/Cryptococcus_neoformans_var_grubii_h99.CNA3.dna.toplevel.fa.gz
</pre></div>
</div>
</div>
<p>And now we have uncompressed versions of the files</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls <span class="si">${</span><span class="nv">GENOME_DIR</span><span class="si">}</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Cryptococcus_neoformans_var_grubii_h99.CNA3.39.gtf
Cryptococcus_neoformans_var_grubii_h99.CNA3.dna.toplevel.fa
</pre></div></div>
</div>
<p>Let’s define variables for the FASTA and GTF files so we have more manageable names to work with</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nv">FASTA</span><span class="o">=</span><span class="si">${</span><span class="nv">GENOME_DIR</span><span class="si">}</span>/Cryptococcus_neoformans_var_grubii_h99.CNA3.dna.toplevel.fa
<span class="nv">GTF</span><span class="o">=</span><span class="si">${</span><span class="nv">GENOME_DIR</span><span class="si">}</span>/Cryptococcus_neoformans_var_grubii_h99.CNA3.39.gtf
</pre></div>
</div>
</div>
<p>Let’s take a quick look at these files. The <code class="docutils literal notranslate"><span class="pre">head</span></code> command shows us the first few lines of a file (default is 10).</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>head <span class="nv">$FASTA</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&gt;1 dna:chromosome chromosome:CNA3:1:1:2291499:1 REF
GAATTCTAAAACAGTTGCATTCTATAATTACAAAATAATTGAAACACTTCCATTCTCTTG
ACTAATATTATTTAATTAGACACCAACTCGACATTCTGTCTTCGACCTATCTTTCTCCTC
TCCCAGTCATCGCCCAGTAGAATTACCAGGCAATGAACCACGGCCTTTCATCCCAACGGC
ACAGCATATGGGTTCACTCCAACAGTGAACCATTCCAAAAGGCCTTGCCTGCGTGGCCAT
CTCCTCACAAACCCACCATCCTGCATCATCTCAGGTATCATACCTTCGATTCCTTCATCA
CCCCAAAGATCTTTCTGTTTGCCTTTGCGATTTGAGTGATACCAAACAAGAGAATACAGA
AAGTTAGGGACAAAGGGGGAGGGCTCTCTGAAGCGGCATGCTCCTCTTTCAACATGAAAA
TGGAAAGCCCCATCATCCATGTGGTATATGGGCAGGCAGCAAACCTTACAAACACCACTT
TTCAGATTTTCCAGAGCATCCAACCAGTCATCAATGTGATCGAGAGGAACCTTTTGACAC
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>head <span class="nv">$GTF</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
#!genome-build CNA3
#!genome-version CNA3
#!genome-date 2015-11
#!genome-build-accession GCA_000149245.3
#!genebuild-last-updated 2015-11
1       ena     gene    100     5645    .       -       .       gene_id &#34;CNAG_04548&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
1       ena     transcript      100     5645    .       -       .       gene_id &#34;CNAG_04548&#34;; transcript_id &#34;AFR92135&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;;
1       ena     exon    5494    5645    .       -       .       gene_id &#34;CNAG_04548&#34;; transcript_id &#34;AFR92135&#34;; exon_number &#34;1&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;; exon_id &#34;AFR92135-1&#34;;
1       ena     CDS     5494    5645    .       -       0       gene_id &#34;CNAG_04548&#34;; transcript_id &#34;AFR92135&#34;; exon_number &#34;1&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;; protein_id &#34;AFR92135&#34;; protein_version &#34;1&#34;;
1       ena     start_codon     5643    5645    .       -       0       gene_id &#34;CNAG_04548&#34;; transcript_id &#34;AFR92135&#34;; exon_number &#34;1&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;;
</pre></div></div>
</div>
<p>Unfortunately the GFF file has long lines, which are wrapping onto the next line, making them hard to read. Another option is to use the command <code class="docutils literal notranslate"><span class="pre">less</span> <span class="pre">-S</span> <span class="pre">${GENOME_DIR}/Cryptococcus_neoformans_var_grubii_h99.CNA3.39.gtf</span></code> in the terminal (after sourcing the configuration script). The “-S” tells <code class="docutils literal notranslate"><span class="pre">less</span></code> to truncate lines instead of wrapping them.</p>
<blockquote>
<div><p class="rubric" id="gtf-a-brief-aside">GTF, a brief aside</p>
<p>You can find one description of the GTF format <a class="reference external" href="https://useast.ensembl.org/info/website/upload/gff.html">here</a>. Unfortunately it is not entirely standard.</p>
</div></blockquote>
</div>
<div class="section" id="Indexing-the-Genome">
<h3>Indexing the Genome<a class="headerlink" href="#Indexing-the-Genome" title="Permalink to this headline">¶</a></h3>
<p>Before we can map reads to the reference genome using STAR, we need to index it. This will generate a transformed version of the genome that allows STAR to efficiently map sequences to it. We run STAR in “genomeGenerate” mode to do this.</p>
<p>So here is how we run STAR for genome indexing:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>STAR <span class="se">\</span>
    --runMode genomeGenerate <span class="se">\</span>
    --genomeDir <span class="nv">$GENOME_DIR</span> <span class="se">\</span>
    --genomeFastaFiles <span class="si">${</span><span class="nv">FASTA</span><span class="si">}</span> <span class="se">\</span>
    --sjdbGTFfile <span class="si">${</span><span class="nv">GTF</span><span class="si">}</span> <span class="se">\</span>
    --outFileNamePrefix <span class="si">${</span><span class="nv">STAR_OUT</span><span class="si">}</span>/genome_ <span class="se">\</span>
    --genomeSAindexNbases <span class="m">11</span> <span class="se">\</span>
    --runThreadN <span class="m">2</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Jun 26 16:05:21 ..... started STAR run
Jun 26 16:05:21 ... starting to generate Genome files
Jun 26 16:05:21 ... starting to sort Suffix Array. This may take a long time...
Jun 26 16:05:21 ... sorting Suffix Array chunks and saving them to disk...
Jun 26 16:05:43 ... loading chunks from disk, packing SA...
Jun 26 16:05:43 ... finished generating suffix array
Jun 26 16:05:43 ... generating Suffix Array index
Jun 26 16:05:45 ... completed Suffix Array index
Jun 26 16:05:45 ..... processing annotations GTF
Jun 26 16:05:46 ..... inserting junctions into the genome indices
Jun 26 16:06:11 ... writing Genome to disk ...
Jun 26 16:06:11 ... writing Suffix Array to disk ...
Jun 26 16:06:11 ... writing SAindex to disk
Jun 26 16:06:11 ..... finished successfully
</pre></div></div>
</div>
<p>STAR has <em>a lot</em> of command line options! So here is what the above command is doing: * –runMode genomeGenerate: index the genome * –genomeDir : output genome index files to this directory * –genomeFastaFiles : genome sequence file (in FASTA format) * –sjdbGTFfile : annotation file (in GTF format) * –outFileNamePrefix : prefix all output files with this string * –genomeSAindexNbases : selects string length for index, needs to be adjusted based on genome size, can also be made smaller
to reduce memory usage at cost of speed * –runThreadN : tells STAR to run using multiple cores. I am using it so we don’t have to wait too long for this to run during class. It is OK to use multiple cores, but before you do this you should be sure that the server is not busy, and even then you should use a reasonable number of cores. Abusing multi-threading is inconsiderate of other users and could crash the server.</p>
<p>Now let’s check to be sure that worked:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls <span class="nv">$GENOME_DIR</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
chrLength.txt
chrNameLength.txt
chrName.txt
chrStart.txt
Cryptococcus_neoformans_var_grubii_h99.CNA3.39.gtf
Cryptococcus_neoformans_var_grubii_h99.CNA3.dna.toplevel.fa
exonGeTrInfo.tab
exonInfo.tab
geneInfo.tab
Genome
genomeParameters.txt
SA
SAindex
sjdbInfo.txt
sjdbList.fromGTF.out.tab
sjdbList.out.tab
transcriptInfo.tab
</pre></div></div>
</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>