

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Prepare Data &mdash; HTS2019 1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../_static/jquery.js"></script>
        <script type="text/javascript" src="../_static/underscore.js"></script>
        <script type="text/javascript" src="../_static/doctools.js"></script>
        <script type="text/javascript" src="../_static/language_data.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "document", "processClass": "math|output_area"}})</script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> HTS2019
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Slides</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../slides.html">Slide sets</a></li>
</ul>
<p class="caption"><span class="caption-text">Quick Intro of Analysis Tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../janice/1_IntroToCompBoot.html">Duke HTS Summer Course 2018</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/2_Intro_to_Unix_and_bash.html">Introduction to Unix and bash</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/3_IntroductionToR.html">Introduction to R</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/4_Rlibraries_and_bioconductor.html">R libraries and Bioconductor</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/5_Tidyverse_HandsOn-Filled.html">Hands-On Tidyverse - with count data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/6_Statistical_Inference.html">Simulations and Statistical Inference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/7_Statistical_Inference_Solutions.html">Estimation and Hypothesis Testing</a></li>
</ul>
<p class="caption"><span class="caption-text">Unix Shell</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix01_File_And_Directory_Solutions.html">Working with Unix</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix02_FileIO_Solutions.html">File I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix03_File_Storage_Solutions.html">File archival and recovery</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix04_Text_Manipulation_Solutinos.html">Working with text</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix05_Variables_Solutions.html">Unix Variables</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix06_Bash_Bioinformatics.html">Using <code class="docutils literal notranslate"><span class="pre">bash</span></code> in bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix07_Capstone_Exercise.html">Unix capstone exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix07_Capstone_Exercise_Solutions.html">Unix capstone exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix_Appendix01_Regular_Expressions.html">The Unix Shell: Regular Expressions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix_Appendix02_Review.html">Unix Review of Basics</a></li>
</ul>
<p class="caption"><span class="caption-text">R Graphics and Data Manipulation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R00_Review_Basics.html">Basics of R (Review)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R00_Review_Basics.html">Basics of R (Review)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R01_Data_Manipulation_Solutions.html">Data Manipulation in the <code class="docutils literal notranslate"><span class="pre">tidyverse</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R02_Tidying_Data_Solutions.html">Tidy Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R03_FileIO.html">File I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R04_Unsupervised_Learning.html">Unsupervised Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R05_Unsupervised_Learning_More_Examples.html">Unsupervised Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R06_Graphics_Overview.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R07_Graphics_Base.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R08_Graphics_ggplot2.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R09_Graphics_Exercise.html">R Graphics Exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R09_Graphics_Exercise_Solutions.html">R Graphics Exercise (Solutions)</a></li>
</ul>
<p class="caption"><span class="caption-text">Analysis of Pilot Data (Gene Count, DESeq2 &amp; DE Analysis)</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/8_Pilot2019_getstarcounts-KO.html">Get Star Count of the Pilot 2019 Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/00_deseq_prepare_data.html">DESeq2: Create count matrix and metadata data frame</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/01_deseq_dataobject.html">DESeq2: Create Data Objects</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/02_deseq_DE.html">DESeq2: Basic Differential Expression (DE) analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/03_deseq_interaction.html">DESeq2: Basic Differential Expression (DE) analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DESeq2/04-deseq2-plots.html">Generate figures from DESEq2 analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">Reproducible Analysis (Git)</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../computing/data_handling_demo.html">Data handling Demo</a></li>
<li class="toctree-l1"><a class="reference internal" href="../computing/git_lesson.html">Git Lesson</a></li>
<li class="toctree-l1"><a class="reference internal" href="../computing/git_in_class.html">Git in class</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">HTS2019</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>Prepare Data</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/bioinformatics/igv_visualization.ipynb.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 5ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast,
.nboutput.nblast {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast + .nbinput {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">source</span> bioinf_intro_config.sh
mkdir -p <span class="nv">$TRIMMED</span> <span class="nv">$STAR_OUT</span> <span class="nv">$IGV_DIR</span>
</pre></div>
</div>
</div>
<div class="section" id="Prepare-Data">
<h1>Prepare Data<a class="headerlink" href="#Prepare-Data" title="Permalink to this headline">¶</a></h1>
<blockquote>
<div><p>This notebook depends on BAM files generated in the <a class="reference internal" href="glob_loop.html#A-globy-pipeline"><span class="std std-ref">A globy pipeline</span></a> section of the <em>Looping with Globs</em> notebook. If you have not already generated the BAM files, please run that notebook now</p>
</div></blockquote>
<p>IGV needs indices for the BAM files. The index allows it to quickly load reads from different parts of the genome.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="k">for</span> BAM in <span class="si">${</span><span class="nv">STAR_OUT</span><span class="si">}</span>/*_Aligned.sortedByCoord.out.bam
    <span class="k">do</span>
        <span class="nb">echo</span> <span class="nv">$BAM</span>
        samtools index <span class="nv">$BAM</span>
    <span class="k">done</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
/home/jovyan/work/scratch/bioinf_intro/star_out/21_2019_P_M1_S21_L001_R1_Aligned.sortedByCoord.out.bam
/home/jovyan/work/scratch/bioinf_intro/star_out/21_2019_P_M1_S21_L001_R1_short_introns_Aligned.sortedByCoord.out.bam
/home/jovyan/work/scratch/bioinf_intro/star_out/21_2019_P_M1_S21_L002_R1_Aligned.sortedByCoord.out.bam
/home/jovyan/work/scratch/bioinf_intro/star_out/21_2019_P_M1_S21_L002_R1_short_introns_Aligned.sortedByCoord.out.bam
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls -ltr <span class="si">${</span><span class="nv">STAR_OUT</span><span class="si">}</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
total 577416
-rw-r--r-- 1 jovyan users    166944 Jun 24 09:50 21_2019_P_M1_S21_L001_R1_short_introns_ReadsPerGene.out.tab
-rw-r--r-- 1 jovyan users 104774284 Jun 24 09:50 21_2019_P_M1_S21_L001_R1_short_introns_Aligned.sortedByCoord.out.bam
-rw-r--r-- 1 jovyan users    918773 Jun 24 09:50 21_2019_P_M1_S21_L001_R1_short_introns_SJ.out.tab
-rw-r--r-- 1 jovyan users       246 Jun 24 09:50 21_2019_P_M1_S21_L001_R1_short_introns_Log.progress.out
-rw-r--r-- 1 jovyan users     18641 Jun 24 09:50 21_2019_P_M1_S21_L001_R1_short_introns_Log.out
-rw-r--r-- 1 jovyan users      1849 Jun 24 09:50 21_2019_P_M1_S21_L001_R1_short_introns_Log.final.out
-rw-r--r-- 1 jovyan users    166808 Jun 24 09:51 21_2019_P_M1_S21_L002_R1_short_introns_ReadsPerGene.out.tab
-rw-r--r-- 1 jovyan users 102828697 Jun 24 09:51 21_2019_P_M1_S21_L002_R1_short_introns_Aligned.sortedByCoord.out.bam
-rw-r--r-- 1 jovyan users    913218 Jun 24 09:51 21_2019_P_M1_S21_L002_R1_short_introns_SJ.out.tab
-rw-r--r-- 1 jovyan users       246 Jun 24 09:51 21_2019_P_M1_S21_L002_R1_short_introns_Log.progress.out
-rw-r--r-- 1 jovyan users     18643 Jun 24 09:51 21_2019_P_M1_S21_L002_R1_short_introns_Log.out
-rw-r--r-- 1 jovyan users      1849 Jun 24 09:51 21_2019_P_M1_S21_L002_R1_short_introns_Log.final.out
-rw-r--r-- 1 jovyan users 168781979 Jun 26 15:11 21_2019_P_M1_S21_L002_R1_Aligned.out.bam
-rw-r--r-- 1 jovyan users   1068750 Jun 26 15:11 multiqc_report.html
drwxr-xr-x 2 jovyan users      4096 Jun 26 15:11 <span class="ansi-blue-intense-fg ansi-bold">multiqc_data</span>
-rw-r--r-- 1 jovyan users     19079 Jun 26 16:06 genome_Log.out
-rw-r--r-- 1 jovyan users    167049 Jun 26 16:29 21_2019_P_M1_S21_L003_R1_ReadsPerGene.out.tab
-rw-r--r-- 1 jovyan users    921520 Jun 26 16:29 21_2019_P_M1_S21_L003_R1_SJ.out.tab
-rw-r--r-- 1 jovyan users       364 Jun 26 16:29 21_2019_P_M1_S21_L003_R1_Log.progress.out
-rw-r--r-- 1 jovyan users     17731 Jun 26 16:29 21_2019_P_M1_S21_L003_R1_Log.out
-rw-r--r-- 1 jovyan users      1849 Jun 26 16:29 21_2019_P_M1_S21_L003_R1_Log.final.out
-rw-r--r-- 1 jovyan users    166942 Jun 26 16:32 21_2019_P_M1_S21_L001_R1_ReadsPerGene.out.tab
-rw-r--r-- 1 jovyan users 104982330 Jun 26 16:32 21_2019_P_M1_S21_L001_R1_Aligned.sortedByCoord.out.bam
-rw-r--r-- 1 jovyan users    923575 Jun 26 16:32 21_2019_P_M1_S21_L001_R1_SJ.out.tab
-rw-r--r-- 1 jovyan users       364 Jun 26 16:32 21_2019_P_M1_S21_L001_R1_Log.progress.out
-rw-r--r-- 1 jovyan users     18155 Jun 26 16:32 21_2019_P_M1_S21_L001_R1_Log.out
-rw-r--r-- 1 jovyan users      1848 Jun 26 16:32 21_2019_P_M1_S21_L001_R1_Log.final.out
-rw-r--r-- 1 jovyan users    166811 Jun 26 16:34 21_2019_P_M1_S21_L002_R1_ReadsPerGene.out.tab
-rw-r--r-- 1 jovyan users 103015847 Jun 26 16:34 21_2019_P_M1_S21_L002_R1_Aligned.sortedByCoord.out.bam
-rw-r--r-- 1 jovyan users    918360 Jun 26 16:34 21_2019_P_M1_S21_L002_R1_SJ.out.tab
-rw-r--r-- 1 jovyan users       364 Jun 26 16:34 21_2019_P_M1_S21_L002_R1_Log.progress.out
-rw-r--r-- 1 jovyan users     18155 Jun 26 16:34 21_2019_P_M1_S21_L002_R1_Log.out
-rw-r--r-- 1 jovyan users      1848 Jun 26 16:34 21_2019_P_M1_S21_L002_R1_Log.final.out
-rw-r--r-- 1 jovyan users     65344 Jun 26 16:40 21_2019_P_M1_S21_L001_R1_Aligned.sortedByCoord.out.bam.bai
-rw-r--r-- 1 jovyan users     32536 Jun 26 16:40 21_2019_P_M1_S21_L001_R1_short_introns_Aligned.sortedByCoord.out.bam.bai
-rw-r--r-- 1 jovyan users     63664 Jun 26 16:40 21_2019_P_M1_S21_L002_R1_Aligned.sortedByCoord.out.bam.bai
-rw-r--r-- 1 jovyan users     32248 Jun 26 16:40 21_2019_P_M1_S21_L002_R1_short_introns_Aligned.sortedByCoord.out.bam.bai
</pre></div></div>
</div>
</div>
<div class="section" id="Downloading-Everything">
<h1>Downloading Everything<a class="headerlink" href="#Downloading-Everything" title="Permalink to this headline">¶</a></h1>
<div class="section" id="Downloading-files-through-Jupyter">
<h2>Downloading files through Jupyter<a class="headerlink" href="#Downloading-files-through-Jupyter" title="Permalink to this headline">¶</a></h2>
<p>We need to download the following files: 1. BAM file(s) 2. Index for each BAM file 3. Genome sequence (FASTA) 4. Genome annotation (GTF)</p>
<div class="section" id="Jupyter-File-Browser">
<h3>Jupyter File Browser<a class="headerlink" href="#Jupyter-File-Browser" title="Permalink to this headline">¶</a></h3>
<p>For each file: 1. Select checkbox next to filename 2. Click download</p>
</div>
<div class="section" id="Packaging-up-files">
<h3>Packaging up files<a class="headerlink" href="#Packaging-up-files" title="Permalink to this headline">¶</a></h3>
<p>Because we want to download several files, it might be easier to package them up using a program called <code class="docutils literal notranslate"><span class="pre">tar</span></code>, then download the resulting package file (commonly called a tarball) containing all of the files we need.</p>
<div class="section" id="Link-Directory">
<h4>Link Directory<a class="headerlink" href="#Link-Directory" title="Permalink to this headline">¶</a></h4>
<p>First we will do a little hack - we will create a directory of links to all the files we need to download. Since the original files are in two different directories, we are essentially makig a single “virtual directory” containing all the files.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ln -s <span class="si">${</span><span class="nv">STAR_OUT</span><span class="si">}</span>/*.bam* <span class="nv">$GENOME_DIR</span>/<span class="nv">$GTF</span> <span class="nv">$GENOME_DIR</span>/<span class="nv">$FA</span> <span class="nv">$IGV_DIR</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="Taring">
<h4>Taring<a class="headerlink" href="#Taring" title="Permalink to this headline">¶</a></h4>
<p>Here are the command line options we will use with <code class="docutils literal notranslate"><span class="pre">tar</span></code></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">--dereference</span></code> treat the soft-links in our virtual directory as if they were the files that are linked to</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--create</span></code> we are creating a tarball, not unpackaging it</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--gzip</span></code> tells tar to also gzip (compress) the file</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--verbose</span></code> tell us what is happening while running</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--file</span> <span class="pre">TARBALL_NAME</span></code> tells tar what to name the tarball it is creating</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--directory</span> <span class="pre">PATH</span></code> the base directory for the files to be tarred</p></li>
<li><p>FILE[S]_TO_PACKAGE</p></li>
</ul>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>tar --dereference <span class="se">\</span>
    --create <span class="se">\</span>
    --gzip <span class="se">\</span>
    --verbose <span class="se">\</span>
    --file <span class="nv">$CUROUT</span>/stuff_for_igv.tgz <span class="se">\</span>
    --directory <span class="nv">$CUROUT</span> <span class="se">\</span>
    <span class="k">$(</span>basename <span class="nv">$IGV_DIR</span><span class="k">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
igv/
igv/21_2019_P_M1_S21_L001_R1_Aligned.sortedByCoord.out.bam
igv/21_2019_P_M1_S21_L002_R1_Aligned.out.bam
igv/Cryptococcus_neoformans_var_grubii_h99.CNA3.39.gtf
igv/21_2019_P_M1_S21_L001_R1_Aligned.sortedByCoord.out.bam.bai
igv/21_2019_P_M1_S21_L002_R1_Aligned.sortedByCoord.out.bam.bai
igv/21_2019_P_M1_S21_L002_R1_short_introns_Aligned.sortedByCoord.out.bam
igv/21_2019_P_M1_S21_L001_R1_short_introns_Aligned.sortedByCoord.out.bam.bai
igv/21_2019_P_M1_S21_L002_R1_Aligned.sortedByCoord.out.bam
igv/Cryptococcus_neoformans_var_grubii_h99.CNA3.dna.toplevel.fa
igv/21_2019_P_M1_S21_L002_R1_short_introns_Aligned.sortedByCoord.out.bam.bai
igv/21_2019_P_M1_S21_L001_R1_short_introns_Aligned.sortedByCoord.out.bam
</pre></div></div>
</div>
<p>Let’s check that it worked …</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">echo</span> <span class="nv">$CUROUT</span>
ls <span class="nv">$CUROUT</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
/home/jovyan/work/scratch/bioinf_intro
<span class="ansi-blue-intense-fg ansi-bold">count_out</span>  <span class="ansi-blue-intense-fg ansi-bold">igv</span>     <span class="ansi-blue-intense-fg ansi-bold">qc_output</span>  <span class="ansi-red-intense-fg ansi-bold">stuff_for_igv_shorter_intron.tgz</span>  <span class="ansi-blue-intense-fg ansi-bold">trimmed_fastqs</span>
<span class="ansi-blue-intense-fg ansi-bold">genome</span>     <span class="ansi-blue-intense-fg ansi-bold">myinfo</span>  <span class="ansi-blue-intense-fg ansi-bold">star_out</span>   <span class="ansi-red-intense-fg ansi-bold">stuff_for_igv.tgz</span>
</pre></div></div>
</div>
</div>
<div class="section" id="Download-the-tarball">
<h4>Download the tarball<a class="headerlink" href="#Download-the-tarball" title="Permalink to this headline">¶</a></h4>
<p>Now you can do one of the following to download the tarball to your laptop:</p>
<ol class="arabic simple">
<li><p>Click on the “Jupyter” logo above to open the Jupyter file browser</p></li>
<li><p>Naviagte your way to the directory where we saved the tarball (see <code class="docutils literal notranslate"><span class="pre">echo</span> <span class="pre">$CUROUT</span></code> above)</p></li>
<li><p>Click the checkbox next to <code class="docutils literal notranslate"><span class="pre">stuff_for_igv.tgz</span></code>, then click on the <em>Download</em> button near the top of the JUpyter windown to download it.</p></li>
</ol>
</div>
<div class="section" id="Unpacking-our-tarball">
<h4>Unpacking our tarball<a class="headerlink" href="#Unpacking-our-tarball" title="Permalink to this headline">¶</a></h4>
<p>On a Mac you can “untar” by double clicking on the file in finder, or at the terminal with the command <code class="docutils literal notranslate"><span class="pre">tar</span> <span class="pre">-zxf</span> <span class="pre">my_notebooks.tgz</span></code>.</p>
<p>On Windows, you can download software that will do it, such as <a class="reference external" href="http://www.7-zip.org/">7-Zip</a></p>
</div>
</div>
</div>
<div class="section" id="Download-IGV">
<h2>Download IGV<a class="headerlink" href="#Download-IGV" title="Permalink to this headline">¶</a></h2>
<p>It is often helpful to use visualization software to interact with an assembly. We will be using Integrative Genomics Viewer (IGV) because it is pretty good, somewhat user friendly, and cross-platform. We need to download Integrative Genomics Viewer (IGV) for visualizing reads on our laptops. See instructions below for the type of computer you are using.</p>
<div class="section" id="OS-X-(Macs)">
<h3>OS X (Macs)<a class="headerlink" href="#OS-X-(Macs)" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple">
<li><p><a class="reference external" href="http://data.broadinstitute.org/igv/projects/downloads/2.4/IGV_2.4.13.app.zip">a Mac only version</a></p></li>
<li><p><a class="reference external" href="http://data.broadinstitute.org/igv/projects/downloads/2.4/IGV_2.4.13.zip">the cross-platform version</a> ### Windows</p></li>
<li><p><a class="reference external" href="http://data.broadinstitute.org/igv/projects/downloads/2.4/IGV_Win_2.4.13.zip">a Windows only version</a></p></li>
<li><p><a class="reference external" href="http://data.broadinstitute.org/igv/projects/downloads/2.4/IGV_2.4.13.zip">the cross-platform version</a> ### Linux</p></li>
<li><p><a class="reference external" href="http://data.broadinstitute.org/igv/projects/downloads/2.4/IGV_2.4.13.zip">the cross-platform version</a></p></li>
</ol>
</div>
</div>
</div>
<div class="section" id="Using-IGV">
<h1>Using IGV<a class="headerlink" href="#Using-IGV" title="Permalink to this headline">¶</a></h1>
<div class="section" id="Run-IGV">
<h2>Run IGV<a class="headerlink" href="#Run-IGV" title="Permalink to this headline">¶</a></h2>
<p>If you downloaded the Mac specific version, just double click. If you have the cross-platform version: unzip the binary distribution archive in a folder of your choosing. IGV is launched from a command prompt: follow instructions in the “readme” file. To launch igv on Mac or Linux platforms use the shell script “igv.sh”. On Windows use “igv.bat”.</p>
</div>
<div class="section" id="Load-Files">
<h2>Load Files<a class="headerlink" href="#Load-Files" title="Permalink to this headline">¶</a></h2>
<p>Once IGV is running do the following within IGV: 1. <strong>Genome Sequence:</strong> Genomes-&gt;Load Genome From File: (select the FASTA file we just downloaded from Jupyter) 2. <strong>Annotation:</strong> File-&gt;Load From File: (select the GTF file we just downloaded from Jupyter) 3. <strong>Bamfile:</strong> File-&gt;Load From File: (select the BAM files we just downloaded from Jupyter)</p>
</div>
<div class="section" id="Configurations-to-explore">
<h2>Configurations to explore<a class="headerlink" href="#Configurations-to-explore" title="Permalink to this headline">¶</a></h2>
<ol class="arabic simple">
<li><p>Zoom in until reads are visible</p></li>
<li><p>Right click -&gt; Color alignments by -&gt; first-of-pair strand</p></li>
<li><p>Right click-&gt;Collapsed</p></li>
</ol>
<!---
2. Right click -> View as pairs
--></div>
<div class="section" id="Look-around">
<h2>Look around<a class="headerlink" href="#Look-around" title="Permalink to this headline">¶</a></h2>
<p>A few things to look for: - read strand relative to annotated gene strand - intron-spanning reads - SNPs - Areas with no reads - Coverage depth plot - antisense reads - non-protein-coding RNAs</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>