

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Using bash in bioinformatics &mdash; HTS2019 1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../_static/jquery.js"></script>
        <script type="text/javascript" src="../_static/underscore.js"></script>
        <script type="text/javascript" src="../_static/doctools.js"></script>
        <script type="text/javascript" src="../_static/language_data.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "document", "processClass": "math|output_area"}})</script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Unix capstone exercise" href="Unix07_Capstone_Exercise.html" />
    <link rel="prev" title="Unix Variables" href="Unix05_Variables_Solutions.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> HTS2019
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Slides</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../slides.html">Slide sets</a></li>
</ul>
<p class="caption"><span class="caption-text">Unix Shell</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Unix01_File_And_Directory_Solutions.html">Working with Unix</a></li>
<li class="toctree-l1"><a class="reference internal" href="Unix02_FileIO_Solutions.html">File I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="Unix03_File_Storage_Solutions.html">File archival and recovery</a></li>
<li class="toctree-l1"><a class="reference internal" href="Unix04_Text_Manipulation_Solutinos.html">Working with text</a></li>
<li class="toctree-l1"><a class="reference internal" href="Unix05_Variables_Solutions.html">Unix Variables</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Using <code class="docutils literal notranslate"><span class="pre">bash</span></code> in bioinformatics</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#0.1.-Files,-directories-and-working-with-text">0.1. Files, directories and working with text</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#This-is-the-file-we-will-warm-up-with">This is the file we will warm up with</a></li>
<li class="toctree-l3"><a class="reference internal" href="#We-can-see-the-contents-of-a-file-with-cat">We can see the contents of a file with <code class="docutils literal notranslate"><span class="pre">cat</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#Using-head-and-tail">Using <code class="docutils literal notranslate"><span class="pre">head</span></code> and <code class="docutils literal notranslate"><span class="pre">tail</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#Pipling-to-chain-commnads">Pipling to chain commnads</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Searching-within-a-file-with-grep">Searching within a file with <code class="docutils literal notranslate"><span class="pre">grep</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#Use-a-regular-expression-with-grep">Use a regular expression with <code class="docutils literal notranslate"><span class="pre">grep</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#Get-rid-of-numbers-with-cut">Get rid of numbers with <code class="docutils literal notranslate"><span class="pre">cut</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#Get-rid-of-numbers-with-grep">Get rid of numbers with <code class="docutils literal notranslate"><span class="pre">grep</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#Convert-DNA-to-RNA-with-tr">Convert DNA to RNA with <code class="docutils literal notranslate"><span class="pre">tr</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#Reverse-complement-with-tr">Reverse complement with <code class="docutils literal notranslate"><span class="pre">tr</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#Split-command-over-multiple-lines">Split command over multiple lines</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#0.2-Getting-files">0.2 Getting files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#0.3-Compressing-ard-archiving-files">0.3 Compressing ard archiving files</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Compressing-with-gzip">Compressing with <code class="docutils literal notranslate"><span class="pre">gzip</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#Uncompressing-with-gzip">Uncompressing with <code class="docutils literal notranslate"><span class="pre">gzip</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#Compressing-with-bgzip">Compressing with <code class="docutils literal notranslate"><span class="pre">bgzip</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#Note-that-bgzip-files-can-be-uncompressed-with-gzip">Note that <code class="docutils literal notranslate"><span class="pre">bgzip</span></code> files can be uncompressed with <code class="docutils literal notranslate"><span class="pre">gzip</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#Viewing-compressed-files">Viewing compressed files</a></li>
<li class="toctree-l4"><a class="reference internal" href="#Archival-with-tar">Archival with <code class="docutils literal notranslate"><span class="pre">tar</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#View-contents-of-archive">View contents of archive</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Remove-direcotry-and-regenerate-from-archive">Remove direcotry and regenerate from archive</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#0.4-Checksums">0.4 Checksums</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#We-check-that-the-files-are-unchanged-wth-the--c-flag">We check that the files are unchanged wth the <code class="docutils literal notranslate"><span class="pre">-c</span></code> flag</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Suppsoe-one-of-the-files-is-corrupted">Suppsoe one of the files is corrupted</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Using-diff-to-find-changes">Using <code class="docutils literal notranslate"><span class="pre">diff</span></code> to find changes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Verify">Verify</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#1.-Working-with-FASTA-files">1. Working with FASTA files</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Set-up-variables-for-files-we-will-be-using">Set up variables for files we will be using</a></li>
<li class="toctree-l3"><a class="reference internal" href="#This-is-how-you-get-the-value-of-a-shell-variable">This is how you get the value of a shell variable</a></li>
<li class="toctree-l3"><a class="reference internal" href="#List-files-in-data-direcotry">List files in data direcotry</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Inspect-the-contents-of-a-zipped-file.">Inspect the contents of a zipped file.</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Show-headers-only">Show headers only</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Show-headers-with-line-numbers">Show headers with line numbers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Count-number-of-headers-(one-for-each-crhomosome)">Count number of headers (one for each crhomosome)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Extract-first-part-of-the-sequence-from-chromosome-7">Extract first part of the sequence from chromosome 7</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Extract-first-line-of-chromosome-7-and-save-to-file">Extract first line of chromosome 7 and save to file</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Inspect-the-file-we-just-saved">Inspect the file we just saved</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Find-complement">Find complement</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Find-reverse-complement">Find reverse complement</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Delete-A-and-T">Delete A and T</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#Only-count-non-A-and-T">Only count non A and T</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#Count-C-and-G">Count C and G</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Specify-data-directory-as-a-shell-variable">Specify data directory as a shell variable</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#2.-Working-with-FASTQ-files">2. Working with FASTQ files</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Specify-target-file-as-a-shell-variable">Specify target file as a shell variable</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Show-the-contents-of-a-zipped-file">Show the contents of a zipped file</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Extracting-just-the-sequence">Extracting just the sequence</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Joining-lines">Joining lines</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#3.Working-wtih-GTF-files">3.Working wtih GTF files</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Inspect-the-GTF-file">Inspect the GTF file</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Ignore-comment-lines-version-1">Ignore comment lines version 1</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Ingore-comment-lines-2">Ingore comment lines 2</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Find-all-unique-feature-types-in-GTF">Find all unique feature types in GTF</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Get-counts-of-all-unique-feature-types-in-GTF">Get counts of all unique feature types in GTF</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Find-the-the-attributes-of-the-first-3-exons">Find the the attributes of the first 3 exons</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Store-attributes-of-all-genes-in-a-file">Store attributes of all genes in a file</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#4.-Looping">4. Looping</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Loop-throgh-first-10-lines">Loop throgh first 10 lines</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Loop-through-first-10-tRNAs">Loop through first 10 tRNAs</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Loop-through-files-in-a-directory-and-print-the-filename-followed-by-the-first-line">Loop through files in a directory and print the filename followed by the first line</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Unix07_Capstone_Exercise.html">Unix capstone exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="Unix07_Capstone_Exercise_Solutions.html">Unix capstone exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="Unix_Appendix01_Regular_Expressions.html">The Unix Shell: Regular Expressions</a></li>
<li class="toctree-l1"><a class="reference internal" href="Unix_Appendix02_Review.html">Unix Review of Basics</a></li>
</ul>
<p class="caption"><span class="caption-text">R Graphics and Data Manipulation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="R00_Review_Basics.html">Basics of R (Review)</a></li>
<li class="toctree-l1"><a class="reference internal" href="R00_Review_Basics.html">Basics of R (Review)</a></li>
<li class="toctree-l1"><a class="reference internal" href="R01_Data_Manipulation_Solutions.html">Data Manipulation in the <code class="docutils literal notranslate"><span class="pre">tidyverse</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="R02_Tidying_Data_Solutions.html">Tidy Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="R03_FileIO.html">File I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="R04_Unsupervised_Learning.html">Unsupervised Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="R05_Unsupervised_Learning_More_Examples.html">Unsupervised Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="R06_Graphics_Overview.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="R07_Graphics_Base.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="R08_Graphics_ggplot2.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="R09_Graphics_Exercise.html">R Graphics Exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="R09_Graphics_Exercise_Solutions.html">R Graphics Exercise (Solutions)</a></li>
</ul>
<p class="caption"><span class="caption-text">Reproducible Analysis (Git)</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../computing/data_handling_demo.html">Data handling Demo</a></li>
<li class="toctree-l1"><a class="reference internal" href="../computing/git_lesson.html">Git Lesson</a></li>
<li class="toctree-l1"><a class="reference internal" href="../computing/git_in_class.html">Git in class</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">HTS2019</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>Using <code class="docutils literal notranslate"><span class="pre">bash</span></code> in bioinformatics</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/cliburn/Unix06_Bash_Bioinformatics.ipynb.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 5ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast,
.nboutput.nblast {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast + .nbinput {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="Using-bash-in-bioinformatics">
<h1>Using <code class="docutils literal notranslate"><span class="pre">bash</span></code> in bioinformatics<a class="headerlink" href="#Using-bash-in-bioinformatics" title="Permalink to this headline">¶</a></h1>
<p>The Unix commands shown in this notebook are quite intimidating at first. There’s a lot of material becaue I try to cover all the stuff you willl need to construct a typicla bioinformatics pipeline.</p>
<p>Try to understand the expressions and play aound to see what varying them will do. You do not need to memorize the commands now - you can always come back to this notebook for reference. You will also get more practice when you do the Bioiinformatics sessions to construct Unix pipelines for processing your RNA-seq data sets.</p>
<p>Hack to handle broken pipes.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>cleanup <span class="o">()</span> <span class="o">{</span>
    :
<span class="o">}</span>

<span class="nb">trap</span> <span class="s2">&quot;cleanup&quot;</span> SIGPIPE
</pre></div>
</div>
</div>
<p>Treat unset variables as an error, and immediately exit.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">set</span> -u
</pre></div>
</div>
</div>
<div class="section" id="0.1.-Files,-directories-and-working-with-text">
<h2>0.1. Files, directories and working with text<a class="headerlink" href="#0.1.-Files,-directories-and-working-with-text" title="Permalink to this headline">¶</a></h2>
<p>Quickly reveiw what the following navigation and file/directory manipulation commands do:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>&gt;, &gt;&gt;, &lt;, <span class="p">|</span>, ls, cd, pwd, cp, mv, rm, mkdir, rmdir
</pre></div>
</div>
<p>This session dives quite deep into Unix shell scripting and invovles quite a bit of text processing. We review some basic commands for working with text in Unix.</p>
<div class="section" id="This-is-the-file-we-will-warm-up-with">
<h3>This is the file we will warm up with<a class="headerlink" href="#This-is-the-file-we-will-warm-up-with" title="Permalink to this headline">¶</a></h3>
<p>We create it using a HEREDOC that allows us to construct multi-line files and redirrect it to the file <code class="docutils literal notranslate"><span class="pre">junk.txt</span></code></p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>cat &gt; junk.txt <span class="s">&lt;&lt;EOF</span>
<span class="s">1. CCCTAACCCCCTAACCCCCTAACCCTCAGTCGGGGAGGCGACAATAGCTG</span>
<span class="s">2. GTCATATGTTCTGTACGTTATTGGCCAACTGATCATACCTGAATCGAGCC</span>
<span class="s">3. GAACCGGGATTATCAAAGACGAACATGGTCGGGTCCTTGAACCAAACGAA</span>
<span class="s">4. TCTCCGTCCGCTGGCGTGTTTTTCTTTTCTCAAGTGGGCAAGTTACCCGG</span>
<span class="s">EOF</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="We-can-see-the-contents-of-a-file-with-cat">
<h3>We can see the contents of a file with <code class="docutils literal notranslate"><span class="pre">cat</span></code><a class="headerlink" href="#We-can-see-the-contents-of-a-file-with-cat" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>cat junk.txt
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1. CCCTAACCCCCTAACCCCCTAACCCTCAGTCGGGGAGGCGACAATAGCTG
2. GTCATATGTTCTGTACGTTATTGGCCAACTGATCATACCTGAATCGAGCC
3. GAACCGGGATTATCAAAGACGAACATGGTCGGGTCCTTGAACCAAACGAA
4. TCTCCGTCCGCTGGCGTGTTTTTCTTTTCTCAAGTGGGCAAGTTACCCGG
</pre></div></div>
</div>
</div>
<div class="section" id="Using-head-and-tail">
<h3>Using <code class="docutils literal notranslate"><span class="pre">head</span></code> and <code class="docutils literal notranslate"><span class="pre">tail</span></code><a class="headerlink" href="#Using-head-and-tail" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>head -3 junk.txt
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1. CCCTAACCCCCTAACCCCCTAACCCTCAGTCGGGGAGGCGACAATAGCTG
2. GTCATATGTTCTGTACGTTATTGGCCAACTGATCATACCTGAATCGAGCC
3. GAACCGGGATTATCAAAGACGAACATGGTCGGGTCCTTGAACCAAACGAA
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>tail -3 junk.txt
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
2. GTCATATGTTCTGTACGTTATTGGCCAACTGATCATACCTGAATCGAGCC
3. GAACCGGGATTATCAAAGACGAACATGGTCGGGTCCTTGAACCAAACGAA
4. TCTCCGTCCGCTGGCGTGTTTTTCTTTTCTCAAGTGGGCAAGTTACCCGG
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>tail +3 junk.txt
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
3. GAACCGGGATTATCAAAGACGAACATGGTCGGGTCCTTGAACCAAACGAA
4. TCTCCGTCCGCTGGCGTGTTTTTCTTTTCTCAAGTGGGCAAGTTACCCGG
</pre></div></div>
</div>
</div>
<div class="section" id="Pipling-to-chain-commnads">
<h3>Pipling to chain commnads<a class="headerlink" href="#Pipling-to-chain-commnads" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>head -3 junk.txt <span class="p">|</span> tail -2
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
2. GTCATATGTTCTGTACGTTATTGGCCAACTGATCATACCTGAATCGAGCC
3. GAACCGGGATTATCAAAGACGAACATGGTCGGGTCCTTGAACCAAACGAA
</pre></div></div>
</div>
</div>
<div class="section" id="Searching-within-a-file-with-grep">
<h3>Searching within a file with <code class="docutils literal notranslate"><span class="pre">grep</span></code><a class="headerlink" href="#Searching-within-a-file-with-grep" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">grep</span></code> takes a REGEX (regular expression) as the query and a file (or files) to search</p></li>
<li><p>Normally grep shows lines that match the REGEX argument</p></li>
<li><p>The <code class="docutils literal notranslate"><span class="pre">-v</span></code> flag shows the lines that DO NOT match the REGEX argument</p></li>
</ul>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>grep <span class="s2">&quot;[2|4]&quot;</span> junk.txt
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
2. GTCATATGTTCTGTACGTTATTGGCCAACTGATCATACCTGAATCGAGCC
4. TCTCCGTCCGCTGGCGTGTTTTTCTTTTCTCAAGTGGGCAAGTTACCCGG
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>grep -v <span class="s2">&quot;[2|4]&quot;</span> junk.txt
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1. CCCTAACCCCCTAACCCCCTAACCCTCAGTCGGGGAGGCGACAATAGCTG
3. GAACCGGGATTATCAAAGACGAACATGGTCGGGTCCTTGAACCAAACGAA
</pre></div></div>
</div>
</div>
<div class="section" id="Use-a-regular-expression-with-grep">
<h3>Use a regular expression with <code class="docutils literal notranslate"><span class="pre">grep</span></code><a class="headerlink" href="#Use-a-regular-expression-with-grep" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>grep <span class="s2">&quot;CCC.*CCC&quot;</span> junk.txt
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1. CCCTAACCCCCTAACCCCCTAACCCTCAGTCGGGGAGGCGACAATAGCTG
</pre></div></div>
</div>
</div>
<div class="section" id="Get-rid-of-numbers-with-cut">
<h3>Get rid of numbers with <code class="docutils literal notranslate"><span class="pre">cut</span></code><a class="headerlink" href="#Get-rid-of-numbers-with-cut" title="Permalink to this headline">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">cut</span></code> extracts columns separated by delimiters - here the delimiter is a space <code class="docutils literal notranslate"><span class="pre">-d</span> <span class="pre">'</span> <span class="pre">'</span></code> - and we cut the second column <code class="docutils literal notranslate"><span class="pre">-f</span> <span class="pre">2</span></code></p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>cut -d <span class="s1">&#39; &#39;</span> -f <span class="m">2</span> junk.txt
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CCCTAACCCCCTAACCCCCTAACCCTCAGTCGGGGAGGCGACAATAGCTG
GTCATATGTTCTGTACGTTATTGGCCAACTGATCATACCTGAATCGAGCC
GAACCGGGATTATCAAAGACGAACATGGTCGGGTCCTTGAACCAAACGAA
TCTCCGTCCGCTGGCGTGTTTTTCTTTTCTCAAGTGGGCAAGTTACCCGG
</pre></div></div>
</div>
</div>
<div class="section" id="Get-rid-of-numbers-with-grep">
<h3>Get rid of numbers with <code class="docutils literal notranslate"><span class="pre">grep</span></code><a class="headerlink" href="#Get-rid-of-numbers-with-grep" title="Permalink to this headline">¶</a></h3>
<p>The <code class="docutils literal notranslate"><span class="pre">-o</span></code> flag to <code class="docutils literal notranslate"><span class="pre">grep</span></code> shows only the strings mathcing the regular expression</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>grep -o <span class="s1">&#39;[A-Z]*&#39;</span> junk.txt
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CCCTAACCCCCTAACCCCCTAACCCTCAGTCGGGGAGGCGACAATAGCTG
GTCATATGTTCTGTACGTTATTGGCCAACTGATCATACCTGAATCGAGCC
GAACCGGGATTATCAAAGACGAACATGGTCGGGTCCTTGAACCAAACGAA
TCTCCGTCCGCTGGCGTGTTTTTCTTTTCTCAAGTGGGCAAGTTACCCGG
</pre></div></div>
</div>
</div>
<div class="section" id="Convert-DNA-to-RNA-with-tr">
<h3>Convert DNA to RNA with <code class="docutils literal notranslate"><span class="pre">tr</span></code><a class="headerlink" href="#Convert-DNA-to-RNA-with-tr" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>cat junk.txt <span class="p">|</span> tr T U
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1. CCCUAACCCCCUAACCCCCUAACCCUCAGUCGGGGAGGCGACAAUAGCUG
2. GUCAUAUGUUCUGUACGUUAUUGGCCAACUGAUCAUACCUGAAUCGAGCC
3. GAACCGGGAUUAUCAAAGACGAACAUGGUCGGGUCCUUGAACCAAACGAA
4. UCUCCGUCCGCUGGCGUGUUUUUCUUUUCUCAAGUGGGCAAGUUACCCGG
</pre></div></div>
</div>
</div>
<div class="section" id="Reverse-complement-with-tr">
<h3>Reverse complement with <code class="docutils literal notranslate"><span class="pre">tr</span></code><a class="headerlink" href="#Reverse-complement-with-tr" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>grep -o <span class="s1">&#39;[ACTG]*&#39;</span> junk.txt <span class="p">|</span>
tr ACGT TGCA <span class="p">|</span>
rev
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CAGCTATTGTCGCCTCCCCGACTGAGGGTTAGGGGGTTAGGGGGTTAGGG
GGCTCGATTCAGGTATGATCAGTTGGCCAATAACGTACAGAACATATGAC
TTCGTTTGGTTCAAGGACCCGACCATGTTCGTCTTTGATAATCCCGGTTC
CCGGGTAACTTGCCCACTTGAGAAAAGAAAAACACGCCAGCGGACGGAGA
</pre></div></div>
</div>
</div>
<div class="section" id="Split-command-over-multiple-lines">
<h3>Split command over multiple lines<a class="headerlink" href="#Split-command-over-multiple-lines" title="Permalink to this headline">¶</a></h3>
<p>Use <code class="docutils literal notranslate"><span class="pre">\</span></code> to continue command to next line</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>grep -o <span class="s1">&#39;[ACTG]*&#39;</span> junk.txt <span class="p">|</span> tr <span class="se">\</span>
ACGT TGCA <span class="p">|</span> rev
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CAGCTATTGTCGCCTCCCCGACTGAGGGTTAGGGGGTTAGGGGGTTAGGG
GGCTCGATTCAGGTATGATCAGTTGGCCAATAACGTACAGAACATATGAC
TTCGTTTGGTTCAAGGACCCGACCATGTTCGTCTTTGATAATCCCGGTTC
CCGGGTAACTTGCCCACTTGAGAAAAGAAAAACACGCCAGCGGACGGAGA
</pre></div></div>
</div>
<p>Actually the line continuation character is not necesssary if you break at a pipe <code class="docutils literal notranslate"><span class="pre">|</span></code></p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>grep -o <span class="s1">&#39;[ACTG]*&#39;</span> junk.txt <span class="p">|</span>
tr ACGT TGCA <span class="p">|</span>
rev
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CAGCTATTGTCGCCTCCCCGACTGAGGGTTAGGGGGTTAGGGGGTTAGGG
GGCTCGATTCAGGTATGATCAGTTGGCCAATAACGTACAGAACATATGAC
TTCGTTTGGTTCAAGGACCCGACCATGTTCGTCTTTGATAATCCCGGTTC
CCGGGTAACTTGCCCACTTGAGAAAAGAAAAACACGCCAGCGGACGGAGA
</pre></div></div>
</div>
</div>
</div>
<div class="section" id="0.2-Getting-files">
<h2>0.2 Getting files<a class="headerlink" href="#0.2-Getting-files" title="Permalink to this headline">¶</a></h2>
<p>We will download Chromosomes 7 and 8 of Crytococcus neoformans in FASTA format from NCBI Nucleotide. We can find their IDs through a <a class="reference external" href="https://www.ncbi.nlm.nih.gov/nuccore/?term=cryptococcus+neoformans+B-3501A">web query</a>.</p>
<p>We find that the IDs are</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">CM000046</span><span class="o">.</span><span class="mi">1</span> <span class="k">for</span> <span class="n">chormosome</span> <span class="mi">7</span>
<span class="n">CM000047</span><span class="o">.</span><span class="mi">1</span> <span class="k">for</span> <span class="n">chormosome</span> <span class="mi">8</span>
</pre></div>
</div>
<p>We will use the NCBI Entrez tool <code class="docutils literal notranslate"><span class="pre">efetch</span></code> to download the FASTA files to a folder <code class="docutils literal notranslate"><span class="pre">seqs</span></code>.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="c1">### setup efetch command</span>
<span class="nb">cd</span> ~
<span class="c1">#/bin/bash</span>
perl -MNet::FTP -e <span class="se">\</span>
    <span class="s1">&#39;$ftp = new Net::FTP(&quot;ftp.ncbi.nlm.nih.gov&quot;, Passive =&gt; 1);</span>
<span class="s1">     $ftp-&gt;login; $ftp-&gt;binary;</span>
<span class="s1">     $ftp-&gt;get(&quot;/entrez/entrezdirect/edirect.tar.gz&quot;);&#39;</span>
gunzip -c edirect.tar.gz <span class="p">|</span> tar xf -
rm edirect.tar.gz
<span class="c1">#builtin exit</span>
<span class="nb">export</span> <span class="nv">PATH</span><span class="o">=</span><span class="si">${</span><span class="nv">PATH</span><span class="si">}</span>:<span class="nv">$HOME</span>/edirect &gt;<span class="p">&amp;</span> /dev/null <span class="o">||</span> setenv PATH <span class="s2">&quot;</span><span class="si">${</span><span class="nv">PATH</span><span class="si">}</span><span class="s2">:</span><span class="nv">$HOME</span><span class="s2">/edirect&quot;</span>
./edirect/setup.sh
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>

Trying to establish local installations of any missing Perl modules
(as logged in /home/jovyan/edirect/setup-deps.log).
Please be patient, as this step may take a little while.

Entrez Direct has been successfully downloaded and installed.

In order to complete the configuration process, please execute the following:

  echo &#34;export PATH=\${PATH}:/home/jovyan/edirect&#34; &gt;&gt; $HOME/.bashrc

or manually edit the PATH variable assignment in your .bash_profile file.

</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>mkdir -p seqs
efetch -db<span class="o">=</span>nuccore -format<span class="o">=</span>fasta -id<span class="o">=</span>CM000046.1 &gt; seqs/CM000046.1.fa
efetch -db<span class="o">=</span>nuccore -format<span class="o">=</span>fasta -id<span class="o">=</span>CM000047.1 &gt; seqs/CM000047.1.fa
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls seqs
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CM000046.1.fa      <span class="ansi-red-intense-fg ansi-bold">cneo040623.scaffs.fa.gz</span>
CM000047.1.fa      <span class="ansi-red-intense-fg ansi-bold">Cryptococcus_neoformans.ASM9104v1.43.gtf.gz</span>
CM000047.1.fa.bak
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>head seqs/CM000046.1.fa
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&gt;CM000046.1 Cryptococcus neoformans var. neoformans B-3501A chromosome 7, whole genome shotgun sequence
TGGTCTTATGAGGAAGAGGAGTTTGGATTATTTTTTCTTTTCTTTAAAAAGTTGTTTATTTAAGTAGTTT
CTTTAATTCGGGTAACACACACGACAACCCAATAAATTAAACAACGAAAAAATGCAACCTCTATAACCCC
CTAACCCTTTTATTCGGGTAACACACACGACAACCCAATAAATTAAACAACGAAAAATGCAACCTCTATA
ACCCCCGAAAAGGATTGGTGTCGAGTTAGTAAAGGCGAGGGCTAGGACGCTGGTCTTGTGAGGAAGAGGA
GGTTGGATTATTTTTTCTTTTCTTTAATAAGTTGTTTATTTAAGTAGTTTCTTTTATTCGGGTAACACAC
ACGACAACCCAATAAATTAAACAACGAAAAATGCAACCTCTATAACCCCCATCCAAGGTGCCACGGTTGA
GATTGAGCCCCACACACTGTCGGGACAAAGGAGAAACGACCTTCGGGTCAGAGGTTCCAGCGCTCTGGCC
TTCACTGACTACGACCTGAAGGTTTACTCCCTCGGGGACCGAGACGCGAGGAGCACAGCCACCCCCAGCA
CCCCCAACAGCAAACTGGCCGACTTCTGCTTGGACCGGTGCGTGAACTGGCTCGACAAGGTGGGTCAGGT
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[22]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>head seqs/CM000047.1.fa
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&gt;CM000047.1 Cryptococcus neoformans var. neoformans B-3501A chromosome 8, whole genome shotgun sequence
ATGGCCTTCCTGACTACGACCTGAGGTTTACTCCCTCGGCGACCGAGACGCGAGGAGCACAGCCACCCCA
AGCACCCCCAACAGCAAGCTGGCCGAATTCTGCTTGGACCGGTGCGTGAACTGGCTCGACAAGGTGGGTC
AGGTCGTCTCGAAGAACGCTCCGAAAGTCACTGGTGGGGTCTTTAAACCGATCATCCTTTCCACTGGTGG
CCTGATGAGCAGGAGCACAGCAGACGAATGGAAGGAGTGGAGGGAGGCGATGCCGGTGGGGGGGTTCGAG
AAAATGGAGAAACGGATTGGTGTCGAGTTAGTAAAGGCGAGGGCTAGGACGCTGGTCTTGTGAGGAAGAG
GAGGTTGGATTATTTTTTCTTTTCTTTAAAAAGTTGTTTATTTAAGTAGTTTCTTTAATTCGGGCAACCC
ACACGACAACCCAATAAATTAAACAACGAAAAATGCAACCTCTATAACCCTCAATAGCTCTGGCTGGTAC
AGATGGGCAGAAACTCACCTTGCTCCCACGGCATACATCCCTCCCAATAAGTATCTGCAGATGGCGAATA
AACCCCATGCTCTGCGCAAATATATTGAAGGCTGACGACTTTGATGTTATGATCCACCCCATAAACGCTC
</pre></div></div>
</div>
</div>
<div class="section" id="0.3-Compressing-ard-archiving-files">
<h2>0.3 Compressing ard archiving files<a class="headerlink" href="#0.3-Compressing-ard-archiving-files" title="Permalink to this headline">¶</a></h2>
<p>In Bioinformatics the most common formats are <code class="docutils literal notranslate"><span class="pre">gzip</span></code> and <code class="docutils literal notranslate"><span class="pre">bgzip</span></code>.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[23]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">cd</span> ~/seqs
ls
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CM000046.1.fa      <span class="ansi-red-intense-fg ansi-bold">cneo040623.scaffs.fa.gz</span>
CM000047.1.fa      <span class="ansi-red-intense-fg ansi-bold">Cryptococcus_neoformans.ASM9104v1.43.gtf.gz</span>
CM000047.1.fa.bak
</pre></div></div>
</div>
<div class="section" id="Compressing-with-gzip">
<h3>Compressing with <code class="docutils literal notranslate"><span class="pre">gzip</span></code><a class="headerlink" href="#Compressing-with-gzip" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[24]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>gzip *fa
ls
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-intense-fg ansi-bold">CM000046.1.fa.gz</span>   <span class="ansi-red-intense-fg ansi-bold">cneo040623.scaffs.fa.gz</span>
CM000047.1.fa.bak  <span class="ansi-red-intense-fg ansi-bold">Cryptococcus_neoformans.ASM9104v1.43.gtf.gz</span>
<span class="ansi-red-intense-fg ansi-bold">CM000047.1.fa.gz</span>
</pre></div></div>
</div>
</div>
<div class="section" id="Uncompressing-with-gzip">
<h3>Uncompressing with <code class="docutils literal notranslate"><span class="pre">gzip</span></code><a class="headerlink" href="#Uncompressing-with-gzip" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[25]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>gunzip *gz
ls
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CM000046.1.fa  CM000047.1.fa.bak     Cryptococcus_neoformans.ASM9104v1.43.gtf
CM000047.1.fa  cneo040623.scaffs.fa
</pre></div></div>
</div>
</div>
<div class="section" id="Compressing-with-bgzip">
<h3>Compressing with <code class="docutils literal notranslate"><span class="pre">bgzip</span></code><a class="headerlink" href="#Compressing-with-bgzip" title="Permalink to this headline">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">bgzip</span></code> (Blocked GNU Zip Format) is commonly used in Bioinformatics, and allows random access of the content. BAM, BCF and VCF file formats are typically <code class="docutils literal notranslate"><span class="pre">bgzip</span></code> compressed.</p>
<p>Unfortunately <code class="docutils literal notranslate"><span class="pre">bgzip</span></code> does not take multiple files, and we have to use the <code class="docutils literal notranslate"><span class="pre">xargs</span></code> command to pipe in one file at a time.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[26]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>xargs --help <span class="p">|</span> grep <span class="s1">&#39;\-n,&#39;</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
  -n, --max-args=MAX-ARGS      use at most MAX-ARGS arguments per command line
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[27]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls *fa <span class="p">|</span> xargs -n1 bgzip
ls
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-intense-fg ansi-bold">CM000046.1.fa.gz</span>   <span class="ansi-red-intense-fg ansi-bold">cneo040623.scaffs.fa.gz</span>
CM000047.1.fa.bak  Cryptococcus_neoformans.ASM9104v1.43.gtf
<span class="ansi-red-intense-fg ansi-bold">CM000047.1.fa.gz</span>
</pre></div></div>
</div>
</div>
<div class="section" id="Note-that-bgzip-files-can-be-uncompressed-with-gzip">
<h3>Note that <code class="docutils literal notranslate"><span class="pre">bgzip</span></code> files can be uncompressed with <code class="docutils literal notranslate"><span class="pre">gzip</span></code><a class="headerlink" href="#Note-that-bgzip-files-can-be-uncompressed-with-gzip" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[28]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>gunzip *gz
ls
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CM000046.1.fa  CM000047.1.fa.bak     Cryptococcus_neoformans.ASM9104v1.43.gtf
CM000047.1.fa  cneo040623.scaffs.fa
</pre></div></div>
</div>
<div class="section" id="Viewing-compressed-files">
<h4>Viewing compressed files<a class="headerlink" href="#Viewing-compressed-files" title="Permalink to this headline">¶</a></h4>
<p>Some utilities can work directly with gzipped files. One that we will be using extensively is <code class="docutils literal notranslate"><span class="pre">zcat</span></code>.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[29]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls *fa <span class="p">|</span> xargs -n1 bgzip
ls
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-intense-fg ansi-bold">CM000046.1.fa.gz</span>   <span class="ansi-red-intense-fg ansi-bold">cneo040623.scaffs.fa.gz</span>
CM000047.1.fa.bak  Cryptococcus_neoformans.ASM9104v1.43.gtf
<span class="ansi-red-intense-fg ansi-bold">CM000047.1.fa.gz</span>
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[30]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat CM000046.1.fa.gz <span class="p">|</span> head
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&gt;CM000046.1 Cryptococcus neoformans var. neoformans B-3501A chromosome 7, whole genome shotgun sequence
TGGTCTTATGAGGAAGAGGAGTTTGGATTATTTTTTCTTTTCTTTAAAAAGTTGTTTATTTAAGTAGTTT
CTTTAATTCGGGTAACACACACGACAACCCAATAAATTAAACAACGAAAAAATGCAACCTCTATAACCCC
CTAACCCTTTTATTCGGGTAACACACACGACAACCCAATAAATTAAACAACGAAAAATGCAACCTCTATA
ACCCCCGAAAAGGATTGGTGTCGAGTTAGTAAAGGCGAGGGCTAGGACGCTGGTCTTGTGAGGAAGAGGA
GGTTGGATTATTTTTTCTTTTCTTTAATAAGTTGTTTATTTAAGTAGTTTCTTTTATTCGGGTAACACAC
ACGACAACCCAATAAATTAAACAACGAAAAATGCAACCTCTATAACCCCCATCCAAGGTGCCACGGTTGA
GATTGAGCCCCACACACTGTCGGGACAAAGGAGAAACGACCTTCGGGTCAGAGGTTCCAGCGCTCTGGCC
TTCACTGACTACGACCTGAAGGTTTACTCCCTCGGGGACCGAGACGCGAGGAGCACAGCCACCCCCAGCA
CCCCCAACAGCAAACTGGCCGACTTCTGCTTGGACCGGTGCGTGAACTGGCTCGACAAGGTGGGTCAGGT
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[31]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>gunzip *gz
ls
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CM000046.1.fa  CM000047.1.fa.bak     Cryptococcus_neoformans.ASM9104v1.43.gtf
CM000047.1.fa  cneo040623.scaffs.fa
</pre></div></div>
</div>
</div>
<div class="section" id="Archival-with-tar">
<h4>Archival with <code class="docutils literal notranslate"><span class="pre">tar</span></code><a class="headerlink" href="#Archival-with-tar" title="Permalink to this headline">¶</a></h4>
<p>Often we want to compress an entire direcotry (possibly with subdirectories). In Unix, this is most commonly done with <code class="docutils literal notranslate"><span class="pre">tar</span></code>. Flags to note</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">c</span></code> compress</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">x</span></code> extract</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">z</span></code> gzip or gunzip</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">f</span></code> files</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">v</span></code> verbose</p></li>
</ul>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[32]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>tar --help <span class="p">|</span> head -8
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Usage: tar [OPTION...] [FILE]...
GNU &#39;tar&#39; saves many files together into a single tape or disk archive, and can
restore individual files from the archive.

Examples:
  tar -cf archive.tar foo bar  # Create archive.tar from files foo and bar.
  tar -tvf archive.tar         # List all files in archive.tar verbosely.
  tar -xf archive.tar          # Extract all files from archive.tar.
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[33]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">cd</span>
ls seqs
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CM000046.1.fa  CM000047.1.fa.bak     Cryptococcus_neoformans.ASM9104v1.43.gtf
CM000047.1.fa  cneo040623.scaffs.fa
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[34]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>tar czvf seqs.tar.gz seqs
ls seqs*
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
seqs/
seqs/CM000047.1.fa.bak
seqs/CM000046.1.fa
seqs/Cryptococcus_neoformans.ASM9104v1.43.gtf
seqs/CM000047.1.fa
seqs/cneo040623.scaffs.fa
<span class="ansi-red-intense-fg ansi-bold">seqs.tar.gz</span>

seqs:
CM000046.1.fa  CM000047.1.fa.bak     Cryptococcus_neoformans.ASM9104v1.43.gtf
CM000047.1.fa  cneo040623.scaffs.fa
</pre></div></div>
</div>
</div>
</div>
<div class="section" id="View-contents-of-archive">
<h3>View contents of archive<a class="headerlink" href="#View-contents-of-archive" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[35]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>tar tzvf seqs.tar.gz
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
drwxr-xr-x jovyan/users      0 2019-06-26 09:39 seqs/
-rw-r--r-- jovyan/users 1194398 2019-06-26 09:30 seqs/CM000047.1.fa.bak
-rw-r--r-- jovyan/users 1434513 2019-06-26 09:38 seqs/CM000046.1.fa
-rw-r--r-- jovyan/users 27773163 2019-06-26 09:30 seqs/Cryptococcus_neoformans.ASM9104v1.43.gtf
-rw-r--r-- jovyan/users  1194378 2019-06-26 09:38 seqs/CM000047.1.fa
-rw-r--r-- jovyan/users 18907235 2019-06-26 09:39 seqs/cneo040623.scaffs.fa
</pre></div></div>
</div>
</div>
<div class="section" id="Remove-direcotry-and-regenerate-from-archive">
<h3>Remove direcotry and regenerate from archive<a class="headerlink" href="#Remove-direcotry-and-regenerate-from-archive" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[36]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>rm -rf seqs
ls seqs*
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-intense-fg ansi-bold">seqs.tar.gz</span>
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[37]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>tar xzvf seqs.tar.gz
ls seqs*
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
seqs/
seqs/CM000047.1.fa.bak
seqs/CM000046.1.fa
seqs/Cryptococcus_neoformans.ASM9104v1.43.gtf
seqs/CM000047.1.fa
seqs/cneo040623.scaffs.fa
<span class="ansi-red-intense-fg ansi-bold">seqs.tar.gz</span>

seqs:
CM000046.1.fa  CM000047.1.fa.bak     Cryptococcus_neoformans.ASM9104v1.43.gtf
CM000047.1.fa  cneo040623.scaffs.fa
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[38]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>rm seqs.tar.gz
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="0.4-Checksums">
<h2>0.4 Checksums<a class="headerlink" href="#0.4-Checksums" title="Permalink to this headline">¶</a></h2>
<p>When working with genomic data, we deal with very large files. There is a small risk that these files will be corrupted over time or during data transfer. To ensure that files are not changed, we use a “checksum” function. This is a function that generates an long, essentially random number called a checksum that represents the contents of the file. When the file contents change, so will the checksum. In theory, there is a very small probability that two different files generate the same
checksum, but in practice the probability is too small to worry about.</p>
<p>There are several different algorithms for generating the checksums, and at least 3 Unix commands to do so, but they all work very similarly for our purposes.</p>
<p>The strategy is:</p>
<ul class="simple">
<li><p>Generate and store a checksum together with a data file whose integrity you care about</p></li>
<li><p>When you use or download the data, re-generate the checksum (using the same algorithm e.g. MD5) and compare with the checksum</p></li>
</ul>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[39]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls ~/seqs
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CM000046.1.fa  CM000047.1.fa.bak     Cryptococcus_neoformans.ASM9104v1.43.gtf
CM000047.1.fa  cneo040623.scaffs.fa
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[40]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>md5sum ~/seqs/*fa &gt; MD5_CHECKSUM
</pre></div>
</div>
</div>
<div class="section" id="We-check-that-the-files-are-unchanged-wth-the--c-flag">
<h3>We check that the files are unchanged wth the <code class="docutils literal notranslate"><span class="pre">-c</span></code> flag<a class="headerlink" href="#We-check-that-the-files-are-unchanged-wth-the--c-flag" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[41]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>md5sum -c MD5_CHECKSUM
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
/home/jovyan/seqs/CM000046.1.fa: OK
/home/jovyan/seqs/CM000047.1.fa: OK
/home/jovyan/seqs/cneo040623.scaffs.fa: OK
</pre></div></div>
</div>
</div>
<div class="section" id="Suppsoe-one-of-the-files-is-corrupted">
<h3>Suppsoe one of the files is corrupted<a class="headerlink" href="#Suppsoe-one-of-the-files-is-corrupted" title="Permalink to this headline">¶</a></h3>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[42]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">echo</span> <span class="s1">&#39;Oops I did it again&#39;</span> &gt;&gt; seqs/CM000047.1.fa
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[43]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>md5sum -c MD5_CHECKSUM  <span class="se">\</span>
<span class="p">|</span> cat <span class="c1"># this is jusst so we can Run All</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
/home/jovyan/seqs/CM000046.1.fa: OK
/home/jovyan/seqs/CM000047.1.fa: FAILED
/home/jovyan/seqs/cneo040623.scaffs.fa: OK
md5sum: WARNING: 1 computed checksum did NOT match
</pre></div></div>
</div>
</div>
<div class="section" id="Using-diff-to-find-changes">
<h3>Using <code class="docutils literal notranslate"><span class="pre">diff</span></code> to find changes<a class="headerlink" href="#Using-diff-to-find-changes" title="Permalink to this headline">¶</a></h3>
<p>Let’s rename the altered file, re-download the orginal and see what was changed.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[44]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>mv seqs/CM000047.1.fa seqs/CM000047.1.fa.bak
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[45]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>efetch -db<span class="o">=</span>nuccore -format<span class="o">=</span>fasta -id<span class="o">=</span>CM000047.1 &gt; seqs/CM000047.1.fa
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[46]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>diff -u seqs/CM000047.1.fa seqs/CM000047.1.fa.bak <span class="se">\</span>
<span class="p">|</span> cat <span class="c1"># this is jusst so we can Run All</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
--- seqs/CM000047.1.fa  2019-06-26 09:39:10.615210391 +0000
+++ seqs/CM000047.1.fa.bak      2019-06-26 09:39:09.187204139 +0000
@@ -16820,3 +16820,4 @@
 CCGTGTCGGCACGCATCTGATTTTTTTCTTATTATTCATAACAACGGCTACAGATCACGCAAGCGCATCG
 ACTCCGCTGCTGCGAAACACCACGAGTACTGCAACCATTGCTGCGGATGAGCTTAGGGGGTTAGGGGGTT
 AGGGGGTTAGGGGGTTAGGGGGTTAGGGGGTTAGGGGGTTAGGGGGTTAGGGG
+Oops I did it again
</pre></div></div>
</div>
<p>Explanaiton from [official documentation]</p>
<p>Header</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">---</span> <span class="n">seqs</span><span class="o">/</span><span class="n">CM000047</span><span class="o">.</span><span class="mf">1.</span><span class="n">fa</span>  <span class="mi">2018</span><span class="o">-</span><span class="mi">06</span><span class="o">-</span><span class="mi">21</span> <span class="mi">09</span><span class="p">:</span><span class="mi">58</span><span class="p">:</span><span class="mf">18.246185758</span> <span class="o">-</span><span class="mi">0400</span>
<span class="o">+++</span> <span class="n">seqs</span><span class="o">/</span><span class="n">CM000047</span><span class="o">.</span><span class="mf">1.</span><span class="n">fa</span><span class="o">.</span><span class="n">bak</span>  <span class="mi">2018</span><span class="o">-</span><span class="mi">06</span><span class="o">-</span><span class="mi">21</span> <span class="mi">09</span><span class="p">:</span><span class="mi">55</span><span class="p">:</span><span class="mf">55.967707764</span> <span class="o">-</span><span class="mi">0400</span>
</pre></div>
</div>
<p>Explanation</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>The unified output format starts with a two-line header, which looks like this:

--- from-file from-file-modification-time
+++ to-file to-file-modification-time
The timestamp looks like ‘2002-02-21 23:30:39.942229878 -0800’ to indicate the date, time with fractional seconds, and time zone. The fractional seconds are omitted on hosts that do not support fractional timestamps.

You can change the header’s content with the --label=label option. See Alternate Names.
</pre></div>
</div>
<p>Hunk</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">@@</span> <span class="o">-</span><span class="mi">16820</span><span class="p">,</span><span class="mi">3</span> <span class="o">+</span><span class="mi">16820</span><span class="p">,</span><span class="mi">4</span> <span class="o">@@</span>
 <span class="n">CCGTGTCGGCACGCATCTGATTTTTTTCTTATTATTCATAACAACGGCTACAGATCACGCAAGCGCATCG</span>
 <span class="n">ACTCCGCTGCTGCGAAACACCACGAGTACTGCAACCATTGCTGCGGATGAGCTTAGGGGGTTAGGGGGTT</span>
 <span class="n">AGGGGGTTAGGGGGTTAGGGGGTTAGGGGGTTAGGGGGTTAGGGGGTTAGGGG</span>
<span class="o">+</span><span class="n">Oops</span> <span class="n">I</span> <span class="n">did</span> <span class="n">it</span> <span class="n">again</span>
</pre></div>
</div>
<p>Explanation</p>
<p>``` &#64;&#64; from-file-line-numbers to-file-line-numbers &#64;&#64; line-from-either-file line-from-either-file…</p>
<p>If a hunk contains just one line, only its start line number appears. Otherwise its line numbers look like ‘start,count’. An empty hunk is considered to start at the line that follows the hunk.</p>
<p>If a hunk and its context contain two or more lines, its line numbers look like ‘start,count’. Otherwise only its end line number appears. An empty hunk is considered to end at the line that precedes the hunk.</p>
<p>The lines common to both files begin with a space character. The lines that actually differ between the two files have one of the following indicator characters in the left print column:</p>
<p>‘+’ A line was added here to the first file.</p>
<p>‘-’ A line was removed here from the first file. ```</p>
</div>
<div class="section" id="Verify">
<h3>Verify<a class="headerlink" href="#Verify" title="Permalink to this headline">¶</a></h3>
<p>We can confirm that a line was added at the position given by <code class="docutils literal notranslate"><span class="pre">diff</span></code></p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[47]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>cat seqs/CM000047.1.fa <span class="p">|</span> tail +16820 <span class="p">|</span> head -3
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CCGTGTCGGCACGCATCTGATTTTTTTCTTATTATTCATAACAACGGCTACAGATCACGCAAGCGCATCG
ACTCCGCTGCTGCGAAACACCACGAGTACTGCAACCATTGCTGCGGATGAGCTTAGGGGGTTAGGGGGTT
AGGGGGTTAGGGGGTTAGGGGGTTAGGGGGTTAGGGGGTTAGGGGGTTAGGGG
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[48]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>cat seqs/CM000047.1.fa.bak <span class="p">|</span> tail +16820 <span class="p">|</span> head -4
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CCGTGTCGGCACGCATCTGATTTTTTTCTTATTATTCATAACAACGGCTACAGATCACGCAAGCGCATCG
ACTCCGCTGCTGCGAAACACCACGAGTACTGCAACCATTGCTGCGGATGAGCTTAGGGGGTTAGGGGGTT
AGGGGGTTAGGGGGTTAGGGGGTTAGGGGGTTAGGGGGTTAGGGGGTTAGGGG
Oops I did it again
</pre></div></div>
</div>
</div>
</div>
<div class="section" id="1.-Working-with-FASTA-files">
<h2>1. Working with FASTA files<a class="headerlink" href="#1.-Working-with-FASTA-files" title="Permalink to this headline">¶</a></h2>
<p>From <a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=BlastHelp">NCBI FASTA specification</a></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">A</span> <span class="n">sequence</span> <span class="ow">in</span> <span class="n">FASTA</span> <span class="nb">format</span> <span class="n">begins</span> <span class="k">with</span> <span class="n">a</span> <span class="n">single</span><span class="o">-</span><span class="n">line</span> <span class="n">description</span><span class="p">,</span> <span class="n">followed</span> <span class="n">by</span> <span class="n">lines</span> <span class="n">of</span> <span class="n">sequence</span> <span class="n">data</span><span class="o">.</span> <span class="n">The</span> <span class="n">description</span> <span class="n">line</span> <span class="p">(</span><span class="n">defline</span><span class="p">)</span> <span class="ow">is</span> <span class="n">distinguished</span> <span class="kn">from</span> <span class="nn">the</span> <span class="n">sequence</span> <span class="n">data</span> <span class="n">by</span> <span class="n">a</span> <span class="n">greater</span><span class="o">-</span><span class="n">than</span> <span class="p">(</span><span class="s2">&quot;&gt;&quot;</span><span class="p">)</span> <span class="n">symbol</span> <span class="n">at</span> <span class="n">the</span> <span class="n">beginning</span><span class="o">.</span> <span class="n">It</span> <span class="ow">is</span> <span class="n">recommended</span> <span class="n">that</span> <span class="nb">all</span> <span class="n">lines</span> <span class="n">of</span> <span class="n">text</span> <span class="n">be</span> <span class="n">shorter</span> <span class="n">than</span> <span class="mi">80</span> <span class="n">characters</span> <span class="ow">in</span> <span class="n">length</span><span class="o">.</span> <span class="n">An</span> <span class="n">example</span> <span class="n">sequence</span> <span class="ow">in</span> <span class="n">FASTA</span> <span class="nb">format</span> <span class="ow">is</span><span class="p">:</span>

        <span class="o">&gt;</span><span class="n">P01013</span> <span class="n">GENE</span> <span class="n">X</span> <span class="n">PROTEIN</span> <span class="p">(</span><span class="n">OVALBUMIN</span><span class="o">-</span><span class="n">RELATED</span><span class="p">)</span>
        <span class="n">QIKDLLVSSSTDLDTTLVLVNAIYFKGMWKTAFNAEDTREMPFHVTKQESKPVQMMCMNNSFNVATLPAE</span>
        <span class="n">KMKILELPFASGDLSMLVLLPDEVSDLERIEKTINFEKLTEWTNPNTMEKRRVKVYLPQMKIEEKYNLTS</span>
        <span class="n">VLMALGMTDLFIPSANLTGISSAESLKISQAVHGAFMELSEDGIEMAGSTGVIEDIKHSPESEQFRADHP</span>
        <span class="n">FLFLIKHNPTNTIVYFGRYWSP</span>
</pre></div>
</div>
<p>Note that a FASTA file may contain mulitple sequences, each with its own description line.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[49]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">cd</span> seqs
wget ftp://ftp.ensemblgenomes.org/pub/release-43/fungi/gtf/cryptococcus_neoformans/Cryptococcus_neoformans.ASM9104v1.43.gtf.gz
wget http://www-sequence.stanford.edu/group/C.neoformans/files/cneo040623.scaffs.fa.gz
<span class="nb">cd</span> -
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
--2019-06-26 09:39:11--  ftp://ftp.ensemblgenomes.org/pub/release-43/fungi/gtf/cryptococcus_neoformans/Cryptococcus_neoformans.ASM9104v1.43.gtf.gz
           =&gt; ‘Cryptococcus_neoformans.ASM9104v1.43.gtf.gz’
Resolving ftp.ensemblgenomes.org (ftp.ensemblgenomes.org)... 193.62.197.94
Connecting to ftp.ensemblgenomes.org (ftp.ensemblgenomes.org)|193.62.197.94|:21... connected.
Logging in as anonymous ... Logged in!
==&gt; SYST ... done.    ==&gt; PWD ... done.
==&gt; TYPE I ... done.  ==&gt; CWD (1) /pub/release-43/fungi/gtf/cryptococcus_neoformans ... done.
==&gt; SIZE Cryptococcus_neoformans.ASM9104v1.43.gtf.gz ... 1546239
==&gt; PASV ... done.    ==&gt; RETR Cryptococcus_neoformans.ASM9104v1.43.gtf.gz ... done.
Length: 1546239 (1.5M) (unauthoritative)

Cryptococcus_neofor 100%[===================&gt;]   1.47M  2.04MB/s    in 0.7s

2019-06-26 09:39:13 (2.04 MB/s) - ‘Cryptococcus_neoformans.ASM9104v1.43.gtf.gz’ saved [1546239]

--2019-06-26 09:39:13--  http://www-sequence.stanford.edu/group/C.neoformans/files/cneo040623.scaffs.fa.gz
Resolving www-sequence.stanford.edu (www-sequence.stanford.edu)... 171.66.102.5
Connecting to www-sequence.stanford.edu (www-sequence.stanford.edu)|171.66.102.5|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 5856251 (5.6M) [text/plain]
Saving to: ‘cneo040623.scaffs.fa.gz’

cneo040623.scaffs.f 100%[===================&gt;]   5.58M  1.65MB/s    in 3.4s

2019-06-26 09:39:16 (1.65 MB/s) - ‘cneo040623.scaffs.fa.gz’ saved [5856251/5856251]

/home/jovyan
</pre></div></div>
</div>
<div class="section" id="Set-up-variables-for-files-we-will-be-using">
<h3>Set up variables for files we will be using<a class="headerlink" href="#Set-up-variables-for-files-we-will-be-using" title="Permalink to this headline">¶</a></h3>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[50]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="c1">#DATA_DIR=&#39;/data/hts/2018/Granek_4837_180427A5&#39;</span>
<span class="nv">DATA_DIR</span><span class="o">=</span><span class="s1">&#39;./seqs&#39;</span>
<span class="nv">FASTA</span><span class="o">=</span><span class="s1">&#39;cneo040623.scaffs.fa.gz&#39;</span>
<span class="nv">FASTQ</span><span class="o">=</span><span class="s1">&#39;10_MA_C_S3_L003_R1_001.fastq.gz&#39;</span>
<span class="nv">GTF</span><span class="o">=</span><span class="s1">&#39;Cryptococcus_neoformans.ASM9104v1.43.gtf.gz&#39;</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="This-is-how-you-get-the-value-of-a-shell-variable">
<h3>This is how you get the value of a shell variable<a class="headerlink" href="#This-is-how-you-get-the-value-of-a-shell-variable" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[51]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">echo</span> <span class="nv">$DATA_DIR</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
./seqs
</pre></div></div>
</div>
</div>
<div class="section" id="List-files-in-data-direcotry">
<h3>List files in data direcotry<a class="headerlink" href="#List-files-in-data-direcotry" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[52]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls <span class="nv">$DATA_DIR</span> <span class="p">|</span> head -3
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CM000046.1.fa
CM000047.1.fa
CM000047.1.fa.bak
</pre></div></div>
</div>
</div>
<div class="section" id="Inspect-the-contents-of-a-zipped-file.">
<h3>Inspect the contents of a zipped file.<a class="headerlink" href="#Inspect-the-contents-of-a-zipped-file." title="Permalink to this headline">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">zcat</span></code> is like <code class="docutils literal notranslate"><span class="pre">cat</span></code> but works on gzipped files.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[53]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$FASTA</span> <span class="p">|</span> head
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&gt;chr01.b3501.040506 5 contigs (2263073 bp)
CCCTAACCCCCTAACCCCCTAACCCTCAGTCGGGGAGGCGACAATAGCTG
GTCATATGTTCTGTACGTTATTGGCCAACTGATCATACCTGAATCGAGCC
GAACCGGGATTATCAAAGACGAACATGGTCGGGTCCTTGAACCAAACGAA
TCTCCGTCCGCTGGCGTGTTTTTCTTTTCTCAAGTGGGCAAGTTACCCGG
GGATTTATACAACGATAAGAGGCTATCCGCTGACTATAAATTGTGTTAGC
TGATCCAGATCCGCAGTTGGCACAAACCGATGTTTCCTTTTCCTCAGGTC
CTGAATATTTCTCTTCCATCAAGGCTCCCAACCCCGAGGGGAGTATTTCC
ACCCGATCTGATTCCAAGAGGTCCAGCGTCAACCAGGTACGTGAATGTTT
TCTTCTTTAGTGGTGGGGACCCTATGCTTGTCCCACAGAACTGCACTGAA
</pre></div></div>
</div>
</div>
<div class="section" id="Show-headers-only">
<h3>Show headers only<a class="headerlink" href="#Show-headers-only" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[54]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$FASTA</span>   <span class="p">|</span> grep <span class="s1">&#39;^&gt;&#39;</span> <span class="p">|</span> head
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&gt;chr01.b3501.040506 5 contigs (2263073 bp)
&gt;chr02.b3501.040506 6 contigs (1607892 bp)
&gt;chr03.b3501.040506 6 contigs (2025976 bp)
&gt;chr04.b3501.040623 5 contigs (1762351 bp)
&gt;chr05.b3501.040506 7 contigs (1432964 bp)
&gt;chr06.b3501.040616 4 contigs (1411173 bp)
&gt;chr07.b3501.040506 6 contigs (1356306 bp)
&gt;chr08.b3501.040506 3 contigs (1153453 bp)
&gt;chr09.b3501.040506 5 contigs (1072068 bp)
&gt;chr10.b3501.040506 3 contigs (1017008 bp)
</pre></div></div>
</div>
</div>
<div class="section" id="Show-headers-with-line-numbers">
<h3>Show headers with line numbers<a class="headerlink" href="#Show-headers-with-line-numbers" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[55]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$FASTA</span><span class="p">|</span> grep -n <span class="s1">&#39;^&gt;&#39;</span> <span class="p">|</span> head
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1:&gt;chr01.b3501.040506 5 contigs (2263073 bp)
45264:&gt;chr02.b3501.040506 6 contigs (1607892 bp)
77423:&gt;chr03.b3501.040506 6 contigs (2025976 bp)
117944:&gt;chr04.b3501.040623 5 contigs (1762351 bp)
153193:&gt;chr05.b3501.040506 7 contigs (1432964 bp)
181854:&gt;chr06.b3501.040616 4 contigs (1411173 bp)
210079:&gt;chr07.b3501.040506 6 contigs (1356306 bp)
237207:&gt;chr08.b3501.040506 3 contigs (1153453 bp)
260278:&gt;chr09.b3501.040506 5 contigs (1072068 bp)
281721:&gt;chr10.b3501.040506 3 contigs (1017008 bp)
</pre></div></div>
</div>
</div>
<div class="section" id="Count-number-of-headers-(one-for-each-crhomosome)">
<h3>Count number of headers (one for each crhomosome)<a class="headerlink" href="#Count-number-of-headers-(one-for-each-crhomosome)" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[56]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$FASTA</span>  <span class="p">|</span> grep -c <span class="s1">&#39;^&gt;&#39;</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
14
</pre></div></div>
</div>
</div>
<div class="section" id="Extract-first-part-of-the-sequence-from-chromosome-7">
<h3>Extract first part of the sequence from chromosome 7<a class="headerlink" href="#Extract-first-part-of-the-sequence-from-chromosome-7" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[57]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$FASTA</span> <span class="p">|</span> sed -n <span class="s1">&#39;210079,210082p&#39;</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&gt;chr07.b3501.040506 6 contigs (1356306 bp)
TGGTCTTATGAGGAAGAGGAGTTTGGATTATTTTTTCTTTTCTTTAAAAA
GTTGTTTATTTAAGTAGTTTCTTTAATTCGGGTAACACACACGACAACCC
AATAAATTAAACAACGAAAAAATGCAACCTCTATAACCCCCTAACCCTTT
</pre></div></div>
</div>
</div>
<div class="section" id="Extract-first-line-of-chromosome-7-and-save-to-file">
<h3>Extract first line of chromosome 7 and save to file<a class="headerlink" href="#Extract-first-line-of-chromosome-7-and-save-to-file" title="Permalink to this headline">¶</a></h3>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[58]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$FASTA</span> <span class="p">|</span> sed -n <span class="s1">&#39;210080p&#39;</span> &gt; dna.txt
</pre></div>
</div>
</div>
</div>
<div class="section" id="Inspect-the-file-we-just-saved">
<h3>Inspect the file we just saved<a class="headerlink" href="#Inspect-the-file-we-just-saved" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[59]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>cat dna.txt
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
TGGTCTTATGAGGAAGAGGAGTTTGGATTATTTTTTCTTTTCTTTAAAAA
</pre></div></div>
</div>
</div>
<div class="section" id="Find-complement">
<h3>Find complement<a class="headerlink" href="#Find-complement" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[60]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>cat dna.txt <span class="p">|</span> tr ACGT TGCA
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
ACCAGAATACTCCTTCTCCTCAAACCTAATAAAAAAGAAAAGAAATTTTT
</pre></div></div>
</div>
</div>
<div class="section" id="Find-reverse-complement">
<h3>Find reverse complement<a class="headerlink" href="#Find-reverse-complement" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[61]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>cat dna.txt <span class="p">|</span> tr ACGT TGCA <span class="p">|</span> rev
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
TTTTTAAAGAAAAGAAAAAATAATCCAAACTCCTCTTCCTCATAAGACCA
</pre></div></div>
</div>
</div>
<div class="section" id="Delete-A-and-T">
<h3>Delete A and T<a class="headerlink" href="#Delete-A-and-T" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[62]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>cat dna.txt <span class="p">|</span> tr -d <span class="s1">&#39;AT&#39;</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
GGCGGGGGGGGGCC
</pre></div></div>
</div>
<div class="section" id="Only-count-non-A-and-T">
<h4>Only count non A and T<a class="headerlink" href="#Only-count-non-A-and-T" title="Permalink to this headline">¶</a></h4>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[63]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>cat dna.txt <span class="p">|</span> tr -d <span class="s1">&#39;AT&#39;</span> <span class="p">|</span> tr -d <span class="s1">&#39;\n&#39;</span> <span class="p">|</span> wc -c
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
14
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[64]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>man tr <span class="p">|</span> grep <span class="s1">&#39;\-d,&#39;</span> -A2
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
bash: man: command not found
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>

</pre></div></div>
</div>
</div>
</div>
<div class="section" id="Count-C-and-G">
<h3>Count C and G<a class="headerlink" href="#Count-C-and-G" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[65]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>grep <span class="s1">&#39;[CG]&#39;</span> -o dna.txt <span class="p">|</span> wc -l
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
14
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[66]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>man grep <span class="p">|</span> grep <span class="s1">&#39;\-o,&#39;</span> -A2
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
bash: man: command not found
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>

</pre></div></div>
</div>
</div>
<div class="section" id="Specify-data-directory-as-a-shell-variable">
<h3>Specify data directory as a shell variable<a class="headerlink" href="#Specify-data-directory-as-a-shell-variable" title="Permalink to this headline">¶</a></h3>
</div>
</div>
<div class="section" id="2.-Working-with-FASTQ-files">
<h2>2. Working with FASTQ files<a class="headerlink" href="#2.-Working-with-FASTQ-files" title="Permalink to this headline">¶</a></h2>
<p>From <a class="reference external" href="https://en.wikipedia.org/wiki/FASTQ_format">Wikipedia</a></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">A</span> <span class="n">FASTQ</span> <span class="n">file</span> <span class="n">normally</span> <span class="n">uses</span> <span class="n">four</span> <span class="n">lines</span> <span class="n">per</span> <span class="n">sequence</span><span class="o">.</span>

<span class="n">Line</span> <span class="mi">1</span> <span class="n">begins</span> <span class="k">with</span> <span class="n">a</span> <span class="s1">&#39;@&#39;</span> <span class="n">character</span> <span class="ow">and</span> <span class="ow">is</span> <span class="n">followed</span> <span class="n">by</span> <span class="n">a</span> <span class="n">sequence</span> <span class="n">identifier</span> <span class="ow">and</span> <span class="n">an</span> <span class="n">optional</span> <span class="n">description</span> <span class="p">(</span><span class="n">like</span> <span class="n">a</span> <span class="n">FASTA</span> <span class="n">title</span> <span class="n">line</span><span class="p">)</span><span class="o">.</span>
<span class="n">Line</span> <span class="mi">2</span> <span class="ow">is</span> <span class="n">the</span> <span class="n">raw</span> <span class="n">sequence</span> <span class="n">letters</span><span class="o">.</span>
<span class="n">Line</span> <span class="mi">3</span> <span class="n">begins</span> <span class="k">with</span> <span class="n">a</span> <span class="s1">&#39;+&#39;</span> <span class="n">character</span> <span class="ow">and</span> <span class="ow">is</span> <span class="n">optionally</span> <span class="n">followed</span> <span class="n">by</span> <span class="n">the</span> <span class="n">same</span> <span class="n">sequence</span> <span class="n">identifier</span> <span class="p">(</span><span class="ow">and</span> <span class="nb">any</span> <span class="n">description</span><span class="p">)</span> <span class="n">again</span><span class="o">.</span>
<span class="n">Line</span> <span class="mi">4</span> <span class="n">encodes</span> <span class="n">the</span> <span class="n">quality</span> <span class="n">values</span> <span class="k">for</span> <span class="n">the</span> <span class="n">sequence</span> <span class="ow">in</span> <span class="n">Line</span> <span class="mi">2</span><span class="p">,</span> <span class="ow">and</span> <span class="n">must</span> <span class="n">contain</span> <span class="n">the</span> <span class="n">same</span> <span class="n">number</span> <span class="n">of</span> <span class="n">symbols</span> <span class="k">as</span> <span class="n">letters</span> <span class="ow">in</span> <span class="n">the</span> <span class="n">sequence</span><span class="o">.</span>
</pre></div>
</div>
<div class="section" id="Specify-target-file-as-a-shell-variable">
<h3>Specify target file as a shell variable<a class="headerlink" href="#Specify-target-file-as-a-shell-variable" title="Permalink to this headline">¶</a></h3>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[67]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nv">DATA_DIR</span><span class="o">=</span><span class="s1">&#39;/data/hts2018_pilot/Granek_4837_180427A5&#39;</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[68]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls <span class="nv">$DATA_DIR</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-intense-fg ansi-bold">10_MA_C_S3_L001_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">26_RZ_C_S16_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">10_MA_C_S3_L002_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">27_MA_P_S38_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">10_MA_C_S3_L003_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">27_MA_P_S38_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">10_MA_C_S3_L004_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">27_MA_P_S38_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">10_RZ_C_S11_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">27_MA_P_S38_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">10_RZ_C_S11_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">27_RZ_P_S46_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">10_RZ_C_S11_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">27_RZ_P_S46_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">10_RZ_C_S11_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">27_RZ_P_S46_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">11_MA_J_S20_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">27_RZ_P_S46_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">11_MA_J_S20_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">2_MA_C_S1_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">11_MA_J_S20_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">2_MA_C_S1_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">11_MA_J_S20_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">2_MA_C_S1_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">11_RZ_J_S28_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">2_MA_C_S1_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">11_RZ_J_S28_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">2_RZ_C_S9_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">11_RZ_J_S28_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">2_RZ_C_S9_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">11_RZ_J_S28_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">2_RZ_C_S9_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">12_MA_P_S36_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">2_RZ_C_S9_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">12_MA_P_S36_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">2_TOT_C_S17_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">12_MA_P_S36_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">2_TOT_C_S17_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">12_MA_P_S36_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">2_TOT_C_S17_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">12_RZ_P_S44_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">2_TOT_C_S17_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">12_RZ_P_S44_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">35_MA_P_S39_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">12_RZ_P_S44_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">35_MA_P_S39_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">12_RZ_P_S44_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">35_MA_P_S39_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">13_MA_J_S21_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">35_MA_P_S39_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">13_MA_J_S21_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">35_RZ_P_S47_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">13_MA_J_S21_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">35_RZ_P_S47_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">13_MA_J_S21_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">35_RZ_P_S47_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">13_RZ_J_S29_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">35_RZ_P_S47_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">13_RZ_J_S29_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">36_MA_J_S24_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">13_RZ_J_S29_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">36_MA_J_S24_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">13_RZ_J_S29_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">36_MA_J_S24_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">14_MA_C_S4_L001_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">36_MA_J_S24_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">14_MA_C_S4_L002_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">36_RZ_J_S32_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">14_MA_C_S4_L003_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">36_RZ_J_S32_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">14_MA_C_S4_L004_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">36_RZ_J_S32_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">14_RZ_C_S12_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">36_RZ_J_S32_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">14_RZ_C_S12_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">38_MA_P_S40_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">14_RZ_C_S12_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">38_MA_P_S40_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">14_RZ_C_S12_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">38_MA_P_S40_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">15_MA_C_S5_L001_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">38_MA_P_S40_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">15_MA_C_S5_L002_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">38_RZ_P_S48_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">15_MA_C_S5_L003_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">38_RZ_P_S48_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">15_MA_C_S5_L004_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">38_RZ_P_S48_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">15_RZ_C_S13_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">38_RZ_P_S48_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">15_RZ_C_S13_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">3_MA_J_S19_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">15_RZ_C_S13_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">3_MA_J_S19_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">15_RZ_C_S13_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">3_MA_J_S19_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">16_MA_P_S37_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">3_MA_J_S19_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">16_MA_P_S37_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">3_RZ_J_S27_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">16_MA_P_S37_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">3_RZ_J_S27_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">16_MA_P_S37_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">3_RZ_J_S27_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">16_RZ_P_S45_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">3_RZ_J_S27_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">16_RZ_P_S45_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">3_TOT_J_S34_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">16_RZ_P_S45_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">3_TOT_J_S34_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">16_RZ_P_S45_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">3_TOT_J_S34_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">1_MA_J_S18_L001_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">3_TOT_J_S34_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">1_MA_J_S18_L002_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">40_MA_J_S25_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">1_MA_J_S18_L003_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">40_MA_J_S25_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">1_MA_J_S18_L004_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">40_MA_J_S25_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">1_RZ_J_S26_L001_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">40_MA_J_S25_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">1_RZ_J_S26_L002_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">40_RZ_J_S33_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">1_RZ_J_S26_L003_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">40_RZ_J_S33_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">1_RZ_J_S26_L004_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">40_RZ_J_S33_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">21_MA_C_S6_L001_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">40_RZ_J_S33_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">21_MA_C_S6_L002_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">45_MA_P_S41_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">21_MA_C_S6_L003_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">45_MA_P_S41_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">21_MA_C_S6_L004_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">45_MA_P_S41_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">21_RZ_C_S14_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">45_MA_P_S41_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">21_RZ_C_S14_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">45_RZ_P_S49_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">21_RZ_C_S14_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">45_RZ_P_S49_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">21_RZ_C_S14_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">45_RZ_P_S49_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">22_MA_C_S7_L001_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">45_RZ_P_S49_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">22_MA_C_S7_L002_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">47_MA_P_S42_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">22_MA_C_S7_L003_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">47_MA_P_S42_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">22_MA_C_S7_L004_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">47_MA_P_S42_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">22_RZ_C_S15_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">47_MA_P_S42_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">22_RZ_C_S15_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">47_RZ_P_S50_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">22_RZ_C_S15_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">47_RZ_P_S50_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">22_RZ_C_S15_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">47_RZ_P_S50_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">23_MA_J_S22_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">47_RZ_P_S50_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">23_MA_J_S22_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">4_MA_P_S35_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">23_MA_J_S22_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">4_MA_P_S35_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">23_MA_J_S22_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">4_MA_P_S35_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">23_RZ_J_S30_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">4_MA_P_S35_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">23_RZ_J_S30_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">4_RZ_P_S43_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">23_RZ_J_S30_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">4_RZ_P_S43_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">23_RZ_J_S30_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">4_RZ_P_S43_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">24_MA_J_S23_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">4_RZ_P_S43_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">24_MA_J_S23_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">4_TOT_P_S51_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">24_MA_J_S23_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">4_TOT_P_S51_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">24_MA_J_S23_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">4_TOT_P_S51_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">24_RZ_J_S31_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">4_TOT_P_S51_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">24_RZ_J_S31_L002_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">9_MA_C_S2_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">24_RZ_J_S31_L003_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">9_MA_C_S2_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">24_RZ_J_S31_L004_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">9_MA_C_S2_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">26_MA_C_S8_L001_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">9_MA_C_S2_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">26_MA_C_S8_L002_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">9_RZ_C_S10_L001_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">26_MA_C_S8_L003_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">9_RZ_C_S10_L002_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">26_MA_C_S8_L004_R1_001.fastq.gz</span>   <span class="ansi-red-intense-fg ansi-bold">9_RZ_C_S10_L003_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">26_RZ_C_S16_L001_R1_001.fastq.gz</span>  <span class="ansi-red-intense-fg ansi-bold">9_RZ_C_S10_L004_R1_001.fastq.gz</span>
<span class="ansi-red-intense-fg ansi-bold">26_RZ_C_S16_L002_R1_001.fastq.gz</span>  Granek_4837_180427A5.checksum
<span class="ansi-red-intense-fg ansi-bold">26_RZ_C_S16_L003_R1_001.fastq.gz</span>  README.rtf
</pre></div></div>
</div>
</div>
<div class="section" id="Show-the-contents-of-a-zipped-file">
<h3>Show the contents of a zipped file<a class="headerlink" href="#Show-the-contents-of-a-zipped-file" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[69]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$FASTQ</span> <span class="p">|</span> head -n <span class="m">10</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
@NB501800:128:HTYKFBGX5:3:11401:19195:1019 1:N:0:ATTACTCG+AGGATAGG
NGTAAGGGTCTGGGGGACGTTGTTGTGGATGGAAACAACCCTGGCCTCTCGGGGCTGAGCAGGGAGTCGCTTGAGC
+
#AAAAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEEEEEEEEEEEEEEEEEEAEEEEEEEEEEEEE/&lt;&lt;
@NB501800:128:HTYKFBGX5:3:11401:24339:1020 1:N:0:ATTACTCG+AGGATAGG
NCTCGCCTGGGGTGCTTGCGTGTCCCTGTCCTTGGCCCTGGTCACCGGGGAGATGCACATCATCTTCTTCTGGCTC
+
#AAAAEAAEEEEEEEEEEEEEEEE/EEE/EEEEEEEEEEEEEEEE&lt;A/EEEE6EEE//EEEAEEEEEEE/EAEEEA
@NB501800:128:HTYKFBGX5:3:11401:24140:1020 1:N:0:ATTACTCG+AGGATAGG
NTGATCACGGGAGACAAGCTAGAGATGCAAAATGCCATAGTACGATCATCCCATACAAGATGGGTGAGAGAAAAAG
</pre></div></div>
</div>
</div>
<div class="section" id="Extracting-just-the-sequence">
<h3>Extracting just the sequence<a class="headerlink" href="#Extracting-just-the-sequence" title="Permalink to this headline">¶</a></h3>
<p>We need to extract every 4th line starting from the 2nd. We cna use a tool called <code class="docutils literal notranslate"><span class="pre">sed</span></code> (stream editor) to do so.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[70]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$FASTQ</span> <span class="p">|</span> sed -n <span class="s1">&#39;2~4p;&#39;</span> <span class="p">|</span> head
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
NGTAAGGGTCTGGGGGACGTTGTTGTGGATGGAAACAACCCTGGCCTCTCGGGGCTGAGCAGGGAGTCGCTTGAGC
NCTCGCCTGGGGTGCTTGCGTGTCCCTGTCCTTGGCCCTGGTCACCGGGGAGATGCACATCATCTTCTTCTGGCTC
NTGATCACGGGAGACAAGCTAGAGATGCAAAATGCCATAGTACGATCATCCCATACAAGATGGGTGAGAGAAAAAG
NGGGCCTTGGTCCTGCCCTCAGCCTTGATAGGACCCTTGCCGTCGATGGGGTTACCCAAAGCGTCAACAACTCGGC
NACCTTTCGGAGACCTCGGTGGGTCTTCCTGGGGAGACGGGTAGTACCCCATCGGGCGGTAACACCCTCGTAACCG
NCCGAACTGTTGAAAAGAGCTTCCACGCTCCAAAGGCAGGGACAAGAGTAAAGAGCCACCACAGCCAATCTCCAAA
NGCAAACTGAACGTCCTCAGTCTCCATTCGCTGGAAGTCCGATTCAGGACGGGCAGCGTCATCGATAGAGAATGGA
NGCATTACCAGATGCACTGATAGCAATGATCGTCTCATCTTCCAAGTACTTGCTCACAGCGATACTATCACTCAAA
GGCGATCTTCCTCTTGGGGAAGTGAACGTTCTTCTTGTGCTGCTTCTCGGTGATAGCAGAAGCCTCCTTGGTAGTC
CGGCGATGATGAGAGTGAAGGGATCTTGCCACATGAGAGGGGATGATCGACGGGGACGAGGACGGATGGGGATGAT
</pre></div></div>
</div>
<p><strong>Explanation</strong></p>
<p><code class="docutils literal notranslate"><span class="pre">sed</span> <span class="pre">-n</span> <span class="pre">'2~4p'</span></code></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-n</span></code> means to suppress all output</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">2~4p</span></code> means print (p) every 4th line starting at line 2</p></li>
</ul>
</div>
<div class="section" id="Joining-lines">
<h3>Joining lines<a class="headerlink" href="#Joining-lines" title="Permalink to this headline">¶</a></h3>
<p>Getting sequence as a single string. We just need to delete newlline (<span class="math">\n</span>) characters.</p>
<p>Note: We can break long commands uisng the line continuation character</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[71]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$FASTQ</span> <span class="p">|</span> <span class="se">\</span>
sed -n <span class="s1">&#39;2~4p;&#39;</span> <span class="p">|</span> <span class="se">\</span>
head <span class="p">|</span> <span class="se">\</span>
tr -d <span class="s1">&#39;\n&#39;</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
NGTAAGGGTCTGGGGGACGTTGTTGTGGATGGAAACAACCCTGGCCTCTCGGGGCTGAGCAGGGAGTCGCTTGAGCNCTCGCCTGGGGTGCTTGCGTGTCCCTGTCCTTGGCCCTGGTCACCGGGGAGATGCACATCATCTTCTTCTGGCTCNTGATCACGGGAGACAAGCTAGAGATGCAAAATGCCATAGTACGATCATCCCATACAAGATGGGTGAGAGAAAAAGNGGGCCTTGGTCCTGCCCTCAGCCTTGATAGGACCCTTGCCGTCGATGGGGTTACCCAAAGCGTCAACAACTCGGCNACCTTTCGGAGACCTCGGTGGGTCTTCCTGGGGAGACGGGTAGTACCCCATCGGGCGGTAACACCCTCGTAACCGNCCGAACTGTTGAAAAGAGCTTCCACGCTCCAAAGGCAGGGACAAGAGTAAAGAGCCACCACAGCCAATCTCCAAANGCAAACTGAACGTCCTCAGTCTCCATTCGCTGGAAGTCCGATTCAGGACGGGCAGCGTCATCGATAGAGAATGGANGCATTACCAGATGCACTGATAGCAATGATCGTCTCATCTTCCAAGTACTTGCTCACAGCGATACTATCACTCAAAGGCGATCTTCCTCTTGGGGAAGTGAACGTTCTTCTTGTGCTGCTTCTCGGTGATAGCAGAAGCCTCCTTGGTAGTCCGGCGATGATGAGAGTGAAGGGATCTTGCCACATGAGAGGGGATGATCGACGGGGACGAGGACGGATGGGGATGAT
</pre></div></div>
</div>
</div>
</div>
<div class="section" id="3.Working-wtih-GTF-files">
<h2>3.Working wtih GTF files<a class="headerlink" href="#3.Working-wtih-GTF-files" title="Permalink to this headline">¶</a></h2>
<p>From <a class="reference external" href="http://genome.ucsc.edu/FAQ/FAQformat.html#format3">UCSD Genome Browser</a></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">The</span> <span class="n">GTF</span> <span class="p">(</span><span class="n">General</span> <span class="n">Transfer</span> <span class="n">Format</span><span class="p">)</span> <span class="ow">is</span> <span class="n">an</span> <span class="n">extension</span> <span class="n">of</span> <span class="n">GFF</span> <span class="n">version</span> <span class="mi">2</span>
<span class="ow">and</span> <span class="n">used</span> <span class="n">to</span> <span class="n">represent</span> <span class="n">transcription</span> <span class="n">models</span><span class="o">.</span> <span class="n">GFF</span> <span class="p">(</span><span class="n">General</span> <span class="n">Feature</span> <span class="n">Format</span><span class="p">)</span>
<span class="n">consists</span> <span class="n">of</span> <span class="n">one</span> <span class="n">line</span> <span class="n">per</span> <span class="n">feature</span><span class="p">,</span> <span class="n">each</span> <span class="n">containing</span> <span class="mi">9</span> <span class="n">columns</span> <span class="n">of</span> <span class="n">data</span><span class="o">.</span>

<span class="n">Fields</span>

<span class="n">Fields</span> <span class="n">are</span> <span class="n">tab</span><span class="o">-</span><span class="n">separated</span><span class="o">.</span> <span class="n">Also</span><span class="p">,</span> <span class="nb">all</span> <span class="n">but</span> <span class="n">the</span> <span class="n">final</span> <span class="n">field</span> <span class="ow">in</span> <span class="n">each</span>
<span class="n">feature</span> <span class="n">line</span> <span class="n">must</span> <span class="n">contain</span> <span class="n">a</span> <span class="n">value</span><span class="p">;</span> <span class="s2">&quot;empty&quot;</span> <span class="n">columns</span> <span class="n">are</span> <span class="n">denoted</span>
<span class="k">with</span> <span class="n">a</span> <span class="s1">&#39;.&#39;</span>

    <span class="n">seqname</span>   <span class="o">-</span> <span class="n">name</span> <span class="n">of</span> <span class="n">the</span> <span class="n">chromosome</span> <span class="ow">or</span> <span class="n">scaffold</span><span class="p">;</span> <span class="n">chromosome</span> <span class="n">names</span>
                <span class="n">without</span> <span class="n">a</span> <span class="s1">&#39;chr&#39;</span>
    <span class="n">source</span>    <span class="o">-</span> <span class="n">name</span> <span class="n">of</span> <span class="n">the</span> <span class="n">program</span> <span class="n">that</span> <span class="n">generated</span> <span class="n">this</span> <span class="n">feature</span><span class="p">,</span> <span class="ow">or</span>
                <span class="n">the</span> <span class="n">data</span> <span class="n">source</span> <span class="p">(</span><span class="n">database</span> <span class="ow">or</span> <span class="n">project</span> <span class="n">name</span><span class="p">)</span>
    <span class="n">feature</span>   <span class="o">-</span> <span class="n">feature</span> <span class="nb">type</span> <span class="n">name</span><span class="o">.</span> <span class="n">Current</span> <span class="n">allowed</span> <span class="n">features</span> <span class="n">are</span>
                <span class="p">{</span><span class="n">gene</span><span class="p">,</span> <span class="n">transcript</span><span class="p">,</span> <span class="n">exon</span><span class="p">,</span> <span class="n">CDS</span><span class="p">,</span> <span class="n">Selenocysteine</span><span class="p">,</span> <span class="n">start_codon</span><span class="p">,</span>
                <span class="n">stop_codon</span> <span class="ow">and</span> <span class="n">UTR</span><span class="p">}</span>
    <span class="n">start</span>     <span class="o">-</span> <span class="n">start</span> <span class="n">position</span> <span class="n">of</span> <span class="n">the</span> <span class="n">feature</span><span class="p">,</span> <span class="k">with</span> <span class="n">sequence</span> <span class="n">numbering</span>
                <span class="n">starting</span> <span class="n">at</span> <span class="mf">1.</span>
    <span class="n">end</span>       <span class="o">-</span> <span class="n">end</span> <span class="n">position</span> <span class="n">of</span> <span class="n">the</span> <span class="n">feature</span><span class="p">,</span> <span class="k">with</span> <span class="n">sequence</span> <span class="n">numbering</span>
                <span class="n">starting</span> <span class="n">at</span> <span class="mf">1.</span>
    <span class="n">score</span>     <span class="o">-</span> <span class="n">a</span> <span class="n">floating</span> <span class="n">point</span> <span class="n">value</span> <span class="n">indiciating</span> <span class="n">the</span> <span class="n">score</span> <span class="n">of</span> <span class="n">a</span> <span class="n">feature</span>
    <span class="n">strand</span>    <span class="o">-</span> <span class="n">defined</span> <span class="k">as</span> <span class="o">+</span> <span class="p">(</span><span class="n">forward</span><span class="p">)</span> <span class="ow">or</span> <span class="o">-</span> <span class="p">(</span><span class="n">reverse</span><span class="p">)</span><span class="o">.</span>
    <span class="n">frame</span>     <span class="o">-</span> <span class="n">one</span> <span class="n">of</span> <span class="s1">&#39;0&#39;</span><span class="p">,</span> <span class="s1">&#39;1&#39;</span> <span class="ow">or</span> <span class="s1">&#39;2&#39;</span><span class="o">.</span> <span class="n">Frame</span> <span class="n">indicates</span> <span class="n">the</span> <span class="n">number</span> <span class="n">of</span> <span class="n">base</span> <span class="n">pairs</span>
                <span class="n">before</span> <span class="n">you</span> <span class="n">encounter</span> <span class="n">a</span> <span class="n">full</span> <span class="n">codon</span><span class="o">.</span> <span class="s1">&#39;0&#39;</span> <span class="n">indicates</span> <span class="n">the</span> <span class="n">feature</span>
                <span class="n">begins</span> <span class="k">with</span> <span class="n">a</span> <span class="n">whole</span> <span class="n">codon</span><span class="o">.</span> <span class="s1">&#39;1&#39;</span> <span class="n">indicates</span> <span class="n">there</span> <span class="ow">is</span> <span class="n">an</span> <span class="n">extra</span>
                <span class="n">base</span> <span class="p">(</span><span class="n">the</span> <span class="mi">3</span><span class="n">rd</span> <span class="n">base</span> <span class="n">of</span> <span class="n">the</span> <span class="n">prior</span> <span class="n">codon</span><span class="p">)</span> <span class="n">at</span> <span class="n">the</span> <span class="n">start</span> <span class="n">of</span> <span class="n">this</span> <span class="n">feature</span><span class="o">.</span>
                <span class="s1">&#39;2&#39;</span> <span class="n">indicates</span> <span class="n">there</span> <span class="n">are</span> <span class="n">two</span> <span class="n">extra</span> <span class="n">bases</span> <span class="p">(</span><span class="mi">2</span><span class="n">nd</span> <span class="ow">and</span> <span class="mi">3</span><span class="n">rd</span> <span class="n">base</span> <span class="n">of</span> <span class="n">the</span>
                <span class="n">prior</span> <span class="n">exon</span><span class="p">)</span> <span class="n">before</span> <span class="n">the</span> <span class="n">first</span> <span class="n">codon</span><span class="o">.</span> <span class="n">All</span> <span class="n">values</span> <span class="n">are</span> <span class="n">given</span> <span class="k">with</span>
                <span class="n">relation</span> <span class="n">to</span> <span class="n">the</span> <span class="mi">5</span><span class="s1">&#39; end.</span>
    <span class="n">attribute</span> <span class="o">-</span> <span class="n">a</span> <span class="n">semicolon</span><span class="o">-</span><span class="n">separated</span> <span class="nb">list</span> <span class="n">of</span> <span class="n">tag</span><span class="o">-</span><span class="n">value</span> <span class="n">pairs</span> <span class="p">(</span><span class="n">separated</span> <span class="n">by</span> <span class="n">a</span> <span class="n">space</span><span class="p">),</span>
                <span class="n">providing</span> <span class="n">additional</span> <span class="n">information</span> <span class="n">about</span> <span class="n">each</span> <span class="n">feature</span><span class="o">.</span> <span class="n">A</span> <span class="n">key</span> <span class="n">can</span> <span class="n">be</span>
                <span class="n">repeated</span> <span class="n">multiple</span> <span class="n">times</span><span class="o">.</span>

<span class="n">Attributes</span>

<span class="n">The</span> <span class="n">following</span> <span class="n">attributes</span> <span class="n">are</span> <span class="n">available</span><span class="o">.</span> <span class="n">All</span> <span class="n">attributes</span> <span class="n">are</span> <span class="n">semi</span><span class="o">-</span><span class="n">colon</span>
<span class="n">separated</span> <span class="n">pairs</span> <span class="n">of</span> <span class="n">keys</span> <span class="ow">and</span> <span class="n">values</span><span class="o">.</span>

<span class="o">-</span> <span class="n">gene_id</span><span class="p">:</span> <span class="n">The</span> <span class="n">stable</span> <span class="n">identifier</span> <span class="k">for</span> <span class="n">the</span> <span class="n">gene</span>
<span class="o">-</span> <span class="n">gene_version</span><span class="p">:</span> <span class="n">The</span> <span class="n">stable</span> <span class="n">identifier</span> <span class="n">version</span> <span class="k">for</span> <span class="n">the</span> <span class="n">gene</span>
<span class="o">-</span> <span class="n">gene_name</span><span class="p">:</span> <span class="n">The</span> <span class="n">official</span> <span class="n">symbol</span> <span class="n">of</span> <span class="n">this</span> <span class="n">gene</span>
<span class="o">-</span> <span class="n">gene_source</span><span class="p">:</span> <span class="n">The</span> <span class="n">annotation</span> <span class="n">source</span> <span class="k">for</span> <span class="n">this</span> <span class="n">gene</span>
<span class="o">-</span> <span class="n">gene_biotype</span><span class="p">:</span> <span class="n">The</span> <span class="n">biotype</span> <span class="n">of</span> <span class="n">this</span> <span class="n">gene</span>
<span class="o">-</span> <span class="n">transcript_id</span><span class="p">:</span> <span class="n">The</span> <span class="n">stable</span> <span class="n">identifier</span> <span class="k">for</span> <span class="n">this</span> <span class="n">transcript</span>
<span class="o">-</span> <span class="n">transcript_version</span><span class="p">:</span> <span class="n">The</span> <span class="n">stable</span> <span class="n">identifier</span> <span class="n">version</span> <span class="k">for</span> <span class="n">this</span> <span class="n">transcript</span>
<span class="o">-</span> <span class="n">transcript_name</span><span class="p">:</span> <span class="n">The</span> <span class="n">symbold</span> <span class="k">for</span> <span class="n">this</span> <span class="n">transcript</span> <span class="n">derived</span> <span class="kn">from</span> <span class="nn">the</span> <span class="n">gene</span> <span class="n">name</span>
<span class="o">-</span> <span class="n">transcript_source</span><span class="p">:</span> <span class="n">The</span> <span class="n">annotation</span> <span class="n">source</span> <span class="k">for</span> <span class="n">this</span> <span class="n">transcript</span>
<span class="o">-</span> <span class="n">transcript_biotype</span><span class="p">:</span> <span class="n">The</span> <span class="n">biotype</span> <span class="k">for</span> <span class="n">this</span> <span class="n">transcript</span>
<span class="o">-</span> <span class="n">exon_id</span><span class="p">:</span> <span class="n">The</span> <span class="n">stable</span> <span class="n">identifier</span> <span class="k">for</span> <span class="n">this</span> <span class="n">exon</span>
<span class="o">-</span> <span class="n">exon_version</span><span class="p">:</span> <span class="n">The</span> <span class="n">stable</span> <span class="n">identifier</span> <span class="n">version</span> <span class="k">for</span> <span class="n">this</span> <span class="n">exon</span>
<span class="o">-</span> <span class="n">exon_number</span><span class="p">:</span> <span class="n">Position</span> <span class="n">of</span> <span class="n">this</span> <span class="n">exon</span> <span class="ow">in</span> <span class="n">the</span> <span class="n">transcript</span>
<span class="o">-</span> <span class="n">ccds_id</span><span class="p">:</span> <span class="n">CCDS</span> <span class="n">identifier</span> <span class="n">linked</span> <span class="n">to</span> <span class="n">this</span> <span class="n">transcript</span>
<span class="o">-</span> <span class="n">protein_id</span><span class="p">:</span> <span class="n">Stable</span> <span class="n">identifier</span> <span class="k">for</span> <span class="n">this</span> <span class="n">transcript</span><span class="s1">&#39;s protein</span>
<span class="o">-</span> <span class="n">protein_version</span><span class="p">:</span> <span class="n">Stable</span> <span class="n">identifier</span> <span class="n">version</span> <span class="k">for</span> <span class="n">this</span> <span class="n">transcript</span><span class="s1">&#39;s protein</span>
<span class="o">-</span> <span class="n">tag</span><span class="p">:</span> <span class="n">A</span> <span class="n">collection</span> <span class="n">of</span> <span class="n">additional</span> <span class="n">key</span> <span class="n">value</span> <span class="n">tags</span>
<span class="o">-</span> <span class="n">transcript_support_level</span><span class="p">:</span> <span class="n">Ranking</span> <span class="n">to</span> <span class="n">assess</span> <span class="n">how</span> <span class="n">well</span> <span class="n">a</span> <span class="n">transcript</span> <span class="ow">is</span> <span class="n">supported</span> <span class="p">(</span><span class="kn">from</span> <span class="mi">1</span> <span class="n">to</span> <span class="mi">5</span><span class="p">)</span>
</pre></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[72]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nv">DATA_DIR</span><span class="o">=</span>~/seqs
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[73]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls <span class="si">${</span><span class="nv">DATA_DIR</span><span class="si">}</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CM000046.1.fa         <span class="ansi-red-intense-fg ansi-bold">cneo040623.scaffs.fa.gz</span>
CM000047.1.fa         Cryptococcus_neoformans.ASM9104v1.43.gtf
CM000047.1.fa.bak     <span class="ansi-red-intense-fg ansi-bold">Cryptococcus_neoformans.ASM9104v1.43.gtf.gz</span>
cneo040623.scaffs.fa
</pre></div></div>
</div>
<div class="section" id="Inspect-the-GTF-file">
<h3>Inspect the GTF file<a class="headerlink" href="#Inspect-the-GTF-file" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[74]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$GTF</span> <span class="p">|</span> head
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
#!genome-build ASM9104v1
#!genome-version ASM9104v1
#!genome-date 2016-06
#!genome-build-accession GCA_000091045.1
#!genebuild-last-updated 2016-06
1       ena     gene    4839    6236    .       +       .       gene_id &#34;CNA00010&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
1       ena     transcript      4839    6236    .       +       .       gene_id &#34;CNA00010&#34;; transcript_id &#34;AAW41410&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;;
1       ena     exon    4839    5178    .       +       .       gene_id &#34;CNA00010&#34;; transcript_id &#34;AAW41410&#34;; exon_number &#34;1&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;; exon_id &#34;AAW41410-1&#34;; exon_version &#34;1&#34;;
1       ena     CDS     4914    5178    .       +       0       gene_id &#34;CNA00010&#34;; transcript_id &#34;AAW41410&#34;; exon_number &#34;1&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;; protein_id &#34;AAW41410&#34;;
1       ena     start_codon     4914    4916    .       +       0       gene_id &#34;CNA00010&#34;; transcript_id &#34;AAW41410&#34;; exon_number &#34;1&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;;
</pre></div></div>
</div>
</div>
<div class="section" id="Ignore-comment-lines-version-1">
<h3>Ignore comment lines version 1<a class="headerlink" href="#Ignore-comment-lines-version-1" title="Permalink to this headline">¶</a></h3>
<p>This uses <code class="docutils literal notranslate"><span class="pre">tail</span></code> to take all lines starting from line 6</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[75]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$GTF</span> <span class="p">|</span> tail +6 <span class="p">|</span> head -3
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1       ena     gene    4839    6236    .       +       .       gene_id &#34;CNA00010&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
1       ena     transcript      4839    6236    .       +       .       gene_id &#34;CNA00010&#34;; transcript_id &#34;AAW41410&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;;
1       ena     exon    4839    5178    .       +       .       gene_id &#34;CNA00010&#34;; transcript_id &#34;AAW41410&#34;; exon_number &#34;1&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;; exon_id &#34;AAW41410-1&#34;; exon_version &#34;1&#34;;
</pre></div></div>
</div>
</div>
<div class="section" id="Ingore-comment-lines-2">
<h3>Ingore comment lines 2<a class="headerlink" href="#Ingore-comment-lines-2" title="Permalink to this headline">¶</a></h3>
<p>This uses <code class="docutils literal notranslate"><span class="pre">grep</span></code> to ignore (<code class="docutils literal notranslate"><span class="pre">-v</span></code>) lines starting with <code class="docutils literal notranslate"><span class="pre">^#</span></code></p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[76]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$GTF</span> <span class="p">|</span> grep -v <span class="s1">&#39;^#&#39;</span> <span class="p">|</span> head -3
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1       ena     gene    4839    6236    .       +       .       gene_id &#34;CNA00010&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
1       ena     transcript      4839    6236    .       +       .       gene_id &#34;CNA00010&#34;; transcript_id &#34;AAW41410&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;;
1       ena     exon    4839    5178    .       +       .       gene_id &#34;CNA00010&#34;; transcript_id &#34;AAW41410&#34;; exon_number &#34;1&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;; exon_id &#34;AAW41410-1&#34;; exon_version &#34;1&#34;;
</pre></div></div>
</div>
</div>
<div class="section" id="Find-all-unique-feature-types-in-GTF">
<h3>Find all unique feature types in GTF<a class="headerlink" href="#Find-all-unique-feature-types-in-GTF" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[77]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$GTF</span> <span class="p">|</span> grep -v <span class="s1">&#39;^#&#39;</span> <span class="p">|</span> cut -f <span class="m">3</span> <span class="p">|</span> sort <span class="p">|</span> uniq
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CDS
exon
five_prime_utr
gene
start_codon
stop_codon
three_prime_utr
transcript
</pre></div></div>
</div>
</div>
<div class="section" id="Get-counts-of-all-unique-feature-types-in-GTF">
<h3>Get counts of all unique feature types in GTF<a class="headerlink" href="#Get-counts-of-all-unique-feature-types-in-GTF" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[78]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$GTF</span> <span class="p">|</span> grep -v <span class="s1">&#39;^#&#39;</span> <span class="p">|</span> cut -f <span class="m">3</span> <span class="p">|</span> sort <span class="p">|</span> uniq -c
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
  44079 CDS
  44271 exon
   6738 five_prime_utr
   6813 gene
   6894 start_codon
   2369 stop_codon
   6718 three_prime_utr
   7044 transcript
</pre></div></div>
</div>
</div>
<div class="section" id="Find-the-the-attributes-of-the-first-3-exons">
<h3>Find the the attributes of the first 3 exons<a class="headerlink" href="#Find-the-the-attributes-of-the-first-3-exons" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[79]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$GTF</span> <span class="p">|</span> grep -v <span class="s1">&#39;^#&#39;</span> <span class="p">|</span> awk <span class="s1">&#39;$3 == &quot;exon&quot;&#39;</span> <span class="p">|</span> head -3
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1       ena     exon    4839    5178    .       +       .       gene_id &#34;CNA00010&#34;; transcript_id &#34;AAW41410&#34;; exon_number &#34;1&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;; exon_id &#34;AAW41410-1&#34;; exon_version &#34;1&#34;;
1       ena     exon    5241    5377    .       +       .       gene_id &#34;CNA00010&#34;; transcript_id &#34;AAW41410&#34;; exon_number &#34;2&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;; exon_id &#34;AAW41410-2&#34;; exon_version &#34;1&#34;;
1       ena     exon    5453    5560    .       +       .       gene_id &#34;CNA00010&#34;; transcript_id &#34;AAW41410&#34;; exon_number &#34;3&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;; transcript_source &#34;ena&#34;; transcript_biotype &#34;protein_coding&#34;; exon_id &#34;AAW41410-3&#34;; exon_version &#34;1&#34;;
</pre></div></div>
</div>
</div>
<div class="section" id="Store-attributes-of-all-genes-in-a-file">
<h3>Store attributes of all genes in a file<a class="headerlink" href="#Store-attributes-of-all-genes-in-a-file" title="Permalink to this headline">¶</a></h3>
<p>It is common to use Unix commands to do simple processing and save the results to a file, then use a scripting language such as R or Python to furhter process the file.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[80]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>zcat <span class="nv">$DATA_DIR</span>/<span class="nv">$GTF</span> <span class="p">|</span> grep -v <span class="s1">&#39;^#&#39;</span> <span class="p">|</span> awk <span class="s1">&#39;$3 == &quot;gene&quot;&#39;</span> <span class="p">|</span> cut -f <span class="m">9</span> &gt; genes.txt
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[81]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>head -6 genes.txt
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
gene_id &#34;CNA00010&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
gene_id &#34;CNA00020&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
gene_id &#34;CNA00030&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
gene_id &#34;CNA00040&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
gene_id &#34;CNA00050&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
gene_id &#34;CNA00060&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
</pre></div></div>
</div>
</div>
</div>
<div class="section" id="4.-Looping">
<h2>4. Looping<a class="headerlink" href="#4.-Looping" title="Permalink to this headline">¶</a></h2>
<div class="section" id="Loop-throgh-first-10-lines">
<h3>Loop throgh first 10 lines<a class="headerlink" href="#Loop-throgh-first-10-lines" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>Creae a counter variable <code class="docutils literal notranslate"><span class="pre">LINE_NUM</span></code> and initialize it to 0</p></li>
<li><p>Redirect <code class="docutils literal notranslate"><span class="pre">genes.txt</span></code> to a while loop</p></li>
<li><p>Whlie loop reads each line into the variable <code class="docutils literal notranslate"><span class="pre">LINE</span></code></p></li>
<li><p>The <code class="docutils literal notranslate"><span class="pre">[[</span> <span class="pre">CONDITION</span> <span class="pre">]]</span></code> tests if CONDITION is true</p></li>
<li><p>If LINE_NUM is less than <code class="docutils literal notranslate"><span class="pre">-lt</span></code> 10 print LINE and increment LINE_NUM</p></li>
<li><p>Note that <code class="docutils literal notranslate"><span class="pre">((</span> <span class="pre">EXPR</span> <span class="pre">))</span></code> is used to evaluate numeric expressions</p></li>
</ul>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[82]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nv">LINE_NUM</span><span class="o">=</span><span class="m">0</span>
<span class="k">while</span> <span class="nb">read</span> LINE
<span class="k">do</span>
    <span class="k">if</span> <span class="o">[[</span> <span class="nv">$LINE_NUM</span> -lt <span class="m">10</span> <span class="o">]]</span><span class="p">;</span> <span class="k">then</span>
        <span class="nb">echo</span> <span class="nv">$LINE</span>
        <span class="o">((</span> LINE_NUM++ <span class="o">))</span>
    <span class="k">fi</span>
<span class="k">done</span> &lt; genes.txt
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
gene_id &#34;CNA00010&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
gene_id &#34;CNA00020&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
gene_id &#34;CNA00030&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
gene_id &#34;CNA00040&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
gene_id &#34;CNA00050&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
gene_id &#34;CNA00060&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
gene_id &#34;CNA00070&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
gene_id &#34;CNA00075&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
gene_id &#34;CNA00080&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
gene_id &#34;CNA00090&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;protein_coding&#34;;
</pre></div></div>
</div>
</div>
<div class="section" id="Loop-through-first-10-tRNAs">
<h3>Loop through first 10 tRNAs<a class="headerlink" href="#Loop-through-first-10-tRNAs" title="Permalink to this headline">¶</a></h3>
<p>Almost the same as previous program but we add a new condition to check tha the gene_biotype is tRNA using a REGULAr EXPRESSION via the match operator <code class="docutils literal notranslate"><span class="pre">=~</span></code></p>
<p><code class="docutils literal notranslate"><span class="pre">.*gene_biotype[[:space:]]\&quot;tRNA\&quot;.*</span></code> means</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">.*</span></code>: Match any number of any character</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">gene_biotype</span></code>: Match the string gene_biotype</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">[[:space:]]</span></code>: Match a single space cahracter</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">\&quot;tRNA\&quot;</span></code>: Math the string “tRNA” - we escape the double quotes to match linteral double quote characters</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.*</span></code>: Match any number of any character</p></li>
</ul>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[83]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nv">LINE_NUM</span><span class="o">=</span><span class="m">0</span>
<span class="k">while</span> <span class="nb">read</span> LINE
<span class="k">do</span>
    <span class="k">if</span> <span class="o">[[</span> <span class="nv">$LINE_NUM</span> -lt <span class="m">10</span> <span class="o">]]</span> <span class="o">&amp;&amp;</span> <span class="o">[[</span> <span class="nv">$LINE</span> <span class="o">=</span>~ .*gene_biotype<span class="o">[[</span>:space:<span class="o">]]</span><span class="se">\&quot;</span>tRNA<span class="se">\&quot;</span>.* <span class="o">]]</span> <span class="p">;</span> <span class="k">then</span>
        <span class="nb">echo</span> <span class="nv">$LINE</span>
        <span class="o">((</span>LINE_NUM++<span class="o">))</span>
    <span class="k">fi</span>
<span class="k">done</span> &lt; genes.txt
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
gene_id &#34;CNA02110&#34;; gene_name &#34;CNA02110&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;tRNA&#34;;
gene_id &#34;EBG00005237496&#34;; gene_name &#34;tRNA&#34;; gene_source &#34;Rfam&#34;; gene_biotype &#34;tRNA&#34;;
gene_id &#34;CNA02190&#34;; gene_name &#34;CNA02190&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;tRNA&#34;;
gene_id &#34;CNA02200&#34;; gene_name &#34;CNA02200&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;tRNA&#34;;
gene_id &#34;CNA02280&#34;; gene_name &#34;CNA02280&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;tRNA&#34;;
gene_id &#34;EBG00005237502&#34;; gene_name &#34;tRNA&#34;; gene_source &#34;Rfam&#34;; gene_biotype &#34;tRNA&#34;;
gene_id &#34;CNA02320&#34;; gene_name &#34;CNA02320&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;tRNA&#34;;
gene_id &#34;CNA03230&#34;; gene_name &#34;CNA03230&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;tRNA&#34;;
gene_id &#34;EBG00005237494&#34;; gene_name &#34;tRNA&#34;; gene_source &#34;Rfam&#34;; gene_biotype &#34;tRNA&#34;;
gene_id &#34;CNA03260&#34;; gene_name &#34;CNA03260&#34;; gene_source &#34;ena&#34;; gene_biotype &#34;tRNA&#34;;
</pre></div></div>
</div>
</div>
<div class="section" id="Loop-through-files-in-a-directory-and-print-the-filename-followed-by-the-first-line">
<h3>Loop through files in a directory and print the filename followed by the first line<a class="headerlink" href="#Loop-through-files-in-a-directory-and-print-the-filename-followed-by-the-first-line" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>Use a COMMAND EXPRESSION <code class="docutils literal notranslate"><span class="pre">$(EXPR)</span></code> to list files matching the glob pattern</p></li>
<li><p>We loop through each file using a <code class="docutils literal notranslate"><span class="pre">for</span></code> statement</p></li>
<li><p>For each file, extract the first line with <code class="docutils literal notranslate"><span class="pre">head</span> <span class="pre">-1</span></code> and save in variable <code class="docutils literal notranslate"><span class="pre">FIRST</span></code></p></li>
<li><p>The <code class="docutils literal notranslate"><span class="pre">2&gt;/dev/null</span></code> ignores output from the <code class="docutils literal notranslate"><span class="pre">stderr</span></code> stream</p></li>
<li><p>We use another COMMAND EXPRESSION to print the filename withou the directory (using <code class="docutils literal notranslate"><span class="pre">basename</span></code>)</p></li>
<li><p>We then print the first line storred in <code class="docutils literal notranslate"><span class="pre">FIRST</span></code></p></li>
</ul>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[84]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nv">DATA_DIR</span><span class="o">=</span><span class="s1">&#39;/data/hts2018_pilot/Granek_4837_180427A5&#39;</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[85]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="k">for</span> FILE in <span class="k">$(</span>ls <span class="nv">$DATA_DIR</span>/*_MA_J_S20_*gz<span class="k">)</span>
<span class="k">do</span>
    <span class="nv">FIRST</span><span class="o">=</span><span class="k">$(</span>zcat <span class="nv">$FILE</span> <span class="m">2</span>&gt;/dev/null <span class="p">|</span> head -1<span class="k">)</span>
    <span class="nb">echo</span> <span class="k">$(</span>basename <span class="nv">$FILE</span><span class="k">)</span>
    <span class="nb">echo</span> <span class="nv">$FIRST</span>
    <span class="nb">echo</span>
<span class="k">done</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
11_MA_J_S20_L001_R1_001.fastq.gz
@NB501800:128:HTYKFBGX5:1:11101:26760:1056 1:N:0:CGCTCATT+NGGATAGG

11_MA_J_S20_L002_R1_001.fastq.gz
@NB501800:128:HTYKFBGX5:2:11101:20327:1041 1:N:0:CGCTCATT+AGGATAGN

11_MA_J_S20_L003_R1_001.fastq.gz
@NB501800:128:HTYKFBGX5:3:11401:10896:1024 1:N:0:CGCTCATT+AGGATAGG

11_MA_J_S20_L004_R1_001.fastq.gz
@NB501800:128:HTYKFBGX5:4:11401:8259:1032 1:N:0:CGCTCATT+AGGATAGG

</pre></div></div>
</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Unix07_Capstone_Exercise.html" class="btn btn-neutral float-right" title="Unix capstone exercise" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Unix05_Variables_Solutions.html" class="btn btn-neutral float-left" title="Unix Variables" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>