

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>DESeq2: Basic Differential Expression (DE) analysis &mdash; HTS2019 1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../_static/jquery.js"></script>
        <script type="text/javascript" src="../_static/underscore.js"></script>
        <script type="text/javascript" src="../_static/doctools.js"></script>
        <script type="text/javascript" src="../_static/language_data.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "document", "processClass": "math|output_area"}})</script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Generate figures from DESEq2 analysis" href="04-deseq2-plots.html" />
    <link rel="prev" title="DESeq2: Basic Differential Expression (DE) analysis" href="02_deseq_DE.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> HTS2019
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Slides</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../slides.html">Slide sets</a></li>
</ul>
<p class="caption"><span class="caption-text">Quick Intro of Analysis Tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../janice/1_IntroToCompBoot.html">Duke HTS Summer Course 2018</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/2_Intro_to_Unix_and_bash.html">Introduction to Unix and bash</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/3_IntroductionToR.html">Introduction to R</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/4_Rlibraries_and_bioconductor.html">R libraries and Bioconductor</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/5_Tidyverse_HandsOn-Filled.html">Hands-On Tidyverse - with count data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/6_Statistical_Inference.html">Simulations and Statistical Inference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/7_Statistical_Inference_Solutions.html">Estimation and Hypothesis Testing</a></li>
</ul>
<p class="caption"><span class="caption-text">Unix Shell</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix01_File_And_Directory_Solutions.html">Working with Unix</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix02_FileIO_Solutions.html">File I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix03_File_Storage_Solutions.html">File archival and recovery</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix04_Text_Manipulation_Solutinos.html">Working with text</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix05_Variables_Solutions.html">Unix Variables</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix06_Bash_Bioinformatics.html">Using <code class="docutils literal notranslate"><span class="pre">bash</span></code> in bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix07_Capstone_Exercise.html">Unix capstone exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix07_Capstone_Exercise_Solutions.html">Unix capstone exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix_Appendix01_Regular_Expressions.html">The Unix Shell: Regular Expressions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix_Appendix02_Review.html">Unix Review of Basics</a></li>
</ul>
<p class="caption"><span class="caption-text">R Graphics and Data Manipulation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R00_Review_Basics.html">Basics of R (Review)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R00_Review_Basics.html">Basics of R (Review)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R01_Data_Manipulation_Solutions.html">Data Manipulation in the <code class="docutils literal notranslate"><span class="pre">tidyverse</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R02_Tidying_Data_Solutions.html">Tidy Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R03_FileIO.html">File I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R04_Unsupervised_Learning.html">Unsupervised Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R05_Unsupervised_Learning_More_Examples.html">Unsupervised Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R06_Graphics_Overview.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R07_Graphics_Base.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R08_Graphics_ggplot2.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R09_Graphics_Exercise.html">R Graphics Exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R09_Graphics_Exercise_Solutions.html">R Graphics Exercise (Solutions)</a></li>
</ul>
<p class="caption"><span class="caption-text">Analysis of Pilot Data (Gene Count, DESeq2 &amp; DE Analysis)</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="8_Pilot2019_getstarcounts-KO.html">Get Star Count of the Pilot 2019 Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="00_deseq_prepare_data.html">DESeq2: Create count matrix and metadata data frame</a></li>
<li class="toctree-l1"><a class="reference internal" href="01_deseq_dataobject.html">DESeq2: Create Data Objects</a></li>
<li class="toctree-l1"><a class="reference internal" href="02_deseq_DE.html">DESeq2: Basic Differential Expression (DE) analysis</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">DESeq2: Basic Differential Expression (DE) analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#Objective:-Carry-out-a-basic-set-of-DE-interaction-analysis-using-DESeq2-and-visualize-the-results">Objective: Carry out a basic set of DE interaction analysis using DESeq2 and visualize the results</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Load-packages">Load packages</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#Load-the-2019-pilot-dds-object-from-image-file">Load the 2019 pilot dds object from image file</a></li>
<li class="toctree-l2"><a class="reference internal" href="#review-the-steps-of-DE-analysis">review the steps of DE analysis</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Before-carrying-out-an-interaction-analysis,-let's-review-the-steps-for-conducting-a-DE-analysis">Before carrying out an interaction analysis, letâ€™s review the steps for conducting a DE analysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Inspect-the-dds2019-object">Inspect the dds2019 object</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Note-that-the-design-is-additive">Note that the design is additive</a></li>
<li class="toctree-l3"><a class="reference internal" href="#The-steps-for-a-basic-analysis-are:-estimate-size-factors,-estimate-dispersion-parameters,-and-then-carry-out-DE-analysis">The steps for a basic analysis are: estimate size factors, estimate dispersion parameters, and then carry out DE analysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Identify-differentially-expressed-genes-with-respect-to-condition-using-pH4-as-reference">Identify differentially expressed genes with respect to condition using pH4 as reference</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#Exercise:-Compare-the-size-factors-and-dispersion-estimates-between-ddsmult-and-ddsad">Exercise: Compare the size factors and dispersion estimates between ddsmult and ddsad</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Estimate-DE-effect-due-to-condition-given-genotype">Estimate DE effect due to condition given genotype</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Define-a-group-variable">Define a group variable</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Set-design">Set design</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Generate-results-for-CNAG_03518">Generate results for CNAG_03518</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="04-deseq2-plots.html">Generate figures from DESEq2 analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">Reproducible Analysis (Git)</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../computing/data_handling_demo.html">Data handling Demo</a></li>
<li class="toctree-l1"><a class="reference internal" href="../computing/git_lesson.html">Git Lesson</a></li>
<li class="toctree-l1"><a class="reference internal" href="../computing/git_in_class.html">Git in class</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">HTS2019</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>DESeq2: Basic Differential Expression (DE) analysis</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/DESeq2/03_deseq_interaction.ipynb.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 5ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast,
.nboutput.nblast {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast + .nbinput {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="DESeq2:-Basic-Differential-Expression-(DE)-analysis">
<h1>DESeq2: Basic Differential Expression (DE) analysis<a class="headerlink" href="#DESeq2:-Basic-Differential-Expression-(DE)-analysis" title="Permalink to this headline">Â¶</a></h1>
<div class="section" id="Objective:-Carry-out-a-basic-set-of-DE-interaction-analysis-using-DESeq2-and-visualize-the-results">
<h2>Objective: Carry out a basic set of DE interaction analysis using DESeq2 and visualize the results<a class="headerlink" href="#Objective:-Carry-out-a-basic-set-of-DE-interaction-analysis-using-DESeq2-and-visualize-the-results" title="Permalink to this headline">Â¶</a></h2>
<p>The objective is to identify genes whose differential effect due to treatment depends on the genotype</p>
<div class="section" id="Load-packages">
<h3>Load packages<a class="headerlink" href="#Load-packages" title="Permalink to this headline">Â¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="kn">library</span><span class="p">(</span>tidyverse<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>DESeq2<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>dendextend<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>RColorBrewer<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
Registered S3 methods overwritten by &#39;ggplot2&#39;:
  method         from
  [.quosures     rlang
  c.quosures     rlang
  print.quosures rlang
â”€â”€ <span class="ansi-bold">Attaching packages</span> â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ tidyverse 1.2.1 â”€â”€
<span class="ansi-green-fg">âœ”</span> <span class="ansi-blue-fg">ggplot2</span> 3.1.1     <span class="ansi-green-fg">âœ”</span> <span class="ansi-blue-fg">purrr  </span> 0.3.2
<span class="ansi-green-fg">âœ”</span> <span class="ansi-blue-fg">tibble </span> 2.1.2     <span class="ansi-green-fg">âœ”</span> <span class="ansi-blue-fg">dplyr  </span> 0.8.1
<span class="ansi-green-fg">âœ”</span> <span class="ansi-blue-fg">tidyr  </span> 0.8.3     <span class="ansi-green-fg">âœ”</span> <span class="ansi-blue-fg">stringr</span> 1.4.0
<span class="ansi-green-fg">âœ”</span> <span class="ansi-blue-fg">readr  </span> 1.3.1     <span class="ansi-green-fg">âœ”</span> <span class="ansi-blue-fg">forcats</span> 0.4.0
â”€â”€ <span class="ansi-bold">Conflicts</span> â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ tidyverse_conflicts() â”€â”€
<span class="ansi-red-fg">âœ–</span> <span class="ansi-blue-fg">dplyr</span>::<span class="ansi-green-fg">filter()</span> masks <span class="ansi-blue-fg">stats</span>::filter()
<span class="ansi-red-fg">âœ–</span> <span class="ansi-blue-fg">dplyr</span>::<span class="ansi-green-fg">lag()</span>    masks <span class="ansi-blue-fg">stats</span>::lag()
Loading required package: S4Vectors
Loading required package: stats4
Loading required package: BiocGenerics
Loading required package: parallel

Attaching package: â€˜BiocGenericsâ€™

The following objects are masked from â€˜package:parallelâ€™:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB

The following objects are masked from â€˜package:dplyrâ€™:

    combine, intersect, setdiff, union

The following objects are masked from â€˜package:statsâ€™:

    IQR, mad, sd, var, xtabs

The following objects are masked from â€˜package:baseâ€™:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which, which.max, which.min


Attaching package: â€˜S4Vectorsâ€™

The following objects are masked from â€˜package:dplyrâ€™:

    first, rename

The following object is masked from â€˜package:tidyrâ€™:

    expand

The following object is masked from â€˜package:baseâ€™:

    expand.grid

Loading required package: IRanges

Attaching package: â€˜IRangesâ€™

The following objects are masked from â€˜package:dplyrâ€™:

    collapse, desc, slice

The following object is masked from â€˜package:purrrâ€™:

    reduce

Loading required package: GenomicRanges
Loading required package: GenomeInfoDb
Loading required package: SummarizedExperiment
Loading required package: Biobase
Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&#34;Biobase&#34;)&#39;, and for packages &#39;citation(&#34;pkgname&#34;)&#39;.

Loading required package: DelayedArray
Loading required package: matrixStats

Attaching package: â€˜matrixStatsâ€™

The following objects are masked from â€˜package:Biobaseâ€™:

    anyMissing, rowMedians

The following object is masked from â€˜package:dplyrâ€™:

    count

Loading required package: BiocParallel

Attaching package: â€˜DelayedArrayâ€™

The following objects are masked from â€˜package:matrixStatsâ€™:

    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges

The following object is masked from â€˜package:purrrâ€™:

    simplify

The following objects are masked from â€˜package:baseâ€™:

    aperm, apply, rowsum


---------------------
Welcome to dendextend version 1.12.0
Type citation(&#39;dendextend&#39;) for how to cite the package.

Type browseVignettes(package = &#39;dendextend&#39;) for the package vignette.
The github page is: https://github.com/talgalili/dendextend/

Suggestions and bug-reports can be submitted at: https://github.com/talgalili/dendextend/issues
Or contact: &lt;tal.galili@gmail.com&gt;

        To suppress this message use:  suppressPackageStartupMessages(library(dendextend))
---------------------


Attaching package: â€˜dendextendâ€™

The following object is masked from â€˜package:statsâ€™:

    cutree

</pre></div></div>
</div>
</div>
</div>
<div class="section" id="Load-the-2019-pilot-dds-object-from-image-file">
<h2>Load the 2019 pilot dds object from image file<a class="headerlink" href="#Load-the-2019-pilot-dds-object-from-image-file" title="Permalink to this headline">Â¶</a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>curdir <span class="o">&lt;-</span> <span class="s">&quot;/home/jovyan/work/scratch/analysis_output&quot;</span>
imgdir <span class="o">&lt;-</span> <span class="kp">file.path</span><span class="p">(</span>curdir<span class="p">,</span> <span class="s">&quot;img&quot;</span><span class="p">)</span>

imgfile <span class="o">&lt;-</span> <span class="kp">file.path</span><span class="p">(</span>imgdir<span class="p">,</span> <span class="s">&quot;pilotdds2019.RData&quot;</span><span class="p">)</span>

imgfile

<span class="kn">attach</span><span class="p">(</span>imgfile<span class="p">)</span>

tools<span class="o">::</span>md5sum<span class="p">(</span>imgfile<span class="p">)</span>

<span class="c1">### List the objects that have been attached</span>
<span class="kp">ls</span><span class="p">(</span><span class="m">2</span><span class="p">)</span>

dds2019 <span class="o">&lt;-</span> dds2019

<span class="kn">detach</span><span class="p">(</span>pos <span class="o">=</span> <span class="m">2</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
'/home/jovyan/work/scratch/analysis_output/img/pilotdds2019.RData'</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<strong>/home/jovyan/work/scratch/analysis_output/img/pilotdds2019.RData:</strong> '6632de5c8a2eed06d8c40c958e6f5d03'</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
'dds2019'</div>
</div>
</div>
<div class="section" id="review-the-steps-of-DE-analysis">
<h2>review the steps of DE analysis<a class="headerlink" href="#review-the-steps-of-DE-analysis" title="Permalink to this headline">Â¶</a></h2>
<div class="section" id="Before-carrying-out-an-interaction-analysis,-let's-review-the-steps-for-conducting-a-DE-analysis">
<h3>Before carrying out an interaction analysis, letâ€™s review the steps for conducting a DE analysis<a class="headerlink" href="#Before-carrying-out-an-interaction-analysis,-let's-review-the-steps-for-conducting-a-DE-analysis" title="Permalink to this headline">Â¶</a></h3>
</div>
<div class="section" id="Inspect-the-dds2019-object">
<h3>Inspect the dds2019 object<a class="headerlink" href="#Inspect-the-dds2019-object" title="Permalink to this headline">Â¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>dds2019
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area highlight-none notranslate"><div class="highlight"><pre>
<span></span>class: DESeqDataSet
dim: 8499 24
metadata(1): version
assays(1): counts
rownames(8499): CNAG_00001 CNAG_00002 ... large_MTrRNA small_MTrRNA
rowData names(0):
colnames(24): 1_2019_P_M1 2_2019_P_M1 ... 23_2019_P_M1 24_2019_P_M1
colData names(22): Label sample_year ... RIN_normal_threshold
  RIN_lowered_threshold
</pre></div>
</div>
</div>
</div>
<div class="section" id="Note-that-the-design-is-additive">
<h3>Note that the design is additive<a class="headerlink" href="#Note-that-the-design-is-additive" title="Permalink to this headline">Â¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>design<span class="p">(</span>dds2019<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area highlight-none notranslate"><div class="highlight"><pre>
<span></span>~condition + genotype
</pre></div>
</div>
</div>
</div>
<div class="section" id="The-steps-for-a-basic-analysis-are:-estimate-size-factors,-estimate-dispersion-parameters,-and-then-carry-out-DE-analysis">
<h3>The steps for a basic analysis are: estimate size factors, estimate dispersion parameters, and then carry out DE analysis<a class="headerlink" href="#The-steps-for-a-basic-analysis-are:-estimate-size-factors,-estimate-dispersion-parameters,-and-then-carry-out-DE-analysis" title="Permalink to this headline">Â¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="c1">### First copy dds2019 to ddsadd</span>
ddsadd <span class="o">&lt;-</span> dds2019
<span class="c1">### Estimate Size Factors</span>
ddsadd <span class="o">&lt;-</span> estimateSizeFactors<span class="p">(</span>ddsadd<span class="p">)</span>
<span class="c1">### Estimate Dispersion parameters (for each gene)</span>
ddsadd <span class="o">&lt;-</span> estimateDispersions<span class="p">(</span>ddsadd<span class="p">)</span>
<span class="c1">### Fit NB MLE model</span>
ddsadd <span class="o">&lt;-</span> DESeq<span class="p">(</span>ddsadd<span class="p">)</span>
<span class="c1">### Rlog &quot;normalized&quot; expressions</span>
<span class="c1">#rldadd &lt;- rlog(ddsadd, blind = TRUE)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
using pre-existing size factors
estimating dispersions
found already estimated dispersions, replacing these
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
fitting model and testing
</pre></div></div>
</div>
</div>
<div class="section" id="Identify-differentially-expressed-genes-with-respect-to-condition-using-pH4-as-reference">
<h3>Identify differentially expressed genes with respect to condition using pH4 as reference<a class="headerlink" href="#Identify-differentially-expressed-genes-with-respect-to-condition-using-pH4-as-reference" title="Permalink to this headline">Â¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>results<span class="p">(</span>ddsadd<span class="p">,</span> contrast <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;condition&quot;</span><span class="p">,</span> <span class="s">&quot;pH8&quot;</span><span class="p">,</span> <span class="s">&quot;pH4&quot;</span><span class="p">),</span> tidy <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span> <span class="o">%&gt;%</span>
    arrange<span class="p">(</span>desc<span class="p">(</span><span class="o">-</span>padj<span class="p">))</span> <span class="o">%&gt;%</span>
        <span class="kp">head</span><span class="p">(</span><span class="m">5</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<table>
<caption>A data.frame: 5 Ã— 7</caption>
<thead>
    <tr><th scope=col>row</th><th scope=col>baseMean</th><th scope=col>log2FoldChange</th><th scope=col>lfcSE</th><th scope=col>stat</th><th scope=col>pvalue</th><th scope=col>padj</th></tr>
    <tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>CNAG_00275</td><td> 1072.9219</td><td>-4.048106</td><td>0.10667356</td><td>-37.94854</td><td>0</td><td>0</td></tr>
    <tr><td>CNAG_00309</td><td>  539.1312</td><td>-3.331524</td><td>0.08444635</td><td>-39.45137</td><td>0</td><td>0</td></tr>
    <tr><td>CNAG_00409</td><td> 1093.4775</td><td>-5.030655</td><td>0.07452918</td><td>-67.49913</td><td>0</td><td>0</td></tr>
    <tr><td>CNAG_00531</td><td>15371.8856</td><td> 3.563691</td><td>0.05663035</td><td> 62.92900</td><td>0</td><td>0</td></tr>
    <tr><td>CNAG_00601</td><td>  506.4752</td><td> 4.661577</td><td>0.09625726</td><td> 48.42832</td><td>0</td><td>0</td></tr>
</tbody>
</table></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="c1">## Interaction analysis</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="c1">### The first step is to specify the design</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="c1">### Additive design</span>
ddsmult <span class="o">&lt;-</span> dds2019
design<span class="p">(</span>ddsmult<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area highlight-none notranslate"><div class="highlight"><pre>
<span></span>~condition + genotype
</pre></div>
</div>
</div>
<p>Update the design by adding the term genotype:condition</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>design<span class="p">(</span>ddsmult<span class="p">)</span> <span class="o">&lt;-</span> formula<span class="p">(</span><span class="o">~</span> genotype <span class="o">+</span> condition <span class="o">+</span> genotype<span class="o">:</span>condition<span class="p">)</span>
design<span class="p">(</span>ddsmult<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area highlight-none notranslate"><div class="highlight"><pre>
<span></span>~genotype + condition + genotype:condition
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="c1">### Now repeat the steps: estimate size factors, estimate dispersion followed by the analysis</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="c1">### Estimate Size Factors</span>
ddsmult <span class="o">&lt;-</span> estimateSizeFactors<span class="p">(</span>ddsmult<span class="p">)</span>
<span class="c1">### Estimate Dispersion parameters (for each gene)</span>
ddsmult <span class="o">&lt;-</span> estimateDispersions<span class="p">(</span>ddsmult<span class="p">)</span>
<span class="c1">### Fit NB MLE model</span>
ddsmultres <span class="o">&lt;-</span> DESeq<span class="p">(</span>ddsmult<span class="p">)</span>
<span class="c1">### Rlog &quot;normalized&quot; expressions</span>
<span class="c1">#rldmult &lt;- rlog(ddsmult, blind = TRUE)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
using pre-existing size factors
estimating dispersions
found already estimated dispersions, replacing these
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
fitting model and testing
</pre></div></div>
</div>
<p>Look at the results (compare the first line to that of the DE analysis</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>results<span class="p">(</span>ddsmultres<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area highlight-none notranslate"><div class="highlight"><pre>
<span></span>log2 fold change (MLE): genotypeWT.conditionpH8
Wald test p-value: genotypeWT.conditionpH8
DataFrame with 8499 rows and 6 columns
                        baseMean     log2FoldChange              lfcSE
                       &lt;numeric&gt;          &lt;numeric&gt;          &lt;numeric&gt;
CNAG_00001                     0                 NA                 NA
CNAG_00002      192.456384076694 -0.103408237109234  0.162208795540166
CNAG_00003      164.990819436254  0.639703164323634  0.158566362520624
CNAG_00004      571.054651467718 -0.149923491484726 0.0875861185206605
CNAG_00005      28.7106195205257   0.38872615882583  0.383575916608188
...                          ...                ...                ...
ENSRNA049551942                0                 NA                 NA
ENSRNA049551964                0                 NA                 NA
ENSRNA049551993                0                 NA                 NA
large_MTrRNA    5333.97356461621   0.25675256982063  0.261942681080193
small_MTrRNA    1705.75472334291   1.15038162089038  0.552570647394538
                              stat               pvalue                padj
                         &lt;numeric&gt;            &lt;numeric&gt;           &lt;numeric&gt;
CNAG_00001                      NA                   NA                  NA
CNAG_00002      -0.637500801142613    0.523798688279165   0.686004971590578
CNAG_00003        4.03429298720548 5.47669483738396e-05 0.00040044394310704
CNAG_00004       -1.71172662993806   0.0869470586782988   0.188582930952285
CNAG_00005        1.01342691757914    0.310856287600497   0.484502484273056
...                            ...                  ...                 ...
ENSRNA049551942                 NA                   NA                  NA
ENSRNA049551964                 NA                   NA                  NA
ENSRNA049551993                 NA                   NA                  NA
large_MTrRNA     0.980186080259394    0.326994274282426   0.501378918298731
small_MTrRNA      2.08187247425215   0.0373541196234004  0.0979329138148396
</pre></div>
</div>
</div>
<p>List the top 5 hits with respect to adjusted P-value</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>results<span class="p">(</span>ddsmultres<span class="p">,</span> tidy <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span> <span class="o">%&gt;%</span>
    arrange<span class="p">(</span>desc<span class="p">(</span><span class="o">-</span>padj<span class="p">))</span> <span class="o">%&gt;%</span>
        <span class="kp">head</span><span class="p">(</span><span class="m">5</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<table>
<caption>A data.frame: 5 Ã— 7</caption>
<thead>
    <tr><th scope=col>row</th><th scope=col>baseMean</th><th scope=col>log2FoldChange</th><th scope=col>lfcSE</th><th scope=col>stat</th><th scope=col>pvalue</th><th scope=col>padj</th></tr>
    <tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>CNAG_06237</td><td>2425.9011</td><td> 1.445970</td><td>0.06535452</td><td> 22.12503</td><td>1.815272e-108</td><td>1.479628e-104</td></tr>
    <tr><td>CNAG_01768</td><td>5493.5145</td><td>-2.012335</td><td>0.09770819</td><td>-20.59535</td><td> 3.020861e-94</td><td> 1.231152e-90</td></tr>
    <tr><td>CNAG_04901</td><td> 475.3418</td><td>-3.273870</td><td>0.16064794</td><td>-20.37916</td><td> 2.560031e-92</td><td> 6.955603e-89</td></tr>
    <tr><td>CNAG_06658</td><td>2332.5964</td><td>-1.550563</td><td>0.08436115</td><td>-18.38006</td><td> 1.897740e-75</td><td> 3.867120e-72</td></tr>
    <tr><td>CNAG_03518</td><td> 399.4864</td><td> 3.091738</td><td>0.16854953</td><td> 18.34320</td><td> 3.741012e-75</td><td> 6.098599e-72</td></tr>
</tbody>
</table></div>
</div>
<p>Visualize the results</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="c1">### Merge gene expression with meta data</span>
myDEplotData <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span>mydds<span class="p">,</span> geneid<span class="p">,</span> mergelab<span class="p">)</span> <span class="p">{</span>
    counts<span class="p">(</span>mydds<span class="p">,</span> normalize <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span> <span class="o">%&gt;%</span>
        as_tibble<span class="p">(</span>rownames<span class="o">=</span><span class="s">&quot;gene&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
        filter<span class="p">(</span>gene <span class="o">==</span> geneid<span class="p">)</span> <span class="o">%&gt;%</span>
        gather<span class="p">(</span>Label<span class="p">,</span> geneexp<span class="p">,</span> <span class="o">-</span>gene<span class="p">)</span> <span class="o">%&gt;%</span>
        select<span class="p">(</span><span class="o">-</span>gene<span class="p">)</span> <span class="o">-&gt;</span> genedat

    colData<span class="p">(</span>mydds<span class="p">)</span> <span class="o">%&gt;%</span>
        as.data.frame <span class="o">%&gt;%</span>
        as_tibble <span class="o">%&gt;%</span>
        full_join<span class="p">(</span>genedat<span class="p">,</span> by <span class="o">=</span> mergelab<span class="p">)</span> <span class="o">-&gt;</span> genedat

    <span class="kr">return</span><span class="p">(</span>genedat<span class="p">)</span>
<span class="p">}</span>


<span class="c1">### Alow for coloring with respect to another factor</span>
myDEplot <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span>mydds<span class="p">,</span> geneid<span class="p">,</span> grpvar<span class="p">,</span> colvar<span class="p">,</span> mergelab<span class="p">)</span> <span class="p">{</span>
    mydat <span class="o">&lt;-</span> myDEplotData<span class="p">(</span>mydds<span class="p">,</span> geneid<span class="p">,</span> mergelab<span class="p">)</span>
    ggplot<span class="p">(</span>mydat<span class="p">,</span> aes_string<span class="p">(</span>x<span class="o">=</span>grpvar<span class="p">,</span> y <span class="o">=</span> <span class="s">&quot;geneexp&quot;</span><span class="p">,</span> col <span class="o">=</span> colvar<span class="p">))</span><span class="o">+</span> geom_point<span class="p">()</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
<p>Compare the DE effect of condition within WT compares to sre1d. This suggests that the DE with respect to condition depends on genotype.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>myDEplot<span class="p">(</span>ddsmult<span class="p">,</span> <span class="s">&quot;CNAG_03518&quot;</span><span class="p">,</span> <span class="s">&quot;genotype&quot;</span><span class="p">,</span> <span class="s">&quot;condition&quot;</span><span class="p">,</span> <span class="s">&quot;Label&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/DESeq2_03_deseq_interaction_30_0.png" src="../_images/DESeq2_03_deseq_interaction_30_0.png" />
</div>
</div>
</div>
</div>
<div class="section" id="Exercise:-Compare-the-size-factors-and-dispersion-estimates-between-ddsmult-and-ddsad">
<h2>Exercise: Compare the size factors and dispersion estimates between ddsmult and ddsad<a class="headerlink" href="#Exercise:-Compare-the-size-factors-and-dispersion-estimates-between-ddsmult-and-ddsad" title="Permalink to this headline">Â¶</a></h2>
</div>
<div class="section" id="Estimate-DE-effect-due-to-condition-given-genotype">
<h2>Estimate DE effect due to condition given genotype<a class="headerlink" href="#Estimate-DE-effect-due-to-condition-given-genotype" title="Permalink to this headline">Â¶</a></h2>
<div class="section" id="Define-a-group-variable">
<h3>Define a group variable<a class="headerlink" href="#Define-a-group-variable" title="Permalink to this headline">Â¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[26]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>ddsgrp <span class="o">&lt;-</span> dds2019
<span class="c1">### DESeq2 requires factors</span>
ddsgrp<span class="o">$</span>group <span class="o">&lt;-</span> <span class="kp">as.factor</span><span class="p">(</span><span class="kp">paste0</span><span class="p">(</span>ddsgrp<span class="o">$</span>condition<span class="p">,</span> ddsgrp<span class="o">$</span>genotype<span class="p">))</span>
ddsgrp<span class="o">$</span>group
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<ol class=list-inline>
    <li>pH4WT</li>
    <li>pH4WT</li>
    <li>pH4WT</li>
    <li>pH4WT</li>
    <li>pH4WT</li>
    <li>pH4WT</li>
    <li>pH4sre1d</li>
    <li>pH4sre1d</li>
    <li>pH4sre1d</li>
    <li>pH4sre1d</li>
    <li>pH4sre1d</li>
    <li>pH4sre1d</li>
    <li>pH8WT</li>
    <li>pH8WT</li>
    <li>pH8WT</li>
    <li>pH8WT</li>
    <li>pH8WT</li>
    <li>pH8WT</li>
    <li>pH8sre1d</li>
    <li>pH8sre1d</li>
    <li>pH8sre1d</li>
    <li>pH8sre1d</li>
    <li>pH8sre1d</li>
    <li>pH8sre1d</li>
</ol>

<details>
    <summary style=display:list-item;cursor:pointer>
            <strong>Levels</strong>:
    </summary>
    <ol class=list-inline>
            <li>'pH4sre1d'</li>
            <li>'pH4WT'</li>
            <li>'pH8sre1d'</li>
            <li>'pH8WT'</li>
    </ol>
</details></div>
</div>
</div>
<div class="section" id="Set-design">
<h3>Set design<a class="headerlink" href="#Set-design" title="Permalink to this headline">Â¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[27]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>design<span class="p">(</span>ddsgrp<span class="p">)</span> <span class="o">&lt;-</span> formula<span class="p">(</span><span class="o">~</span>group<span class="p">)</span>
ddsgrp<span class="o">@</span>design
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area highlight-none notranslate"><div class="highlight"><pre>
<span></span>~group
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[28]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>ddsgrp
<span class="c1">### Estimate Size Factors</span>
ddsgrp <span class="o">&lt;-</span> estimateSizeFactors<span class="p">(</span>ddsgrp<span class="p">)</span>
<span class="c1">### Estimate Dispersion parameters (for each gene)</span>
ddsgrp <span class="o">&lt;-</span> estimateDispersions<span class="p">(</span>ddsgrp<span class="p">)</span>
<span class="c1">### Fit NB MLE model</span>
ddsgrpres <span class="o">&lt;-</span> DESeq<span class="p">(</span>ddsgrp<span class="p">)</span>
<span class="c1">### Rlog &quot;normalized&quot; expressions</span>
<span class="c1">#rldmult &lt;- rlog(ddsmult, blind = TRUE)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area highlight-none notranslate"><div class="highlight"><pre>
<span></span>class: DESeqDataSet
dim: 8499 24
metadata(1): version
assays(1): counts
rownames(8499): CNAG_00001 CNAG_00002 ... large_MTrRNA small_MTrRNA
rowData names(0):
colnames(24): 1_2019_P_M1 2_2019_P_M1 ... 23_2019_P_M1 24_2019_P_M1
colData names(22): Label sample_year ... RIN_normal_threshold
  RIN_lowered_threshold
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
using pre-existing size factors
estimating dispersions
found already estimated dispersions, replacing these
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
fitting model and testing
</pre></div></div>
</div>
</div>
<div class="section" id="Generate-results-for-CNAG_03518">
<h3>Generate results for CNAG_03518<a class="headerlink" href="#Generate-results-for-CNAG_03518" title="Permalink to this headline">Â¶</a></h3>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[31]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="c1">### Results for interaction analysis</span>
r0<span class="o">&lt;-</span>results<span class="p">(</span>ddsgrpres<span class="p">,</span> tidy<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="c1">### Contrast treatments within WT strain</span>
r1<span class="o">&lt;-</span>results<span class="p">(</span>ddsgrpres<span class="p">,</span> contrast <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;group&quot;</span><span class="p">,</span> <span class="s">&quot;pH8WT&quot;</span><span class="p">,</span> <span class="s">&quot;pH4WT&quot;</span><span class="p">),</span> tidy<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="c1">### Contrast treatments within sre1d strain</span>
r2<span class="o">&lt;-</span>results<span class="p">(</span>ddsgrpres<span class="p">,</span> contrast <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;group&quot;</span><span class="p">,</span> <span class="s">&quot;pH8sre1d&quot;</span><span class="p">,</span> <span class="s">&quot;pH4sre1d&quot;</span><span class="p">),</span> tidy<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[33]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>r0 <span class="o">%&gt;%</span> filter<span class="p">(</span>row <span class="o">==</span> <span class="s">&quot;CNAG_03518&quot;</span><span class="p">)</span>
r1 <span class="o">%&gt;%</span> filter<span class="p">(</span>row <span class="o">==</span> <span class="s">&quot;CNAG_03518&quot;</span><span class="p">)</span>
r2 <span class="o">%&gt;%</span> filter<span class="p">(</span>row <span class="o">==</span> <span class="s">&quot;CNAG_03518&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<table>
<caption>A data.frame: 1 Ã— 7</caption>
<thead>
    <tr><th scope=col>row</th><th scope=col>baseMean</th><th scope=col>log2FoldChange</th><th scope=col>lfcSE</th><th scope=col>stat</th><th scope=col>pvalue</th><th scope=col>padj</th></tr>
    <tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>CNAG_03518</td><td>399.4864</td><td>-4.15704</td><td>0.1141264</td><td>-36.42487</td><td>1.720166e-290</td><td>3.78948e-288</td></tr>
</tbody>
</table></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<table>
<caption>A data.frame: 1 Ã— 7</caption>
<thead>
    <tr><th scope=col>row</th><th scope=col>baseMean</th><th scope=col>log2FoldChange</th><th scope=col>lfcSE</th><th scope=col>stat</th><th scope=col>pvalue</th><th scope=col>padj</th></tr>
    <tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>CNAG_03518</td><td>399.4864</td><td>-1.106679</td><td>0.1195981</td><td>-9.253317</td><td>2.176323e-20</td><td>9.58358e-20</td></tr>
</tbody>
</table></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<table>
<caption>A data.frame: 1 Ã— 7</caption>
<thead>
    <tr><th scope=col>row</th><th scope=col>baseMean</th><th scope=col>log2FoldChange</th><th scope=col>lfcSE</th><th scope=col>stat</th><th scope=col>pvalue</th><th scope=col>padj</th></tr>
    <tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>CNAG_03518</td><td>399.4864</td><td>-4.198417</td><td>0.1187655</td><td>-35.35049</td><td>9.853941e-274</td><td>1.460354e-271</td></tr>
</tbody>
</table></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>sessionInfo<span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area highlight-none notranslate"><div class="highlight"><pre>
<span></span>R version 3.6.0 (2019-04-26)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 9 (stretch)

Matrix products: default
BLAS:   /usr/lib/openblas-base/libblas.so.3
LAPACK: /usr/lib/libopenblasp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets
[8] methods   base

other attached packages:
 [1] RColorBrewer_1.1-2          dendextend_1.12.0
 [3] DESeq2_1.24.0               SummarizedExperiment_1.14.0
 [5] DelayedArray_0.10.0         BiocParallel_1.18.0
 [7] matrixStats_0.54.0          Biobase_2.44.0
 [9] GenomicRanges_1.36.0        GenomeInfoDb_1.20.0
[11] IRanges_2.18.1              S4Vectors_0.22.0
[13] BiocGenerics_0.30.0         forcats_0.4.0
[15] stringr_1.4.0               dplyr_0.8.1
[17] purrr_0.3.2                 readr_1.3.1
[19] tidyr_0.8.3                 tibble_2.1.2
[21] ggplot2_3.1.1               tidyverse_1.2.1

loaded via a namespace (and not attached):
 [1] colorspace_1.4-1       IRdisplay_0.7.0        htmlTable_1.13.1
 [4] XVector_0.24.0         base64enc_0.1-3        rstudioapi_0.10
 [7] bit64_0.9-7            AnnotationDbi_1.46.0   lubridate_1.7.4
[10] xml2_1.2.0             splines_3.6.0          geneplotter_1.62.0
[13] knitr_1.23             zeallot_0.1.0          IRkernel_1.0.1
[16] Formula_1.2-3          jsonlite_1.6           broom_0.5.2
[19] annotate_1.62.0        cluster_2.0.8          compiler_3.6.0
[22] httr_1.4.0             backports_1.1.4        assertthat_0.2.1
[25] Matrix_1.2-17          lazyeval_0.2.2         cli_1.1.0
[28] acepack_1.4.1          htmltools_0.3.6        tools_3.6.0
[31] gtable_0.3.0           glue_1.3.1             GenomeInfoDbData_1.2.1
[34] Rcpp_1.0.1             cellranger_1.1.0       vctrs_0.1.0
[37] nlme_3.1-139           xfun_0.7               rvest_0.3.4
[40] XML_3.98-1.19          zlibbioc_1.30.0        scales_1.0.0
[43] hms_0.4.2              memoise_1.1.0          gridExtra_2.3
[46] rpart_4.1-15           latticeExtra_0.6-28    stringi_1.4.3
[49] RSQLite_2.1.1          genefilter_1.66.0      checkmate_1.9.3
[52] repr_1.0.1             rlang_0.3.4            pkgconfig_2.0.2
[55] bitops_1.0-6           evaluate_0.14          lattice_0.20-38
[58] labeling_0.3           htmlwidgets_1.3        bit_1.1-14
[61] tidyselect_0.2.5       plyr_1.8.4             magrittr_1.5
[64] R6_2.4.0               generics_0.0.2         Hmisc_4.2-0
[67] pbdZMQ_0.3-3           DBI_1.0.0              pillar_1.4.1
[70] haven_2.1.0            foreign_0.8-71         withr_2.1.2
[73] survival_2.44-1.1      RCurl_1.95-4.12        nnet_7.3-12
[76] modelr_0.1.4           crayon_1.3.4           uuid_0.1-2
[79] viridis_0.5.1          locfit_1.5-9.1         grid_3.6.0
[82] readxl_1.3.1           data.table_1.12.2      blob_1.1.1
[85] digest_0.6.19          xtable_1.8-4           munsell_0.5.0
[88] viridisLite_0.3.0
</pre></div>
</div>
</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="04-deseq2-plots.html" class="btn btn-neutral float-right" title="Generate figures from DESEq2 analysis" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="02_deseq_DE.html" class="btn btn-neutral float-left" title="DESeq2: Basic Differential Expression (DE) analysis" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>