

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>DESeq2: Create count matrix and metadata data frame &mdash; HTS2019 1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../_static/jquery.js"></script>
        <script type="text/javascript" src="../_static/underscore.js"></script>
        <script type="text/javascript" src="../_static/doctools.js"></script>
        <script type="text/javascript" src="../_static/language_data.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "document", "processClass": "math|output_area"}})</script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="DESeq2: Create Data Objects" href="01_deseq_dataobject.html" />
    <link rel="prev" title="Get Star Count of the Pilot 2019 Data" href="8_Pilot2019_getstarcounts-KO.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> HTS2019
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Slides</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../slides.html">Slide sets</a></li>
</ul>
<p class="caption"><span class="caption-text">Quick Intro of Analysis Tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../janice/1_IntroToCompBoot.html">Duke HTS Summer Course 2018</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/2_Intro_to_Unix_and_bash.html">Introduction to Unix and bash</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/3_IntroductionToR.html">Introduction to R</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/4_Rlibraries_and_bioconductor.html">R libraries and Bioconductor</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/5_Tidyverse_HandsOn-Filled.html">Hands-On Tidyverse - with count data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/6_Statistical_Inference.html">Simulations and Statistical Inference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/7_Statistical_Inference_Solutions.html">Estimation and Hypothesis Testing</a></li>
</ul>
<p class="caption"><span class="caption-text">Unix Shell</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix01_File_And_Directory_Solutions.html">Working with Unix</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix02_FileIO_Solutions.html">File I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix03_File_Storage_Solutions.html">File archival and recovery</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix04_Text_Manipulation_Solutinos.html">Working with text</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix05_Variables_Solutions.html">Unix Variables</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix06_Bash_Bioinformatics.html">Using <code class="docutils literal notranslate"><span class="pre">bash</span></code> in bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix07_Capstone_Exercise.html">Unix capstone exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix07_Capstone_Exercise_Solutions.html">Unix capstone exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix_Appendix01_Regular_Expressions.html">The Unix Shell: Regular Expressions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix_Appendix02_Review.html">Unix Review of Basics</a></li>
</ul>
<p class="caption"><span class="caption-text">R Graphics and Data Manipulation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R00_Review_Basics.html">Basics of R (Review)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R00_Review_Basics.html">Basics of R (Review)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R01_Data_Manipulation_Solutions.html">Data Manipulation in the <code class="docutils literal notranslate"><span class="pre">tidyverse</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R02_Tidying_Data_Solutions.html">Tidy Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R03_FileIO.html">File I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R04_Unsupervised_Learning.html">Unsupervised Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R05_Unsupervised_Learning_More_Examples.html">Unsupervised Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R06_Graphics_Overview.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R07_Graphics_Base.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R08_Graphics_ggplot2.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R09_Graphics_Exercise.html">R Graphics Exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R09_Graphics_Exercise_Solutions.html">R Graphics Exercise (Solutions)</a></li>
</ul>
<p class="caption"><span class="caption-text">Analysis of Pilot Data (Gene Count, DESeq2 &amp; DE Analysis)</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="8_Pilot2019_getstarcounts-KO.html">Get Star Count of the Pilot 2019 Data</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">DESeq2: Create count matrix and metadata data frame</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#Objective:-create-a-count-matrix-and-metadata-frame-for-the-2019-pilot-data">Objective: create a count matrix and metadata frame for the 2019 pilot data</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Load-packages">Load packages</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Import-Counts">Import Counts</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Filter-out-2018-libraries-from-count-file">Filter out 2018 libraries from count file</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Import-metadata-file">Import metadata file</a></li>
<li class="toctree-l3"><a class="reference internal" href="#filter-out-2018-samples">filter out 2018 samples</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Add-a-label-column-to-the-counts-object-so-that-it-can-be-merged-with-the-metadata">Add a label column to the counts object so that it can be merged with the metadata</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Add-counts-across-lanes">Add counts across lanes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Check-to-make-sure-that-every-label-in-your-count-file-has-an-entry-in-the-metadata-file">Check to make sure that every label in your count file has an entry in the metadata file</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Save-objects-to-image-file">Save objects to image file</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="01_deseq_dataobject.html">DESeq2: Create Data Objects</a></li>
<li class="toctree-l1"><a class="reference internal" href="02_deseq_DE.html">DESeq2: Basic Differential Expression (DE) analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="03_deseq_interaction.html">DESeq2: Basic Differential Expression (DE) analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="04-deseq2-plots.html">Generate figures from DESEq2 analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">Reproducible Analysis (Git)</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../computing/data_handling_demo.html">Data handling Demo</a></li>
<li class="toctree-l1"><a class="reference internal" href="../computing/git_lesson.html">Git Lesson</a></li>
<li class="toctree-l1"><a class="reference internal" href="../computing/git_in_class.html">Git in class</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">HTS2019</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>DESeq2: Create count matrix and metadata data frame</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/DESeq2/00_deseq_prepare_data.ipynb.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 5ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast,
.nboutput.nblast {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast + .nbinput {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="DESeq2:-Create-count-matrix-and-metadata-data-frame">
<h1>DESeq2: Create count matrix and metadata data frame<a class="headerlink" href="#DESeq2:-Create-count-matrix-and-metadata-data-frame" title="Permalink to this headline">¶</a></h1>
<div class="section" id="Objective:-create-a-count-matrix-and-metadata-frame-for-the-2019-pilot-data">
<h2>Objective: create a count matrix and metadata frame for the 2019 pilot data<a class="headerlink" href="#Objective:-create-a-count-matrix-and-metadata-frame-for-the-2019-pilot-data" title="Permalink to this headline">¶</a></h2>
<div class="section" id="Load-packages">
<h3>Load packages<a class="headerlink" href="#Load-packages" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="kn">library</span><span class="p">(</span>tidyverse<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
Registered S3 methods overwritten by &#39;ggplot2&#39;:
  method         from
  [.quosures     rlang
  c.quosures     rlang
  print.quosures rlang
── <span class="ansi-bold">Attaching packages</span> ─────────────────────────────────────── tidyverse 1.2.1 ──
<span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">ggplot2</span> 3.1.1     <span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">purrr  </span> 0.3.2
<span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">tibble </span> 2.1.2     <span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">dplyr  </span> 0.8.1
<span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">tidyr  </span> 0.8.3     <span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">stringr</span> 1.4.0
<span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">readr  </span> 1.3.1     <span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">forcats</span> 0.4.0
── <span class="ansi-bold">Conflicts</span> ────────────────────────────────────────── tidyverse_conflicts() ──
<span class="ansi-red-fg">✖</span> <span class="ansi-blue-fg">dplyr</span>::<span class="ansi-green-fg">filter()</span> masks <span class="ansi-blue-fg">stats</span>::filter()
<span class="ansi-red-fg">✖</span> <span class="ansi-blue-fg">dplyr</span>::<span class="ansi-green-fg">lag()</span>    masks <span class="ansi-blue-fg">stats</span>::lag()
</pre></div></div>
</div>
</div>
<div class="section" id="Import-Counts">
<h3>Import Counts<a class="headerlink" href="#Import-Counts" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="c1">### Due to time constraints, we skip the import of the count files (see HTS tidyverse notebook)</span>

cntfile <span class="o">&lt;-</span> <span class="s">&#39;/home/jovyan/work/scratch/analysis_output/out/hts-pilot-2019.RData&#39;</span>
<span class="kn">attach</span><span class="p">(</span>cntfile<span class="p">)</span>
tools<span class="o">::</span>md5sum<span class="p">(</span>cntfile<span class="p">)</span>
<span class="kp">ls</span><span class="p">(</span><span class="m">2</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<strong>/home/jovyan/work/scratch/analysis_output/out/hts-pilot-2019.RData:</strong> 'f617195665ef950ffb36a89f2d18a9ca'</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<ol class=list-inline>
    <li>'genecounts'</li>
    <li>'mapresults'</li>
</ol></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="c1">### Look at the first five genes across the libraries</span>

genecounts<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">8</span><span class="p">,</span> <span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span>
<span class="kp">dim</span><span class="p">(</span>genecounts<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<table>
<caption>A tibble: 8 × 5</caption>
<thead>
    <tr><th scope=col>expid</th><th scope=col>CNAG_00001</th><th scope=col>CNAG_00002</th><th scope=col>CNAG_00003</th><th scope=col>CNAG_00004</th></tr>
    <tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
    <tr><td>1_2019_P_M1_S1_L001_ReadsPerGene.out.tab  </td><td>0</td><td>35</td><td>48</td><td>223</td></tr>
    <tr><td>1_2019_P_M1_S1_L002_ReadsPerGene.out.tab  </td><td>0</td><td>43</td><td>46</td><td>227</td></tr>
    <tr><td>1_2019_P_M1_S1_L003_ReadsPerGene.out.tab  </td><td>0</td><td>46</td><td>49</td><td>232</td></tr>
    <tr><td>1_2019_P_M1_S1_L004_ReadsPerGene.out.tab  </td><td>0</td><td>34</td><td>58</td><td>222</td></tr>
    <tr><td>10_2019_P_M1_S10_L001_ReadsPerGene.out.tab</td><td>0</td><td>30</td><td>36</td><td>130</td></tr>
    <tr><td>10_2019_P_M1_S10_L002_ReadsPerGene.out.tab</td><td>0</td><td>37</td><td>37</td><td>117</td></tr>
    <tr><td>10_2019_P_M1_S10_L003_ReadsPerGene.out.tab</td><td>0</td><td>29</td><td>31</td><td>135</td></tr>
    <tr><td>10_2019_P_M1_S10_L004_ReadsPerGene.out.tab</td><td>0</td><td>23</td><td>27</td><td>131</td></tr>
</tbody>
</table></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<ol class=list-inline>
    <li>144</li>
    <li>8500</li>
</ol></div>
</div>
</div>
<div class="section" id="Filter-out-2018-libraries-from-count-file">
<h3>Filter out 2018 libraries from count file<a class="headerlink" href="#Filter-out-2018-libraries-from-count-file" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>genecounts <span class="o">%&gt;%</span>
    filter<span class="p">(</span>str_detect<span class="p">(</span>expid<span class="p">,</span> <span class="s">&quot;2019&quot;</span><span class="p">))</span> <span class="o">-&gt;</span>
        genecounts2019

<span class="kp">dim</span><span class="p">(</span>genecounts2019<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<ol class=list-inline>
    <li>96</li>
    <li>8500</li>
</ol></div>
</div>
</div>
<div class="section" id="Import-metadata-file">
<h3>Import metadata file<a class="headerlink" href="#Import-metadata-file" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>metadtfile <span class="o">&lt;-</span> <span class="s">&#39;/data/hts_2019_data/hts2019_pilot_rawdata/2019_pilot_metadata.tsv&#39;</span>
tools<span class="o">::</span>md5sum<span class="p">(</span>metadtfile<span class="p">)</span>

mtdf<span class="o">&lt;-</span>readr<span class="o">::</span>read_tsv<span class="p">(</span>metadtfile<span class="p">)</span>

mtdf
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<strong>/data/hts_2019_data/hts2019_pilot_rawdata/2019_pilot_metadata.tsv:</strong> '757e63c2306c3c68a70b81dfcdbf7f22'</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
Parsed with column specification:
cols(
  .default = col_character(),
  sample_year = <span class="ansi-green-fg">col_double()</span>,
  enrich_rep = <span class="ansi-green-fg">col_double()</span>,
  RNA_sample_num = <span class="ansi-green-fg">col_double()</span>,
  library_num = <span class="ansi-green-fg">col_double()</span>,
  bio_replicate = <span class="ansi-green-fg">col_double()</span>,
  Nanodrop_260_280 = <span class="ansi-green-fg">col_double()</span>,
  Nanodrop_260_230 = <span class="ansi-green-fg">col_double()</span>,
  Nanodrop_concentration_ng_ul = <span class="ansi-green-fg">col_double()</span>,
  Bioanalyzer_concentration_ng_ul = <span class="ansi-green-fg">col_double()</span>,
  RIN_lowered_threshold = <span class="ansi-green-fg">col_double()</span>
)
See spec(...) for full column specifications.
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<table>
<caption>A spec_tbl_df: 36 × 22</caption>
<thead>
    <tr><th scope=col>Label</th><th scope=col>sample_year</th><th scope=col>group</th><th scope=col>enrich_rep</th><th scope=col>RNA_sample_num</th><th scope=col>genotype</th><th scope=col>condition</th><th scope=col>libprep_person</th><th scope=col>enrichment_method</th><th scope=col>enrichment_short</th><th scope=col>⋯</th><th scope=col>i5_primer</th><th scope=col>i7_primer</th><th scope=col>library_num</th><th scope=col>bio_replicate</th><th scope=col>Nanodrop_260_280</th><th scope=col>Nanodrop_260_230</th><th scope=col>Nanodrop_concentration_ng_ul</th><th scope=col>Bioanalyzer_concentration_ng_ul</th><th scope=col>RIN_normal_threshold</th><th scope=col>RIN_lowered_threshold</th></tr>
    <tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>1_2019_P_M1 </td><td>2019</td><td>P</td><td>1</td><td> 1</td><td>WT   </td><td>pH4</td><td>C</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i501</td><td>i701</td><td> 1</td><td>1</td><td>2.14</td><td>1.52</td><td>293</td><td> 197</td><td>N/A</td><td> 9.8</td></tr>
    <tr><td>2_2019_P_M1 </td><td>2019</td><td>P</td><td>1</td><td> 2</td><td>WT   </td><td>pH4</td><td>C</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i502</td><td>i701</td><td> 2</td><td>2</td><td>2.12</td><td>1.79</td><td>290</td><td> 225</td><td>N/A</td><td> 9.9</td></tr>
    <tr><td>3_2019_P_M1 </td><td>2019</td><td>P</td><td>1</td><td> 3</td><td>WT   </td><td>pH4</td><td>C</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i503</td><td>i701</td><td> 3</td><td>3</td><td>2.11</td><td>2.49</td><td>302</td><td> 241</td><td>N/A</td><td> 9.9</td></tr>
    <tr><td>4_2019_P_M1 </td><td>2019</td><td>P</td><td>1</td><td> 4</td><td>WT   </td><td>pH4</td><td>P</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i504</td><td>i701</td><td> 4</td><td>4</td><td>2.13</td><td>1.15</td><td>296</td><td> 189</td><td>N/A</td><td> 9.7</td></tr>
    <tr><td>5_2019_P_M1 </td><td>2019</td><td>P</td><td>1</td><td> 5</td><td>WT   </td><td>pH4</td><td>P</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i505</td><td>i701</td><td> 5</td><td>5</td><td>2.09</td><td>2.42</td><td>337</td><td> 268</td><td>10 </td><td>10.0</td></tr>
    <tr><td>6_2019_P_M1 </td><td>2019</td><td>P</td><td>1</td><td> 6</td><td>WT   </td><td>pH4</td><td>P</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i506</td><td>i701</td><td> 6</td><td>6</td><td>2.08</td><td>2.40</td><td>319</td><td> 276</td><td>10 </td><td>10.0</td></tr>
    <tr><td>7_2019_P_M1 </td><td>2019</td><td>P</td><td>1</td><td> 7</td><td>sre1d</td><td>pH4</td><td>C</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i507</td><td>i701</td><td> 7</td><td>1</td><td>2.13</td><td>2.23</td><td>232</td><td> 127</td><td>N/A</td><td> 9.9</td></tr>
    <tr><td>8_2019_P_M1 </td><td>2019</td><td>P</td><td>1</td><td> 8</td><td>sre1d</td><td>pH4</td><td>C</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i508</td><td>i701</td><td> 8</td><td>2</td><td>2.08</td><td>2.24</td><td>320</td><td> 311</td><td>N/A</td><td>10.0</td></tr>
    <tr><td>9_2019_P_M1 </td><td>2019</td><td>P</td><td>1</td><td> 9</td><td>sre1d</td><td>pH4</td><td>C</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i501</td><td>i702</td><td> 9</td><td>3</td><td>2.09</td><td>1.46</td><td>342</td><td> 326</td><td>9.6</td><td> 9.6</td></tr>
    <tr><td>10_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>10</td><td>sre1d</td><td>pH4</td><td>P</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i502</td><td>i702</td><td>10</td><td>4</td><td>2.16</td><td>1.25</td><td>262</td><td> 168</td><td>10 </td><td>10.0</td></tr>
    <tr><td>11_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>11</td><td>sre1d</td><td>pH4</td><td>P</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i503</td><td>i702</td><td>11</td><td>5</td><td>2.12</td><td>2.28</td><td>212</td><td> 236</td><td>10 </td><td>10.0</td></tr>
    <tr><td>12_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>12</td><td>sre1d</td><td>pH4</td><td>P</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i504</td><td>i702</td><td>12</td><td>6</td><td>2.12</td><td>2.57</td><td>246</td><td> 312</td><td>10 </td><td>10.0</td></tr>
    <tr><td>13_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>13</td><td>WT   </td><td>pH8</td><td>C</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i505</td><td>i702</td><td>13</td><td>1</td><td>1.93</td><td>2.28</td><td>465</td><td> 436</td><td>N/A</td><td>10.0</td></tr>
    <tr><td>14_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>14</td><td>WT   </td><td>pH8</td><td>C</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i506</td><td>i702</td><td>14</td><td>2</td><td>2.13</td><td>1.88</td><td>170</td><td>  99</td><td>N/A</td><td> 9.9</td></tr>
    <tr><td>15_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>15</td><td>WT   </td><td>pH8</td><td>C</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i507</td><td>i702</td><td>15</td><td>3</td><td>2.08</td><td>0.97</td><td>300</td><td> 190</td><td>N/A</td><td>10.0</td></tr>
    <tr><td>16_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>16</td><td>WT   </td><td>pH8</td><td>P</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i508</td><td>i702</td><td>16</td><td>4</td><td>2.15</td><td>2.13</td><td>236</td><td> 129</td><td>N/A</td><td>10.0</td></tr>
    <tr><td>17_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>17</td><td>WT   </td><td>pH8</td><td>P</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i501</td><td>i703</td><td>17</td><td>5</td><td>2.13</td><td>0.89</td><td>233</td><td> 111</td><td>N/A</td><td>10.0</td></tr>
    <tr><td>18_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>18</td><td>WT   </td><td>pH8</td><td>P</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i502</td><td>i703</td><td>18</td><td>6</td><td>2.14</td><td>2.01</td><td>213</td><td> 134</td><td>N/A</td><td>10.0</td></tr>
    <tr><td>19_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>19</td><td>sre1d</td><td>pH8</td><td>C</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i503</td><td>i703</td><td>19</td><td>1</td><td>2.14</td><td>2.19</td><td>236</td><td>  80</td><td>N/A</td><td>10.0</td></tr>
    <tr><td>20_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>20</td><td>sre1d</td><td>pH8</td><td>C</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i504</td><td>i703</td><td>20</td><td>2</td><td>2.14</td><td>2.39</td><td>172</td><td> 131</td><td>N/A</td><td>10.0</td></tr>
    <tr><td>21_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>21</td><td>sre1d</td><td>pH8</td><td>C</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i505</td><td>i703</td><td>21</td><td>3</td><td>2.01</td><td>2.22</td><td>397</td><td> 295</td><td>N/A</td><td>10.0</td></tr>
    <tr><td>22_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>22</td><td>sre1d</td><td>pH8</td><td>P</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i506</td><td>i703</td><td>22</td><td>4</td><td>2.22</td><td>2.33</td><td>567</td><td> 499</td><td>10 </td><td>10.0</td></tr>
    <tr><td>23_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>23</td><td>sre1d</td><td>pH8</td><td>P</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i507</td><td>i703</td><td>23</td><td>5</td><td>2.13</td><td>2.56</td><td>254</td><td> 206</td><td>N/A</td><td>10.0</td></tr>
    <tr><td>24_2019_P_M1</td><td>2019</td><td>P</td><td>1</td><td>24</td><td>sre1d</td><td>pH8</td><td>P</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i508</td><td>i703</td><td>24</td><td>6</td><td>2.11</td><td>2.16</td><td>215</td><td> 336</td><td>10 </td><td>10.0</td></tr>
    <tr><td>2_2018_P_H1 </td><td>2018</td><td>P</td><td>1</td><td> 2</td><td>WT   </td><td>YPD</td><td>J</td><td>RNaseH</td><td>H</td><td>⋯</td><td>i501</td><td>i704</td><td>25</td><td>2</td><td>  NA</td><td>  NA</td><td> NA</td><td>2120</td><td>10 </td><td>  NA</td></tr>
    <tr><td>3_2018_P_H1 </td><td>2018</td><td>P</td><td>1</td><td> 3</td><td>WT   </td><td>YPD</td><td>J</td><td>RNaseH</td><td>H</td><td>⋯</td><td>i502</td><td>i704</td><td>26</td><td>3</td><td>  NA</td><td>  NA</td><td> NA</td><td>1500</td><td>10 </td><td>  NA</td></tr>
    <tr><td>4_2018_P_H1 </td><td>2018</td><td>P</td><td>1</td><td> 4</td><td>WT   </td><td>YPD</td><td>J</td><td>RNaseH</td><td>H</td><td>⋯</td><td>i503</td><td>i704</td><td>27</td><td>4</td><td>  NA</td><td>  NA</td><td> NA</td><td>1490</td><td>10 </td><td>  NA</td></tr>
    <tr><td>2_2018_P_H2 </td><td>2018</td><td>P</td><td>2</td><td> 2</td><td>WT   </td><td>YPD</td><td>J</td><td>RNaseH</td><td>H</td><td>⋯</td><td>i504</td><td>i704</td><td>28</td><td>2</td><td>  NA</td><td>  NA</td><td> NA</td><td>2120</td><td>10 </td><td>  NA</td></tr>
    <tr><td>3_2018_P_H2 </td><td>2018</td><td>P</td><td>2</td><td> 3</td><td>WT   </td><td>YPD</td><td>J</td><td>RNaseH</td><td>H</td><td>⋯</td><td>i505</td><td>i704</td><td>29</td><td>3</td><td>  NA</td><td>  NA</td><td> NA</td><td>1500</td><td>10 </td><td>  NA</td></tr>
    <tr><td>4_2018_P_H2 </td><td>2018</td><td>P</td><td>2</td><td> 4</td><td>WT   </td><td>YPD</td><td>J</td><td>RNaseH</td><td>H</td><td>⋯</td><td>i506</td><td>i704</td><td>30</td><td>4</td><td>  NA</td><td>  NA</td><td> NA</td><td>1490</td><td>10 </td><td>  NA</td></tr>
    <tr><td>2_2018_P_T1 </td><td>2018</td><td>P</td><td>1</td><td> 2</td><td>WT   </td><td>YPD</td><td>J</td><td>Total </td><td>T</td><td>⋯</td><td>i507</td><td>i704</td><td>31</td><td>2</td><td>  NA</td><td>  NA</td><td> NA</td><td>2120</td><td>10 </td><td>  NA</td></tr>
    <tr><td>3_2018_P_T1 </td><td>2018</td><td>P</td><td>1</td><td> 3</td><td>WT   </td><td>YPD</td><td>J</td><td>Total </td><td>T</td><td>⋯</td><td>i508</td><td>i704</td><td>32</td><td>3</td><td>  NA</td><td>  NA</td><td> NA</td><td>1500</td><td>10 </td><td>  NA</td></tr>
    <tr><td>4_2018_P_T1 </td><td>2018</td><td>P</td><td>1</td><td> 4</td><td>WT   </td><td>YPD</td><td>J</td><td>Total </td><td>T</td><td>⋯</td><td>i501</td><td>i705</td><td>33</td><td>4</td><td>  NA</td><td>  NA</td><td> NA</td><td>1490</td><td>10 </td><td>  NA</td></tr>
    <tr><td>2_2018_P_M1 </td><td>2018</td><td>P</td><td>1</td><td> 2</td><td>WT   </td><td>YPD</td><td>J</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i502</td><td>i705</td><td>34</td><td>2</td><td>  NA</td><td>  NA</td><td> NA</td><td>2120</td><td>10 </td><td>  NA</td></tr>
    <tr><td>3_2018_P_M1 </td><td>2018</td><td>P</td><td>1</td><td> 3</td><td>WT   </td><td>YPD</td><td>J</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i503</td><td>i705</td><td>35</td><td>3</td><td>  NA</td><td>  NA</td><td> NA</td><td>1500</td><td>10 </td><td>  NA</td></tr>
    <tr><td>4_2018_P_M1 </td><td>2018</td><td>P</td><td>1</td><td> 4</td><td>WT   </td><td>YPD</td><td>J</td><td>mRNA  </td><td>M</td><td>⋯</td><td>i504</td><td>i705</td><td>36</td><td>4</td><td>  NA</td><td>  NA</td><td> NA</td><td>1490</td><td>10 </td><td>  NA</td></tr>
</tbody>
</table></div>
</div>
</div>
<div class="section" id="filter-out-2018-samples">
<h3>filter out 2018 samples<a class="headerlink" href="#filter-out-2018-samples" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>mtdf <span class="o">%&gt;%</span>
    filter<span class="p">(</span>str_detect<span class="p">(</span>Label<span class="p">,</span> <span class="s">&quot;2019&quot;</span><span class="p">))</span> <span class="o">-&gt;</span>
        mtdf2019

<span class="kp">dim</span><span class="p">(</span>mtdf2019<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<ol class=list-inline>
    <li>24</li>
    <li>22</li>
</ol></div>
</div>
</div>
<div class="section" id="Add-a-label-column-to-the-counts-object-so-that-it-can-be-merged-with-the-metadata">
<h3>Add a label column to the counts object so that it can be merged with the metadata<a class="headerlink" href="#Add-a-label-column-to-the-counts-object-so-that-it-can-be-merged-with-the-metadata" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>myregex <span class="o">&lt;-</span> <span class="s">&quot;_S[1-9][0-9]{0,2}_L00[1-4]_ReadsPerGene.out.tab&quot;</span>


<span class="c1">### dim before adding label column</span>
<span class="kp">dim</span><span class="p">(</span>genecounts2019<span class="p">)</span>

genecounts2019 <span class="o">%&gt;%</span>
    mutate<span class="p">(</span>Label<span class="o">=</span>str_replace<span class="p">(</span>expid<span class="p">,</span> myregex<span class="p">,</span> <span class="s">&quot;&quot;</span><span class="p">))</span> <span class="o">-&gt;</span>
        genecounts2019

<span class="c1">### dim before adding label column</span>

<span class="kp">dim</span><span class="p">(</span>genecounts2019<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<ol class=list-inline>
    <li>96</li>
    <li>8500</li>
</ol></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<ol class=list-inline>
    <li>96</li>
    <li>8501</li>
</ol></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="c1">### Look at the first five genes across the libraries</span>

genecounts2019<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">8</span><span class="p">,</span> <span class="kt">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">,</span> <span class="kp">ncol</span><span class="p">(</span>genecounts2019<span class="p">))]</span>
<span class="kp">dim</span><span class="p">(</span>genecounts2019<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<table>
<caption>A tibble: 8 × 6</caption>
<thead>
    <tr><th scope=col>expid</th><th scope=col>CNAG_00001</th><th scope=col>CNAG_00002</th><th scope=col>CNAG_00003</th><th scope=col>CNAG_00004</th><th scope=col>Label</th></tr>
    <tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
    <tr><td>1_2019_P_M1_S1_L001_ReadsPerGene.out.tab  </td><td>0</td><td>35</td><td>48</td><td>223</td><td>1_2019_P_M1 </td></tr>
    <tr><td>1_2019_P_M1_S1_L002_ReadsPerGene.out.tab  </td><td>0</td><td>43</td><td>46</td><td>227</td><td>1_2019_P_M1 </td></tr>
    <tr><td>1_2019_P_M1_S1_L003_ReadsPerGene.out.tab  </td><td>0</td><td>46</td><td>49</td><td>232</td><td>1_2019_P_M1 </td></tr>
    <tr><td>1_2019_P_M1_S1_L004_ReadsPerGene.out.tab  </td><td>0</td><td>34</td><td>58</td><td>222</td><td>1_2019_P_M1 </td></tr>
    <tr><td>10_2019_P_M1_S10_L001_ReadsPerGene.out.tab</td><td>0</td><td>30</td><td>36</td><td>130</td><td>10_2019_P_M1</td></tr>
    <tr><td>10_2019_P_M1_S10_L002_ReadsPerGene.out.tab</td><td>0</td><td>37</td><td>37</td><td>117</td><td>10_2019_P_M1</td></tr>
    <tr><td>10_2019_P_M1_S10_L003_ReadsPerGene.out.tab</td><td>0</td><td>29</td><td>31</td><td>135</td><td>10_2019_P_M1</td></tr>
    <tr><td>10_2019_P_M1_S10_L004_ReadsPerGene.out.tab</td><td>0</td><td>23</td><td>27</td><td>131</td><td>10_2019_P_M1</td></tr>
</tbody>
</table></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<ol class=list-inline>
    <li>96</li>
    <li>8501</li>
</ol></div>
</div>
</div>
<div class="section" id="Add-counts-across-lanes">
<h3>Add counts across lanes<a class="headerlink" href="#Add-counts-across-lanes" title="Permalink to this headline">¶</a></h3>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>genecounts2019 <span class="o">%&gt;%</span>
    group_by<span class="p">(</span>Label<span class="p">)</span> <span class="o">%&gt;%</span>
        summarize_each<span class="p">(</span><span class="kp">sum</span><span class="p">,</span> <span class="o">-</span>expid<span class="p">)</span> <span class="o">-&gt;</span>
            cnt2019
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>cnt2019<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">,</span> <span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<table>
<caption>A tibble: 4 × 5</caption>
<thead>
    <tr><th scope=col>Label</th><th scope=col>CNAG_00001</th><th scope=col>CNAG_00002</th><th scope=col>CNAG_00003</th><th scope=col>CNAG_00004</th></tr>
    <tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
    <tr><td>1_2019_P_M1 </td><td>0</td><td>158</td><td>201</td><td>904</td></tr>
    <tr><td>10_2019_P_M1</td><td>0</td><td>119</td><td>131</td><td>513</td></tr>
    <tr><td>11_2019_P_M1</td><td>0</td><td> 90</td><td>121</td><td>573</td></tr>
    <tr><td>12_2019_P_M1</td><td>0</td><td> 81</td><td>151</td><td>533</td></tr>
</tbody>
</table></div>
</div>
</div>
<div class="section" id="Check-to-make-sure-that-every-label-in-your-count-file-has-an-entry-in-the-metadata-file">
<h3>Check to make sure that every label in your count file has an entry in the metadata file<a class="headerlink" href="#Check-to-make-sure-that-every-label-in-your-count-file-has-an-entry-in-the-metadata-file" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="kp">setdiff</span><span class="p">(</span>cnt2019<span class="o">$</span>Label<span class="p">,</span> mtdf2019<span class="o">$</span>Label<span class="p">)</span>
<span class="kp">setdiff</span><span class="p">(</span>mtdf2019<span class="o">$</span>Label<span class="p">,</span> cnt2019<span class="o">$</span>Label<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
</div>
</div>
</div>
<div class="section" id="Save-objects-to-image-file">
<h3>Save objects to image file<a class="headerlink" href="#Save-objects-to-image-file" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>curdir <span class="o">&lt;-</span> <span class="s">&quot;/home/jovyan/work/scratch/analysis_output&quot;</span>
imgdir <span class="o">&lt;-</span> <span class="kp">file.path</span><span class="p">(</span>curdir<span class="p">,</span> <span class="s">&quot;img&quot;</span><span class="p">)</span>

imgfile <span class="o">&lt;-</span> <span class="kp">file.path</span><span class="p">(</span>imgdir<span class="p">,</span> <span class="s">&quot;pilotcnt2019.RData&quot;</span><span class="p">)</span>

imgfile
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
'/home/jovyan/work/scratch/analysis_output/img/pilotcnt2019.RData'</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="kp">save</span><span class="p">(</span>cnt2019<span class="p">,</span> mtdf2019<span class="p">,</span> file <span class="o">=</span> imgfile<span class="p">)</span>

tools<span class="o">::</span>md5sum<span class="p">(</span>imgfile<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<strong>/home/jovyan/work/scratch/analysis_output/img/pilotcnt2019.RData:</strong> '0d1cdfc8bb97baf52ca4f8f33e2c0ae4'</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>sessionInfo<span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area highlight-none notranslate"><div class="highlight"><pre>
<span></span>R version 3.6.0 (2019-04-26)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 9 (stretch)

Matrix products: default
BLAS:   /usr/lib/openblas-base/libblas.so.3
LAPACK: /usr/lib/libopenblasp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] forcats_0.4.0   stringr_1.4.0   dplyr_0.8.1     purrr_0.3.2
[5] readr_1.3.1     tidyr_0.8.3     tibble_2.1.2    ggplot2_3.1.1
[9] tidyverse_1.2.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1       cellranger_1.1.0 plyr_1.8.4       pillar_1.4.1
 [5] compiler_3.6.0   base64enc_0.1-3  tools_3.6.0      zeallot_0.1.0
 [9] digest_0.6.19    uuid_0.1-2       lubridate_1.7.4  jsonlite_1.6
[13] evaluate_0.14    nlme_3.1-139     gtable_0.3.0     lattice_0.20-38
[17] pkgconfig_2.0.2  rlang_0.3.4      cli_1.1.0        rstudioapi_0.10
[21] IRdisplay_0.7.0  IRkernel_1.0.1   haven_2.1.0      withr_2.1.2
[25] xml2_1.2.0       httr_1.4.0       repr_1.0.1       vctrs_0.1.0
[29] hms_0.4.2        generics_0.0.2   grid_3.6.0       tidyselect_0.2.5
[33] glue_1.3.1       R6_2.4.0         readxl_1.3.1     pbdZMQ_0.3-3
[37] modelr_0.1.4     magrittr_1.5     backports_1.1.4  scales_1.0.0
[41] htmltools_0.3.6  rvest_0.3.4      assertthat_0.2.1 colorspace_1.4-1
[45] stringi_1.4.3    lazyeval_0.2.2   munsell_0.5.0    broom_0.5.2
[49] crayon_1.3.4
</pre></div>
</div>
</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="01_deseq_dataobject.html" class="btn btn-neutral float-right" title="DESeq2: Create Data Objects" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="8_Pilot2019_getstarcounts-KO.html" class="btn btn-neutral float-left" title="Get Star Count of the Pilot 2019 Data" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>