

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Get Star Count of the Pilot 2019 Data &mdash; HTS2019 1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../_static/jquery.js"></script>
        <script type="text/javascript" src="../_static/underscore.js"></script>
        <script type="text/javascript" src="../_static/doctools.js"></script>
        <script type="text/javascript" src="../_static/language_data.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "document", "processClass": "math|output_area"}})</script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="DESeq2: Create count matrix and metadata data frame" href="00_deseq_prepare_data.html" />
    <link rel="prev" title="R Graphics Exercise (Solutions)" href="../cliburn/R09_Graphics_Exercise_Solutions.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> HTS2019
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Slides</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../slides.html">Slide sets</a></li>
</ul>
<p class="caption"><span class="caption-text">Quick Intro of Analysis Tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../janice/1_IntroToCompBoot.html">Duke HTS Summer Course 2018</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/2_Intro_to_Unix_and_bash.html">Introduction to Unix and bash</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/3_IntroductionToR.html">Introduction to R</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/4_Rlibraries_and_bioconductor.html">R libraries and Bioconductor</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/5_Tidyverse_HandsOn-Filled.html">Hands-On Tidyverse - with count data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/6_Statistical_Inference.html">Simulations and Statistical Inference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../janice/7_Statistical_Inference_Solutions.html">Estimation and Hypothesis Testing</a></li>
</ul>
<p class="caption"><span class="caption-text">Unix Shell</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix01_File_And_Directory_Solutions.html">Working with Unix</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix02_FileIO_Solutions.html">File I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix03_File_Storage_Solutions.html">File archival and recovery</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix04_Text_Manipulation_Solutinos.html">Working with text</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix05_Variables_Solutions.html">Unix Variables</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix06_Bash_Bioinformatics.html">Using <code class="docutils literal notranslate"><span class="pre">bash</span></code> in bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix07_Capstone_Exercise.html">Unix capstone exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix07_Capstone_Exercise_Solutions.html">Unix capstone exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix_Appendix01_Regular_Expressions.html">The Unix Shell: Regular Expressions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/Unix_Appendix02_Review.html">Unix Review of Basics</a></li>
</ul>
<p class="caption"><span class="caption-text">R Graphics and Data Manipulation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R00_Review_Basics.html">Basics of R (Review)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R00_Review_Basics.html">Basics of R (Review)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R01_Data_Manipulation_Solutions.html">Data Manipulation in the <code class="docutils literal notranslate"><span class="pre">tidyverse</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R02_Tidying_Data_Solutions.html">Tidy Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R03_FileIO.html">File I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R04_Unsupervised_Learning.html">Unsupervised Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R05_Unsupervised_Learning_More_Examples.html">Unsupervised Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R06_Graphics_Overview.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R07_Graphics_Base.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R08_Graphics_ggplot2.html">R Graphics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R09_Graphics_Exercise.html">R Graphics Exercise</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cliburn/R09_Graphics_Exercise_Solutions.html">R Graphics Exercise (Solutions)</a></li>
</ul>
<p class="caption"><span class="caption-text">Analysis of Pilot Data (Gene Count, DESeq2 &amp; DE Analysis)</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Get Star Count of the Pilot 2019 Data</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#Load-paths-and-libraries">Load paths and libraries</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Create-directories">Create directories</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Load-STAR-Count-Data">Load STAR Count Data</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Gather-and-spread-the-genes-to-get-a-count-matrix-(genecounts)">Gather and spread the genes to get a count matrix (genecounts)</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#Gather-and-spread-the-first-four-rows-to-nmisc">Gather and spread the first four rows to nmisc</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Gather-and-spread-the-gene-rows">Gather and spread the gene rows</a></li>
<li class="toctree-l2"><a class="reference internal" href="#merge-in-the-4-misc-counts-and-add-summaries">merge in the 4 misc counts and add summaries</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="00_deseq_prepare_data.html">DESeq2: Create count matrix and metadata data frame</a></li>
<li class="toctree-l1"><a class="reference internal" href="01_deseq_dataobject.html">DESeq2: Create Data Objects</a></li>
<li class="toctree-l1"><a class="reference internal" href="02_deseq_DE.html">DESeq2: Basic Differential Expression (DE) analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="03_deseq_interaction.html">DESeq2: Basic Differential Expression (DE) analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="04-deseq2-plots.html">Generate figures from DESEq2 analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">Reproducible Analysis (Git)</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../computing/data_handling_demo.html">Data handling Demo</a></li>
<li class="toctree-l1"><a class="reference internal" href="../computing/git_lesson.html">Git Lesson</a></li>
<li class="toctree-l1"><a class="reference internal" href="../computing/git_in_class.html">Git in class</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">HTS2019</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>Get Star Count of the Pilot 2019 Data</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/DESeq2/8_Pilot2019_getstarcounts-KO.ipynb.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 5ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast,
.nboutput.nblast {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast + .nbinput {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="Get-Star-Count-of-the-Pilot-2019-Data">
<h1>Get Star Count of the Pilot 2019 Data<a class="headerlink" href="#Get-Star-Count-of-the-Pilot-2019-Data" title="Permalink to this headline">¶</a></h1>
<div class="section" id="Load-paths-and-libraries">
<h2>Load paths and libraries<a class="headerlink" href="#Load-paths-and-libraries" title="Permalink to this headline">¶</a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="kn">library</span><span class="p">(</span>foreach<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>tidyverse<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
Registered S3 methods overwritten by &#39;ggplot2&#39;:
  method         from
  [.quosures     rlang
  c.quosures     rlang
  print.quosures rlang
── <span class="ansi-bold">Attaching packages</span> ─────────────────────────────────────── tidyverse 1.2.1 ──
<span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">ggplot2</span> 3.1.1     <span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">purrr  </span> 0.3.2
<span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">tibble </span> 2.1.2     <span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">dplyr  </span> 0.8.1
<span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">tidyr  </span> 0.8.3     <span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">stringr</span> 1.4.0
<span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">readr  </span> 1.3.1     <span class="ansi-green-fg">✔</span> <span class="ansi-blue-fg">forcats</span> 0.4.0
── <span class="ansi-bold">Conflicts</span> ────────────────────────────────────────── tidyverse_conflicts() ──
<span class="ansi-red-fg">✖</span> <span class="ansi-blue-fg">purrr</span>::<span class="ansi-green-fg">accumulate()</span> masks <span class="ansi-blue-fg">foreach</span>::accumulate()
<span class="ansi-red-fg">✖</span> <span class="ansi-blue-fg">dplyr</span>::<span class="ansi-green-fg">filter()</span>     masks <span class="ansi-blue-fg">stats</span>::filter()
<span class="ansi-red-fg">✖</span> <span class="ansi-blue-fg">dplyr</span>::<span class="ansi-green-fg">lag()</span>        masks <span class="ansi-blue-fg">stats</span>::lag()
<span class="ansi-red-fg">✖</span> <span class="ansi-blue-fg">purrr</span>::<span class="ansi-green-fg">when()</span>       masks <span class="ansi-blue-fg">foreach</span>::when()
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>data_root_dir <span class="o">&lt;-</span> <span class="s">&quot;/data/hts_2019_data/&quot;</span>
raw_fastq_dir <span class="o">&lt;-</span> <span class="kp">paste</span><span class="p">(</span>data_root_dir<span class="p">,</span> <span class="s">&quot;hts2019_pilot_rawdata&quot;</span><span class="p">,</span> sep <span class="o">=</span> <span class="s">&quot;&quot;</span><span class="p">)</span>
metadata_file <span class="o">&lt;-</span> <span class="kp">file.path</span><span class="p">(</span>raw_fastq_dir<span class="p">,</span> <span class="s">&quot;2019_pilot_metadata.tsv&quot;</span><span class="p">)</span>
count_dir <span class="o">&lt;-</span> <span class="kp">paste</span><span class="p">(</span>data_root_dir<span class="p">,</span> <span class="s">&quot;hts2019_pilot_counts&quot;</span><span class="p">,</span> sep <span class="o">=</span> <span class="s">&quot;&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="Create-directories">
<h3>Create directories<a class="headerlink" href="#Create-directories" title="Permalink to this headline">¶</a></h3>
<p>Below is an illustration of our folder structure.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>scratch
└── bioinf_intro
└── analysis_output
    ├── out -&gt; the folder to store all our output data files
    └── img -&gt; the folder to store all our images
</pre></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>curdir <span class="o">&lt;-</span> <span class="s">&quot;/home/jovyan/work/scratch/analysis_output&quot;</span>
outdir <span class="o">&lt;-</span> <span class="kp">file.path</span><span class="p">(</span>curdir<span class="p">,</span> <span class="s">&quot;out&quot;</span><span class="p">)</span>
imgdir <span class="o">&lt;-</span> <span class="kp">file.path</span><span class="p">(</span>curdir<span class="p">,</span> <span class="s">&quot;img&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="kp">system</span><span class="p">(</span><span class="kp">paste</span><span class="p">(</span><span class="s">&quot;mkdir -p&quot;</span><span class="p">,</span> outdir<span class="p">),</span> intern <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="kp">system</span><span class="p">(</span><span class="kp">paste</span><span class="p">(</span><span class="s">&quot;mkdir -p&quot;</span><span class="p">,</span> imgdir<span class="p">),</span> intern <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>count_suffix <span class="o">&lt;-</span> <span class="s">&quot;_ReadsPerGene.out.tab&quot;</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="kp">list.files</span><span class="p">(</span>count_dir<span class="p">,</span> pattern <span class="o">=</span> <span class="kp">paste0</span><span class="p">(</span>count_suffix<span class="p">,</span><span class="s">&quot;$&quot;</span><span class="p">),</span> full.names <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span> <span class="o">-&gt;</span> countfiles
</pre></div>
</div>
</div>
</div>
<div class="section" id="Load-STAR-Count-Data">
<h3>Load STAR Count Data<a class="headerlink" href="#Load-STAR-Count-Data" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="kp">list.files</span><span class="p">(</span>count_dir<span class="p">,</span> pattern <span class="o">=</span> <span class="kp">paste0</span><span class="p">(</span>count_suffix<span class="p">,</span><span class="s">&quot;$&quot;</span><span class="p">),</span> full.names <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span> <span class="o">-&gt;</span> countfiles
countfiles
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<ol class=list-inline>
    <li>'1_2019_P_M1_S1_L001_ReadsPerGene.out.tab'</li>
    <li>'1_2019_P_M1_S1_L002_ReadsPerGene.out.tab'</li>
    <li>'1_2019_P_M1_S1_L003_ReadsPerGene.out.tab'</li>
    <li>'1_2019_P_M1_S1_L004_ReadsPerGene.out.tab'</li>
    <li>'10_2019_P_M1_S10_L001_ReadsPerGene.out.tab'</li>
    <li>'10_2019_P_M1_S10_L002_ReadsPerGene.out.tab'</li>
    <li>'10_2019_P_M1_S10_L003_ReadsPerGene.out.tab'</li>
    <li>'10_2019_P_M1_S10_L004_ReadsPerGene.out.tab'</li>
    <li>'11_2019_P_M1_S11_L001_ReadsPerGene.out.tab'</li>
    <li>'11_2019_P_M1_S11_L002_ReadsPerGene.out.tab'</li>
    <li>'11_2019_P_M1_S11_L003_ReadsPerGene.out.tab'</li>
    <li>'11_2019_P_M1_S11_L004_ReadsPerGene.out.tab'</li>
    <li>'12_2019_P_M1_S12_L001_ReadsPerGene.out.tab'</li>
    <li>'12_2019_P_M1_S12_L002_ReadsPerGene.out.tab'</li>
    <li>'12_2019_P_M1_S12_L003_ReadsPerGene.out.tab'</li>
    <li>'12_2019_P_M1_S12_L004_ReadsPerGene.out.tab'</li>
    <li>'13_2019_P_M1_S13_L001_ReadsPerGene.out.tab'</li>
    <li>'13_2019_P_M1_S13_L002_ReadsPerGene.out.tab'</li>
    <li>'13_2019_P_M1_S13_L003_ReadsPerGene.out.tab'</li>
    <li>'13_2019_P_M1_S13_L004_ReadsPerGene.out.tab'</li>
    <li>'14_2019_P_M1_S14_L001_ReadsPerGene.out.tab'</li>
    <li>'14_2019_P_M1_S14_L002_ReadsPerGene.out.tab'</li>
    <li>'14_2019_P_M1_S14_L003_ReadsPerGene.out.tab'</li>
    <li>'14_2019_P_M1_S14_L004_ReadsPerGene.out.tab'</li>
    <li>'15_2019_P_M1_S15_L001_ReadsPerGene.out.tab'</li>
    <li>'15_2019_P_M1_S15_L002_ReadsPerGene.out.tab'</li>
    <li>'15_2019_P_M1_S15_L003_ReadsPerGene.out.tab'</li>
    <li>'15_2019_P_M1_S15_L004_ReadsPerGene.out.tab'</li>
    <li>'16_2019_P_M1_S16_L001_ReadsPerGene.out.tab'</li>
    <li>'16_2019_P_M1_S16_L002_ReadsPerGene.out.tab'</li>
    <li>'16_2019_P_M1_S16_L003_ReadsPerGene.out.tab'</li>
    <li>'16_2019_P_M1_S16_L004_ReadsPerGene.out.tab'</li>
    <li>'17_2019_P_M1_S17_L001_ReadsPerGene.out.tab'</li>
    <li>'17_2019_P_M1_S17_L002_ReadsPerGene.out.tab'</li>
    <li>'17_2019_P_M1_S17_L003_ReadsPerGene.out.tab'</li>
    <li>'17_2019_P_M1_S17_L004_ReadsPerGene.out.tab'</li>
    <li>'18_2019_P_M1_S18_L001_ReadsPerGene.out.tab'</li>
    <li>'18_2019_P_M1_S18_L002_ReadsPerGene.out.tab'</li>
    <li>'18_2019_P_M1_S18_L003_ReadsPerGene.out.tab'</li>
    <li>'18_2019_P_M1_S18_L004_ReadsPerGene.out.tab'</li>
    <li>'19_2019_P_M1_S19_L001_ReadsPerGene.out.tab'</li>
    <li>'19_2019_P_M1_S19_L002_ReadsPerGene.out.tab'</li>
    <li>'19_2019_P_M1_S19_L003_ReadsPerGene.out.tab'</li>
    <li>'19_2019_P_M1_S19_L004_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_H1_S25_L001_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_H1_S25_L002_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_H1_S25_L003_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_H1_S25_L004_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_H2_S28_L001_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_H2_S28_L002_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_H2_S28_L003_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_H2_S28_L004_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_M1_S34_L001_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_M1_S34_L002_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_M1_S34_L003_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_M1_S34_L004_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_T1_S31_L001_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_T1_S31_L002_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_T1_S31_L003_ReadsPerGene.out.tab'</li>
    <li>'2_2018_P_T1_S31_L004_ReadsPerGene.out.tab'</li>
    <li>'2_2019_P_M1_S2_L001_ReadsPerGene.out.tab'</li>
    <li>'2_2019_P_M1_S2_L002_ReadsPerGene.out.tab'</li>
    <li>'2_2019_P_M1_S2_L003_ReadsPerGene.out.tab'</li>
    <li>'2_2019_P_M1_S2_L004_ReadsPerGene.out.tab'</li>
    <li>'20_2019_P_M1_S20_L001_ReadsPerGene.out.tab'</li>
    <li>'20_2019_P_M1_S20_L002_ReadsPerGene.out.tab'</li>
    <li>'20_2019_P_M1_S20_L003_ReadsPerGene.out.tab'</li>
    <li>'20_2019_P_M1_S20_L004_ReadsPerGene.out.tab'</li>
    <li>'21_2019_P_M1_S21_L001_ReadsPerGene.out.tab'</li>
    <li>'21_2019_P_M1_S21_L002_ReadsPerGene.out.tab'</li>
    <li>'21_2019_P_M1_S21_L003_ReadsPerGene.out.tab'</li>
    <li>'21_2019_P_M1_S21_L004_ReadsPerGene.out.tab'</li>
    <li>'22_2019_P_M1_S22_L001_ReadsPerGene.out.tab'</li>
    <li>'22_2019_P_M1_S22_L002_ReadsPerGene.out.tab'</li>
    <li>'22_2019_P_M1_S22_L003_ReadsPerGene.out.tab'</li>
    <li>'22_2019_P_M1_S22_L004_ReadsPerGene.out.tab'</li>
    <li>'23_2019_P_M1_S23_L001_ReadsPerGene.out.tab'</li>
    <li>'23_2019_P_M1_S23_L002_ReadsPerGene.out.tab'</li>
    <li>'23_2019_P_M1_S23_L003_ReadsPerGene.out.tab'</li>
    <li>'23_2019_P_M1_S23_L004_ReadsPerGene.out.tab'</li>
    <li>'24_2019_P_M1_S24_L001_ReadsPerGene.out.tab'</li>
    <li>'24_2019_P_M1_S24_L002_ReadsPerGene.out.tab'</li>
    <li>'24_2019_P_M1_S24_L003_ReadsPerGene.out.tab'</li>
    <li>'24_2019_P_M1_S24_L004_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_H1_S26_L001_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_H1_S26_L002_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_H1_S26_L003_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_H1_S26_L004_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_H2_S29_L001_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_H2_S29_L002_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_H2_S29_L003_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_H2_S29_L004_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_M1_S35_L001_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_M1_S35_L002_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_M1_S35_L003_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_M1_S35_L004_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_T1_S32_L001_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_T1_S32_L002_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_T1_S32_L003_ReadsPerGene.out.tab'</li>
    <li>'3_2018_P_T1_S32_L004_ReadsPerGene.out.tab'</li>
    <li>'3_2019_P_M1_S3_L001_ReadsPerGene.out.tab'</li>
    <li>'3_2019_P_M1_S3_L002_ReadsPerGene.out.tab'</li>
    <li>'3_2019_P_M1_S3_L003_ReadsPerGene.out.tab'</li>
    <li>'3_2019_P_M1_S3_L004_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_H1_S27_L001_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_H1_S27_L002_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_H1_S27_L003_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_H1_S27_L004_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_H2_S30_L001_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_H2_S30_L002_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_H2_S30_L003_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_H2_S30_L004_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_M1_S36_L001_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_M1_S36_L002_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_M1_S36_L003_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_M1_S36_L004_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_T1_S33_L001_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_T1_S33_L002_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_T1_S33_L003_ReadsPerGene.out.tab'</li>
    <li>'4_2018_P_T1_S33_L004_ReadsPerGene.out.tab'</li>
    <li>'4_2019_P_M1_S4_L001_ReadsPerGene.out.tab'</li>
    <li>'4_2019_P_M1_S4_L002_ReadsPerGene.out.tab'</li>
    <li>'4_2019_P_M1_S4_L003_ReadsPerGene.out.tab'</li>
    <li>'4_2019_P_M1_S4_L004_ReadsPerGene.out.tab'</li>
    <li>'5_2019_P_M1_S5_L001_ReadsPerGene.out.tab'</li>
    <li>'5_2019_P_M1_S5_L002_ReadsPerGene.out.tab'</li>
    <li>'5_2019_P_M1_S5_L003_ReadsPerGene.out.tab'</li>
    <li>'5_2019_P_M1_S5_L004_ReadsPerGene.out.tab'</li>
    <li>'6_2019_P_M1_S6_L001_ReadsPerGene.out.tab'</li>
    <li>'6_2019_P_M1_S6_L002_ReadsPerGene.out.tab'</li>
    <li>'6_2019_P_M1_S6_L003_ReadsPerGene.out.tab'</li>
    <li>'6_2019_P_M1_S6_L004_ReadsPerGene.out.tab'</li>
    <li>'7_2019_P_M1_S7_L001_ReadsPerGene.out.tab'</li>
    <li>'7_2019_P_M1_S7_L002_ReadsPerGene.out.tab'</li>
    <li>'7_2019_P_M1_S7_L003_ReadsPerGene.out.tab'</li>
    <li>'7_2019_P_M1_S7_L004_ReadsPerGene.out.tab'</li>
    <li>'8_2019_P_M1_S8_L001_ReadsPerGene.out.tab'</li>
    <li>'8_2019_P_M1_S8_L002_ReadsPerGene.out.tab'</li>
    <li>'8_2019_P_M1_S8_L003_ReadsPerGene.out.tab'</li>
    <li>'8_2019_P_M1_S8_L004_ReadsPerGene.out.tab'</li>
    <li>'9_2019_P_M1_S9_L001_ReadsPerGene.out.tab'</li>
    <li>'9_2019_P_M1_S9_L002_ReadsPerGene.out.tab'</li>
    <li>'9_2019_P_M1_S9_L003_ReadsPerGene.out.tab'</li>
    <li>'9_2019_P_M1_S9_L004_ReadsPerGene.out.tab'</li>
</ol></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="kp">length</span><span class="p">(</span>countfiles<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
144</div>
</div>
<p>So there are 96 files about 2019 pilot data under this folder. Each file is generated from a fastq file using STAR.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>mycombine <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span>df1<span class="p">,</span> df2<span class="p">)</span> <span class="p">{</span>
    <span class="c1"># Combine two data frames by gene names</span>
    <span class="c1">#</span>
    <span class="c1"># Args:</span>
    <span class="c1">#   df1 (Dataframe): the first count data</span>
    <span class="c1">#   df2 (Dataframe): the second count data</span>
    <span class="c1">#</span>
    <span class="c1"># Returns:</span>
    <span class="c1">#   (Dataframe) The combined data frame of df1 and df2</span>
    full_join<span class="p">(</span>df1<span class="p">,</span> df2<span class="p">,</span> by <span class="o">=</span> <span class="s">&quot;gene&quot;</span><span class="p">)</span>
<span class="p">}</span>


mystarfile <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span>rootdir<span class="p">,</span> stardir<span class="p">)</span> <span class="p">{</span>
    <span class="c1"># Get the absolute paths of a file</span>
    <span class="c1">#</span>
    <span class="c1"># Args:</span>
    <span class="c1">#   rootdir  (Character): the directory of the folder</span>
    <span class="c1">#   stardir (Character): the filename</span>
    <span class="c1">#</span>
    <span class="c1"># Returns:</span>
    <span class="c1">#   (Character) the directory of the input file</span>
    <span class="kp">file.path</span><span class="p">(</span>rootdir<span class="p">,</span> stardir<span class="p">)</span>
<span class="p">}</span>

<span class="c1"># Data type for each column</span>
coltypes <span class="o">&lt;-</span> <span class="kt">list</span><span class="p">(</span>col_character<span class="p">(),</span> col_integer<span class="p">(),</span> col_integer<span class="p">(),</span> col_integer<span class="p">())</span>
</pre></div>
</div>
</div>
<p>read the count files and combine them</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>out <span class="o">&lt;-</span> foreach<span class="p">(</span>file <span class="o">=</span> countfiles<span class="p">,</span> <span class="m">.</span>combine <span class="o">=</span> mycombine<span class="p">)</span> <span class="o">%do%</span> <span class="p">{</span>
    cntfile <span class="o">&lt;-</span> mystarfile<span class="p">(</span>count_dir<span class="p">,</span> <span class="kp">file</span><span class="p">)</span>
    readr<span class="o">::</span>read_tsv<span class="p">(</span>cntfile<span class="p">,</span> col_names <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> col_types <span class="o">=</span> coltypes <span class="p">)</span> <span class="o">%&gt;%</span>
        dplyr<span class="o">::</span>select<span class="p">(</span>X1<span class="p">,</span> X4<span class="p">)</span> <span class="o">%&gt;%</span>
            dplyr<span class="o">::</span>rename_<span class="p">(</span><span class="m">.</span>dots<span class="o">=</span>setNames<span class="p">(</span><span class="kp">names</span><span class="p">(</span><span class="m">.</span><span class="p">),</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;gene&quot;</span><span class="p">,</span> <span class="kp">file</span><span class="p">)))</span>
<span class="p">}</span>

</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
Warning message:
“rename_() is deprecated.
Please use rename() instead

The &#39;programming&#39; vignette or the tidyeval book can help you
to program with rename() : https://tidyeval.tidyverse.org
<span class="ansi-black-intense-fg">This warning is displayed once per session.</span>”
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span><span class="kp">dim</span><span class="p">(</span>out<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<ol class=list-inline>
    <li>8503</li>
    <li>145</li>
</ol></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>out<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">6</span><span class="p">,</span> <span class="m">1</span><span class="o">:</span><span class="m">6</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<table>
<caption>A tibble: 6 × 6</caption>
<thead>
    <tr><th scope=col>gene</th><th scope=col>1_2019_P_M1_S1_L001_ReadsPerGene.out.tab</th><th scope=col>1_2019_P_M1_S1_L002_ReadsPerGene.out.tab</th><th scope=col>1_2019_P_M1_S1_L003_ReadsPerGene.out.tab</th><th scope=col>1_2019_P_M1_S1_L004_ReadsPerGene.out.tab</th><th scope=col>10_2019_P_M1_S10_L001_ReadsPerGene.out.tab</th></tr>
    <tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
    <tr><td>N_unmapped    </td><td>21162</td><td>19644</td><td>19307</td><td>17356</td><td> 6456</td></tr>
    <tr><td>N_multimapping</td><td>95540</td><td>94070</td><td>97787</td><td>98723</td><td>75280</td></tr>
    <tr><td>N_noFeature   </td><td>26923</td><td>26053</td><td>27257</td><td>26775</td><td>26470</td></tr>
    <tr><td>N_ambiguous   </td><td>  606</td><td>  638</td><td>  615</td><td>  607</td><td>  464</td></tr>
    <tr><td>CNAG_04548    </td><td>    0</td><td>    0</td><td>    0</td><td>    0</td><td>    0</td></tr>
    <tr><td>CNAG_07303    </td><td>    0</td><td>    0</td><td>    0</td><td>    0</td><td>    0</td></tr>
</tbody>
</table></div>
</div>
</div>
<div class="section" id="Gather-and-spread-the-genes-to-get-a-count-matrix-(genecounts)">
<h3>Gather and spread the genes to get a count matrix (genecounts)<a class="headerlink" href="#Gather-and-spread-the-genes-to-get-a-count-matrix-(genecounts)" title="Permalink to this headline">¶</a></h3>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>out <span class="o">%&gt;%</span>
    slice<span class="p">(</span><span class="o">-</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">))</span> <span class="o">%&gt;%</span>
    gather<span class="p">(</span>expid<span class="p">,</span> value<span class="p">,</span> <span class="o">-</span>gene<span class="p">)</span> <span class="o">%&gt;%</span>
    spread<span class="p">(</span>gene<span class="p">,</span> value<span class="p">)</span> <span class="o">-&gt;</span> genecounts

genecounts<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">6</span><span class="p">,</span> <span class="m">1</span><span class="o">:</span><span class="m">6</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<table>
<caption>A tibble: 6 × 6</caption>
<thead>
    <tr><th scope=col>expid</th><th scope=col>CNAG_00001</th><th scope=col>CNAG_00002</th><th scope=col>CNAG_00003</th><th scope=col>CNAG_00004</th><th scope=col>CNAG_00005</th></tr>
    <tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
    <tr><td>1_2019_P_M1_S1_L001_ReadsPerGene.out.tab  </td><td>0</td><td>35</td><td>48</td><td>223</td><td>5</td></tr>
    <tr><td>1_2019_P_M1_S1_L002_ReadsPerGene.out.tab  </td><td>0</td><td>43</td><td>46</td><td>227</td><td>7</td></tr>
    <tr><td>1_2019_P_M1_S1_L003_ReadsPerGene.out.tab  </td><td>0</td><td>46</td><td>49</td><td>232</td><td>8</td></tr>
    <tr><td>1_2019_P_M1_S1_L004_ReadsPerGene.out.tab  </td><td>0</td><td>34</td><td>58</td><td>222</td><td>2</td></tr>
    <tr><td>10_2019_P_M1_S10_L001_ReadsPerGene.out.tab</td><td>0</td><td>30</td><td>36</td><td>130</td><td>5</td></tr>
    <tr><td>10_2019_P_M1_S10_L002_ReadsPerGene.out.tab</td><td>0</td><td>37</td><td>37</td><td>117</td><td>7</td></tr>
</tbody>
</table></div>
</div>
</div>
</div>
<div class="section" id="Gather-and-spread-the-first-four-rows-to-nmisc">
<h2>Gather and spread the first four rows to nmisc<a class="headerlink" href="#Gather-and-spread-the-first-four-rows-to-nmisc" title="Permalink to this headline">¶</a></h2>
<p>For nmisc, we will take the first 4 rows of out since those are the summarizing features. Next, we want to transform the data frame so that it is in statistical format (the samples are the rows and the feature types are the columns). Using a combination of gather and spread, we can transpose our matrix into the desired format.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>out <span class="o">%&gt;%</span>
    slice<span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">)</span> <span class="o">%&gt;%</span>
    gather<span class="p">(</span>expid<span class="p">,</span> value<span class="p">,</span> <span class="o">-</span>gene<span class="p">)</span> <span class="o">%&gt;%</span>
    spread<span class="p">(</span>gene<span class="p">,</span> value<span class="p">)</span> <span class="o">%&gt;%</span>
    rename_<span class="p">(</span><span class="m">.</span>dots<span class="o">=</span>setNames<span class="p">(</span><span class="kp">names</span><span class="p">(</span><span class="m">.</span><span class="p">),</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;expid&quot;</span><span class="p">,</span> <span class="s">&quot;namb&quot;</span><span class="p">,</span> <span class="s">&quot;nmulti&quot;</span><span class="p">,</span> <span class="s">&quot;nnofeat&quot;</span><span class="p">,</span><span class="s">&quot;nunmap&quot;</span><span class="p">)))</span> <span class="o">-&gt;</span> nmisc

nmisc<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">6</span><span class="p">,</span> <span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<table>
<caption>A tibble: 6 × 5</caption>
<thead>
    <tr><th scope=col>expid</th><th scope=col>namb</th><th scope=col>nmulti</th><th scope=col>nnofeat</th><th scope=col>nunmap</th></tr>
    <tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
    <tr><td>1_2019_P_M1_S1_L001_ReadsPerGene.out.tab  </td><td>606</td><td>95540</td><td>26923</td><td>21162</td></tr>
    <tr><td>1_2019_P_M1_S1_L002_ReadsPerGene.out.tab  </td><td>638</td><td>94070</td><td>26053</td><td>19644</td></tr>
    <tr><td>1_2019_P_M1_S1_L003_ReadsPerGene.out.tab  </td><td>615</td><td>97787</td><td>27257</td><td>19307</td></tr>
    <tr><td>1_2019_P_M1_S1_L004_ReadsPerGene.out.tab  </td><td>607</td><td>98723</td><td>26775</td><td>17356</td></tr>
    <tr><td>10_2019_P_M1_S10_L001_ReadsPerGene.out.tab</td><td>464</td><td>75280</td><td>26470</td><td> 6456</td></tr>
    <tr><td>10_2019_P_M1_S10_L002_ReadsPerGene.out.tab</td><td>445</td><td>73190</td><td>25919</td><td> 6097</td></tr>
</tbody>
</table></div>
</div>
</div>
<div class="section" id="Gather-and-spread-the-gene-rows">
<h2>Gather and spread the gene rows<a class="headerlink" href="#Gather-and-spread-the-gene-rows" title="Permalink to this headline">¶</a></h2>
<p>For each samples, we want to sum up all the counts, so we can create a variable denoting the number of total genes mapped for each sample by summing across the rows.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>out <span class="o">%&gt;%</span>
    slice<span class="p">(</span><span class="o">-</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">))</span> <span class="o">%&gt;%</span>
    gather<span class="p">(</span>expid<span class="p">,</span> value<span class="p">,</span> <span class="o">-</span>gene<span class="p">)</span> <span class="o">%&gt;%</span>
    spread<span class="p">(</span>gene<span class="p">,</span> value<span class="p">)</span> <span class="o">%&gt;%</span>
    mutate<span class="p">(</span>ngenemap<span class="o">=</span><span class="kp">rowSums</span><span class="p">(</span><span class="m">.</span><span class="p">[</span><span class="m">-1</span><span class="p">]))</span> <span class="o">%&gt;%</span>
    select<span class="p">(</span>expid<span class="p">,</span> ngenemap<span class="p">)</span> <span class="o">-&gt;</span> ngene

ngene<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">6</span><span class="p">,</span> <span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<table>
<caption>A tibble: 6 × 2</caption>
<thead>
    <tr><th scope=col>expid</th><th scope=col>ngenemap</th></tr>
    <tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>1_2019_P_M1_S1_L001_ReadsPerGene.out.tab  </td><td>4660549</td></tr>
    <tr><td>1_2019_P_M1_S1_L002_ReadsPerGene.out.tab  </td><td>4591006</td></tr>
    <tr><td>1_2019_P_M1_S1_L003_ReadsPerGene.out.tab  </td><td>4715846</td></tr>
    <tr><td>1_2019_P_M1_S1_L004_ReadsPerGene.out.tab  </td><td>4681095</td></tr>
    <tr><td>10_2019_P_M1_S10_L001_ReadsPerGene.out.tab</td><td>3261459</td></tr>
    <tr><td>10_2019_P_M1_S10_L002_ReadsPerGene.out.tab</td><td>3208423</td></tr>
</tbody>
</table></div>
</div>
</div>
<div class="section" id="merge-in-the-4-misc-counts-and-add-summaries">
<h2>merge in the 4 misc counts and add summaries<a class="headerlink" href="#merge-in-the-4-misc-counts-and-add-summaries" title="Permalink to this headline">¶</a></h2>
<p>So far, we can create a comprehensive data frame mapresults which will combine ngene with nmisc. This data frame will have summarizing mapping features in addition to proportion features.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>ngene <span class="o">%&gt;%</span>
    full_join<span class="p">(</span>nmisc<span class="p">,</span> by<span class="o">=</span><span class="s">&quot;expid&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
    mutate<span class="p">(</span>depth <span class="o">=</span> <span class="kp">as.integer</span><span class="p">(</span>ngenemap <span class="o">+</span> namb <span class="o">+</span> nmulti <span class="o">+</span> nnofeat <span class="o">+</span> nunmap<span class="p">))</span> <span class="o">%&gt;%</span>
    mutate<span class="p">(</span>prob.gene <span class="o">=</span> ngenemap <span class="o">/</span> depth<span class="p">)</span> <span class="o">%&gt;%</span>
    mutate<span class="p">(</span>prob.nofeat <span class="o">=</span> nnofeat <span class="o">/</span> depth<span class="p">)</span> <span class="o">%&gt;%</span>
    mutate<span class="p">(</span>prob.unique <span class="o">=</span> <span class="p">(</span>ngenemap<span class="o">+</span>nnofeat<span class="p">)</span> <span class="o">/</span> depth<span class="p">)</span> <span class="o">-&gt;</span> mapresults

mapresults<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">6</span><span class="p">,</span> <span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<table>
<caption>A tibble: 6 × 10</caption>
<thead>
    <tr><th scope=col>expid</th><th scope=col>ngenemap</th><th scope=col>namb</th><th scope=col>nmulti</th><th scope=col>nnofeat</th><th scope=col>nunmap</th><th scope=col>depth</th><th scope=col>prob.gene</th><th scope=col>prob.nofeat</th><th scope=col>prob.unique</th></tr>
    <tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>1_2019_P_M1_S1_L001_ReadsPerGene.out.tab  </td><td>4660549</td><td>606</td><td>95540</td><td>26923</td><td>21162</td><td>4804780</td><td>0.9699818</td><td>0.005603378</td><td>0.9755851</td></tr>
    <tr><td>1_2019_P_M1_S1_L002_ReadsPerGene.out.tab  </td><td>4591006</td><td>638</td><td>94070</td><td>26053</td><td>19644</td><td>4731411</td><td>0.9703249</td><td>0.005506391</td><td>0.9758313</td></tr>
    <tr><td>1_2019_P_M1_S1_L003_ReadsPerGene.out.tab  </td><td>4715846</td><td>615</td><td>97787</td><td>27257</td><td>19307</td><td>4860812</td><td>0.9701766</td><td>0.005607499</td><td>0.9757841</td></tr>
    <tr><td>1_2019_P_M1_S1_L004_ReadsPerGene.out.tab  </td><td>4681095</td><td>607</td><td>98723</td><td>26775</td><td>17356</td><td>4824556</td><td>0.9702644</td><td>0.005549733</td><td>0.9758141</td></tr>
    <tr><td>10_2019_P_M1_S10_L001_ReadsPerGene.out.tab</td><td>3261459</td><td>464</td><td>75280</td><td>26470</td><td> 6456</td><td>3370129</td><td>0.9677549</td><td>0.007854299</td><td>0.9756092</td></tr>
    <tr><td>10_2019_P_M1_S10_L002_ReadsPerGene.out.tab</td><td>3208423</td><td>445</td><td>73190</td><td>25919</td><td> 6097</td><td>3314074</td><td>0.9681205</td><td>0.007820888</td><td>0.9759414</td></tr>
</tbody>
</table></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre>
<span></span>outfile <span class="o">&lt;-</span> <span class="kp">file.path</span><span class="p">(</span>outdir<span class="p">,</span> <span class="s">&quot;hts-pilot-2019.RData&quot;</span><span class="p">)</span>
<span class="kp">save</span><span class="p">(</span>mapresults<span class="p">,</span> genecounts<span class="p">,</span> file<span class="o">=</span>outfile<span class="p">)</span>
tools<span class="o">::</span>md5sum<span class="p">(</span>outfile<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<strong>/home/jovyan/work/scratch/analysis_output/out/hts-pilot-2019.RData:</strong> 'f617195665ef950ffb36a89f2d18a9ca'</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="00_deseq_prepare_data.html" class="btn btn-neutral float-right" title="DESeq2: Create count matrix and metadata data frame" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../cliburn/R09_Graphics_Exercise_Solutions.html" class="btn btn-neutral float-left" title="R Graphics Exercise (Solutions)" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>